From 65a8492832b7335ffabd01f5f64d89dec757c260 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hubert=20Figui=C3=A8re?= <hub@figuiere.net>
Date: Mon, 14 Aug 2017 23:57:51 -0400
Subject: Bug 102151 - RIFF: fix an infinite loop cause by an overflow

---
 XMPFiles/source/FormatSupport/RIFF.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/XMPFiles/source/FormatSupport/RIFF.cpp b/XMPFiles/source/FormatSupport/RIFF.cpp
index 4d9a0c1..93a318c 100644
--- a/XMPFiles/source/FormatSupport/RIFF.cpp
+++ b/XMPFiles/source/FormatSupport/RIFF.cpp
@@ -167,7 +167,8 @@ Chunk::Chunk( ContainerChunk* parent, RIFF_MetaHandler* handler, bool skip, Chun
 
 	this->oldPos = file->Offset();
 	this->id = XIO::ReadUns32_LE( file );
-	this->oldSize = XIO::ReadUns32_LE( file ) + 8;
+	this->oldSize = XIO::ReadUns32_LE( file );
+	this->oldSize += 8;
 
 	// Make sure the size is within expected bounds.
 	XMP_Int64 chunkEnd = this->oldPos + this->oldSize;
-- 
cgit v1.2.1

