diff --git a/includes/me_plugin_argument_validate_me_alias.inc b/includes/me_plugin_argument_validate_me_alias.inc
index 9dbc508c1144de9e01c2d56a32d9a3d1a9d96aa9..dc3c1d42ddaaea48e9d6408fb3e8865aaf2115ea 100644
--- a/includes/me_plugin_argument_validate_me_alias.inc
+++ b/includes/me_plugin_argument_validate_me_alias.inc
@@ -88,6 +88,14 @@ class me_plugin_argument_validate_me_alias extends views_plugin_argument_validat
       // and check to be sure.
       foreach ($uid_args as $key => $uid_arg) {
         $uid_args[$key] = _me_check_arg($uid_arg, $this->argument->options['me_validate_user_argument_type'] == 'name', FALSE);
+        //Make sure we only allow access to the current user
+        if (is_numeric($uid_args[$key])) {
+          if ($uid_args[$key] != $GLOBALS['user']->uid) {
+            return FALSE;
+          }
+        } else if ($uid_args[$key] != $GLOBALS['user']->name) {
+            return FALSE;
+        }
       }
 
       $argument = implode($seperator, $uid_args);
