From 5647afb35e4bcba2060148e1a2a47bc43cc240f2 Mon Sep 17 00:00:00 2001
From: Rex Dieter <rdieter@gmail.com>
Date: Wed, 9 May 2018 14:14:47 -0500
Subject: changelog for prior commit

---
 ChangeLog | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index f518c79..d7ce2ad 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,8 @@
 === xdg-utils 1.1.3 ===
 
+2018-05-09
+   * xdg-open: Argument injection in xdg-open open_envvar (BR103807)
+
 2018-05-02
    * xdg-settings: check_browser is broken under kde when just the binary is specified (BR106343)
 
-- 
cgit v1.2.1

