From fe59605d3520bf2ca4e0a963d194f10e9fee5806 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hubert=20Figui=C3=A8re?= <hub@figuiere.net>
Date: Fri, 2 Feb 2018 09:55:53 -0500
Subject: Bug 102484 - Fix an infinite loop in ASF parser.

---
 XMPFiles/source/FormatSupport/ASF_Support.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/XMPFiles/source/FormatSupport/ASF_Support.cpp b/XMPFiles/source/FormatSupport/ASF_Support.cpp
index 709aea5..95fd192 100644
--- a/XMPFiles/source/FormatSupport/ASF_Support.cpp
+++ b/XMPFiles/source/FormatSupport/ASF_Support.cpp
@@ -270,6 +270,8 @@ bool ASF_Support::ReadHeaderObject ( XMP_IO* fileRef, ObjectState& inOutObjectSt
 
 				this->ReadHeaderExtensionObject ( fileRef, inOutObjectState, pos, objectBase );
 
+			} else if (objectBase.size == 0) {
+				break;
 			}
 
 			pos += objectBase.size;
-- 
cgit v1.2.1

