diff --git a/aws_amazon.module b/aws_amazon.module
index b1650a42adbf053a659f4a0f25834dfd9b357338..b12e2869b464126cb0245855361bcc822b96cdad 100644
--- a/aws_amazon.module
+++ b/aws_amazon.module
@@ -76,8 +76,9 @@ function aws_amazon_menu() {
     'type' => MENU_CALLBACK,
   );
   global $base_url;
+   
   //generate a unique URL per site that cannot be faked.
-  $token = hash('sha1', $base_url . variable_get('aws_amazon_access_key', ''));
+  $token = hash('sha1', $base_url . drupal_get_hash_salt() . variable_get('aws_amazon_access_key', drupal_get_hash_salt()));
   
   $items['aws-amazon-cron/' . $token] = array(
     'type' => MENU_CALLBACK,
