From 8e3feea862db68d3ca0886f46cd99fab45d2db7c Mon Sep 17 00:00:00 2001
From: Mauro Carvalho Chehab <mchehab+huawei@kernel.org>
Date: Sat, 16 May 2020 01:32:25 +0200
Subject: Bump version to 3.107

Signed-off-by: Mauro Carvalho Chehab <mchehab+huawei@kernel.org>
---
 Changes    | 6 ++++++
 xawtv.spec | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/Changes b/Changes
index a7e9a75..b5c17c0 100644
--- a/Changes
+++ b/Changes
@@ -1,3 +1,9 @@
+3.107
+=====
+
+* Solve build issues with gcc 10.1
+* Check if file exists before opening it
+
 3.106
 =====
 
diff --git a/xawtv.spec b/xawtv.spec
index cd8d196..bc80fc8 100644
--- a/xawtv.spec
+++ b/xawtv.spec
@@ -1,7 +1,7 @@
 Name:         xawtv
 Group:        Applications/Multimedia
 Autoreqprov:  on
-Version:      3.106
+Version:      3.107
 Release:      0
 License:      GPL
 Summary:      v4l applications
-- 
cgit v1.2.1

