From b7a8b32600e60759a7ad3921ec4a2750bf173482 Mon Sep 17 00:00:00 2001
From: hbkhan <hbkhan@github.com>
Date: Tue, 23 Jun 2020 17:02:33 -0500
Subject: [PATCH] Fix Remote Command Execution

---
 lib/git.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/git.js b/lib/git.js
index acf6297..2a488ff 100644
--- a/lib/git.js
+++ b/lib/git.js
@@ -61,7 +61,7 @@ module.exports.commit = function (files, message, newVer, tagName, callback) {
 
     function (done) {
       cp.exec(
-        [gitApp, "tag", "-a", tagName, "-m", message].join(" "),
+        [gitApp, "tag", "-a", escapeQuotes(tagName), "-m", message].join(" "),
         gitExtra,
         done
       );
