From 75205f93009db3cf8c0b0f4f1fc8ab82d70da8ad Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Peter=20S=CC=8Cirka?= <petersirka@gmail.com>
Date: Wed, 13 Feb 2019 20:29:24 +0100
Subject: [PATCH] Fixed XSS.

---
 themes/admin/public/ui.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/themes/admin/public/ui.js b/themes/admin/public/ui.js
index 2667788..a893966 100755
--- a/themes/admin/public/ui.js
+++ b/themes/admin/public/ui.js
@@ -559,7 +559,7 @@ COMPONENT('grid', 'filter:true;external:false;fillcount:50;filterlabel:Filtering
 			for (var j = 0, jl = columns.length; j < jl; j++) {
 				var column = columns[j];
 				var val = items[i][column.name];
-				m.value = column.template ? column.template(items[i], column) : column.render ? column.render(val, column, items[i]) : val == null ? '' : (column.format ? val.format(column.format) : val);
+				m.value = column.template ? column.template(items[i], column) : column.render ? column.render(val, column, items[i]) : val == null ? '' : Thelpers.encode((column.format ? val.format(column.format) : val));
 				m.index = j;
 				m.align = column.align;
 				m.background = column.background;
