From f13c6b54a879aaa771ec64b5a066b939e8f8e7f0 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 28 May 2017 06:42:25 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/501

---
 coders/jp2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coders/jp2.c b/coders/jp2.c
index 94b775e1ca..b9fe4ab001 100644
--- a/coders/jp2.c
+++ b/coders/jp2.c
@@ -391,7 +391,7 @@ static Image *ReadJP2Image(const ImageInfo *image_info,ExceptionInfo *exception)
         (jp2_image->comps[0].dy != jp2_image->comps[i].dy) ||
         (jp2_image->comps[0].prec != jp2_image->comps[i].prec) ||
         (jp2_image->comps[0].sgnd != jp2_image->comps[i].sgnd) ||
-        (jp2_image->comps[i].data == NULL))
+        ((image->ping == MagickFalse) && (jp2_image->comps[i].data == NULL)))
       {
         opj_destroy_codec(jp2_codec);
         opj_image_destroy(jp2_image);
