From 848b87e5bea4f8473eea85261a5ff922d6ebd2b6 Mon Sep 17 00:00:00 2001
From: Joel Griffith <joel.griffith@elastic.co>
Date: Mon, 26 Oct 2020 10:28:01 -0700
Subject: [PATCH] Ensure workspace files are _in_ the workspace DIR

---
 src/routes.ts | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/routes.ts b/src/routes.ts
index e426faa6b..031fc4b11 100644
--- a/src/routes.ts
+++ b/src/routes.ts
@@ -157,9 +157,12 @@ export const getRoutes = ({
       }
 
       const filePath = path.join(workspaceDir, file);
-
       const hasFile = await exists(filePath);
 
+      if (!filePath.includes(workspaceDir)) {
+        return res.sendStatus(404);
+      }
+
       if (!hasFile) {
         return res.sendStatus(404);
       }
@@ -185,6 +188,10 @@ export const getRoutes = ({
       const filePath = path.join(workspaceDir, file);
       const hasFile = await exists(filePath);
 
+      if (!filePath.includes(workspaceDir)) {
+        return res.sendStatus(404);
+      }
+
       if (!hasFile) {
         return res.sendStatus(404);
       }
