From 759644cf9c9d846b824683a05f3452f7bfacb894 Mon Sep 17 00:00:00 2001
From: Pierre RAMBAUD <pierre.rambaud@prestashop.com>
Date: Mon, 8 Feb 2021 11:35:58 +0100
Subject: [PATCH] Make sure session is cleared using soft logout

---
 classes/Cookie.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/classes/Cookie.php b/classes/Cookie.php
index 109eb85d8fde..11e55131ebe0 100644
--- a/classes/Cookie.php
+++ b/classes/Cookie.php
@@ -262,6 +262,7 @@ public function logout()
      */
     public function mylogout()
     {
+        $this->deleteSession();
         unset(
             $this->_content['id_customer'],
             $this->_content['id_guest'],
@@ -514,7 +515,7 @@ public function deleteSession()
      */
     public function isSessionAlive()
     {
-        if (!isset($this->session_id, $this->session_token)) {
+        if (!isset($this->session_id) || !isset($this->session_token)) {
             return false;
         }
 
