From 25a3703ef2e015bbe1d1f16f6b2f63bb10dd34f4 Mon Sep 17 00:00:00 2001
From: pancake <pancake@nopcode.org>
Date: Wed, 18 Apr 2018 13:16:04 +0200
Subject: [PATCH] Fix invalid free in RAnal.avr

---
 libr/anal/p/anal_avr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libr/anal/p/anal_avr.c b/libr/anal/p/anal_avr.c
index 4e62158513c..e7717beb7ca 100644
--- a/libr/anal/p/anal_avr.c
+++ b/libr/anal/p/anal_avr.c
@@ -1287,7 +1287,7 @@ INST_HANDLER (sbrx) {	// SBRC Rr, b
 			// SBRS Rr, b
 	int b = buf[0] & 0x7;
 	int r = ((buf[0] >> 4) & 0xf) | ((buf[1] & 0x01) << 4);
-	RAnalOp next_op;
+	RAnalOp next_op = {0};
 
 	// calculate next instruction size (call recursively avr_op_analyze)
 	// and free next_op's esil string (we dont need it now)
