From 1352365e8bf5ea49da3dc82b1ccf7ddb659ae960 Mon Sep 17 00:00:00 2001
From: Joas Schilling <coding@schilljs.com>
Date: Thu, 1 Sep 2016 10:17:15 +0200
Subject: [PATCH] Add PhantomJS to gitignore

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 0e9a5dd05372..46e1174b7920 100644
--- a/.gitignore
+++ b/.gitignore
@@ -104,6 +104,7 @@ nbproject
 /build/lib/
 /build/jsdocs/
 /npm-debug.log
+/PhantomJS_*
 
 # puphpet
 puphpet
