From f41efbcac1f9262a161ebc1babfcf55fae9e939a Mon Sep 17 00:00:00 2001
From: Naoyoshi Aikawa <n.aikawa91@gmail.com>
Date: Sun, 8 Nov 2015 22:32:09 +0900
Subject: [PATCH] Escape html to prevent XSS

---
 Gemfile | 3 +++
 main.rb | 5 +++++
 2 files changed, 8 insertions(+)

diff --git a/Gemfile b/Gemfile
index 0c15556..a8e6acd 100644
--- a/Gemfile
+++ b/Gemfile
@@ -23,6 +23,9 @@ gem "engtagger"
 gem "faraday"
 gem "faraday_middleware"
 
+# 自動HTMLエスケープ
+gem "erubis"
+
 # JSON
 gem "json"
 
diff --git a/main.rb b/main.rb
index 2accc9a..0f647e5 100644
--- a/main.rb
+++ b/main.rb
@@ -7,6 +7,11 @@
   require 'sinatra/reloader'
 end
 
+# ERBテンプレートで変数を自動エスケープ
+# cf. http://www.sinatrarb.com/faq.html#auto_escape_html
+require 'erubis'
+set :erb, :escape_html => true
+
 # クラス
 require './class/vine.rb'
 require './class/itunes.rb'
