From c0dc63a49d8605649f1d8e4a96c9b468b0bff660 Mon Sep 17 00:00:00 2001
From: Alexandre Rebert <alex@forallsecure.com>
Date: Tue, 6 Jun 2017 14:19:43 -0400
Subject: [PATCH] Fix heap buffer overflow in de_dotdot

---
 src/libhttpd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libhttpd.c b/src/libhttpd.c
index 2ccb2fb..fa42c10 100644
--- a/src/libhttpd.c
+++ b/src/libhttpd.c
@@ -2410,7 +2410,7 @@ de_dotdot( char* file )
     while ( strncmp( file, "./", 2 ) == 0 )
 	(void) memmove( file, file + 2, strlen( file ) - 1 );
     while ( ( cp = strstr( file, "/./") ) != (char*) 0 )
-	(void) memmove( cp, cp + 2, strlen( file ) - 1 );
+	(void) memmove( cp, cp + 2, strlen( cp ) - 1 );
 
     /* Alternate between removing leading ../ and removing xxx/../ */
     for (;;)
