From b84c2cab55948a5ee70860779b2640913e3ee1ed Mon Sep 17 00:00:00 2001
From: Reini Urban <rurban@cpan.org>
Date: Tue, 31 Dec 2019 08:49:07 +0100
Subject: [PATCH] encode: protect from stack under-flow

From GH #178 fuzzing
---
 src/bits.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bits.c b/src/bits.c
index 014b04fe4..a3692bdc6 100644
--- a/src/bits.c
+++ b/src/bits.c
@@ -953,7 +953,7 @@ bit_write_UMC (Bit_Chain *dat, BITCODE_UMC val)
     if (byte[i] & 0x7f)
       break;
 
-  if (byte[i] & 0x40)
+  if (byte[i] & 0x40 && i > 0)
     i--;
   byte[i] &= 0x7f;
   for (j = 4; j >= i; j--)
