From f3809fc8a97ee305d721662a75f788f9e9d21938 Mon Sep 17 00:00:00 2001
From: Joshua Sangmeister <joshua.sangmeister@gmail.com>
Date: Fri, 20 Nov 2020 13:38:37 +0100
Subject: [PATCH] Fixed HTML attribute validation

---
 server/openslides/utils/validate.py | 38 +++++++++++++++++++++++++----
 1 file changed, 33 insertions(+), 5 deletions(-)

diff --git a/server/openslides/utils/validate.py b/server/openslides/utils/validate.py
index 14dacee5dc..1aa61edb62 100644
--- a/server/openslides/utils/validate.py
+++ b/server/openslides/utils/validate.py
@@ -43,12 +43,40 @@
     "video",
 ]
 
+allowed_attributes = [
+    "align",
+    "alt",
+    "autoplay",
+    "background",
+    "bgcolor",
+    "border",
+    "class",
+    "colspan",
+    "controls",
+    "dir",
+    "height",
+    "hidden",
+    "href",
+    "hreflang",
+    "id",
+    "lang",
+    "loop",
+    "muted",
+    "poster",
+    "preload",
+    "rel",
+    "rowspan",
+    "scope",
+    "sizes",
+    "src",
+    "srcset",
+    "start",
+    "style",
+    "target",
+    "title",
+    "width",
+]
 
-def allow_all(tag: str, name: str, value: str) -> bool:
-    return True
-
-
-allowed_attributes = allow_all
 allowed_styles = [
     "color",
     "background-color",
