From a2ed226637f8e66c9b089784b5e58eccf2e2fb30 Mon Sep 17 00:00:00 2001
From: life <lealife@outlook.com>
Date: Wed, 29 Nov 2017 11:21:35 +0800
Subject: [PATCH] Starred notes XXS

---
 public/js/app/note.js | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/public/js/app/note.js b/public/js/app/note.js
index e649be69..e7ea05cd 100644
--- a/public/js/app/note.js
+++ b/public/js/app/note.js
@@ -956,8 +956,10 @@ Note.renderNote = function(note) {
     if (!note) {
         return;
     }
+    var title = note.Title || '';
+    title = title.replace(/&lt;/g, '<').replace(/&gt;/g, '>');
     // title
-    $("#noteTitle").val(note.Title);
+    $("#noteTitle").val(title);
 
     // 当前正在编辑的
     // tags
@@ -1997,7 +1999,7 @@ Note.renderStars = function(notes) {
     me.starNotesO.html('');
     for (var i = 0; i < notes.length; ++i) {
         var note = notes[i];
-        var t = tt(me.starItemT, note.NoteId, note.Title || getMsg('Untitled'));
+        var t = tt(me.starItemT, note.NoteId, trimTitle(note.Title) || getMsg('Untitled'));
         me.starNotesO.append(t);
     }
 
