From 241988ca28139ad970c1d9717c419f41e360ddb0 Mon Sep 17 00:00:00 2001
From: root <356986351@qq.com>
Date: Wed, 13 Sep 2017 13:49:52 +0000
Subject: [PATCH] fix memory leak in ReadYCBCRImage as SetImageExtent failure

---
 coders/ycbcr.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/coders/ycbcr.c b/coders/ycbcr.c
index a2623e8429..7e7d4b4574 100644
--- a/coders/ycbcr.c
+++ b/coders/ycbcr.c
@@ -212,6 +212,7 @@ static Image *ReadYCBCRImage(const ImageInfo *image_info,
     if (status == MagickFalse)
     {
       quantum_info=DestroyQuantumInfo(quantum_info);
+      canvas_image=DestroyImage(canvas_image);
       return(DestroyImageList(image));
     }
     SetImageColorspace(image,YCbCrColorspace,exception);
