From 86f649affc12b5078efc86d9439d67d98f4cb2f6 Mon Sep 17 00:00:00 2001
From: Eric <eric.espie@combodo.com>
Date: Thu, 27 May 2021 16:13:27 +0200
Subject: [PATCH] =?UTF-8?q?N=C2=B04002=20-=20code=20hardening?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 pages/run_query.php | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pages/run_query.php b/pages/run_query.php
index c6bc61648..b3bfa2161 100644
--- a/pages/run_query.php
+++ b/pages/run_query.php
@@ -112,7 +112,6 @@ function ShowExamples($oP, $sExpression)
 	if ($sEncoding == 'crypted')
 	{
 		// Translate $sExpression into a oql expression
-		$sClearText = base64_decode($sExpression);
 		$oFilter = DBObjectSearch::unserialize($sExpression);
 		$sExpression = $oFilter->ToOQL();
 	}
