From 91187aef36363a870d60b0a3c1bf8507af34c9e4 Mon Sep 17 00:00:00 2001
From: Bob den Otter <bob@twokings.nl>
Date: Wed, 3 Apr 2019 17:56:47 +0200
Subject: [PATCH] Blacklist some common PHP file extensions

---
 src/Config.php | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/Config.php b/src/Config.php
index f2dabbd549..c7473418c9 100644
--- a/src/Config.php
+++ b/src/Config.php
@@ -308,6 +308,10 @@ protected function parseGeneral()
                 // as a regex-like string, and we switched to an array. If we find the old style, fall back to the defaults.
                 unset($general['accept_file_types']);
             }
+
+            // Just.. Say no to these.
+            $general['accept_file_types'] = array_diff($general['accept_file_types'], ['sh', 'asp', 'cgi', 'php', 'php3', 'ph3', 'php4', 'ph4', 'php5', 'ph5' ,'phtm', 'phtml']);
+
             // accept uppercase and lowercase file extensions.
             $general['accept_file_types'] = array_values(
                 array_unique(
