From cb1845743bd11ba74a49b6b522c080df86a17d51 Mon Sep 17 00:00:00 2001
From: Shawn A <spam@rumental.com>
Date: Tue, 16 Jun 2015 09:09:12 -0500
Subject: [PATCH] fix #1059

---
 admin/filebrowser.php | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/admin/filebrowser.php b/admin/filebrowser.php
index f1ef74c8..4ad27195 100644
--- a/admin/filebrowser.php
+++ b/admin/filebrowser.php
@@ -20,7 +20,7 @@
 $subPath = (isset($_GET['path'])) ? $_GET['path'] : "";
 if(!path_is_safe($path,GSDATAUPLOADPATH)) die();
 $returnid = isset($_GET['returnid']) ? var_out($_GET['returnid']) : "";
-$func = (isset($_GET['func'])) ? $_GET['func'] : "";
+$func = (isset($_GET['func'])) ? var_out($_GET['func']) : "";
 $path = tsl($path);
 // check if host uses Linux (used for displaying permissions
 $isUnixHost = (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN' ? false : true);
@@ -52,7 +52,6 @@ function submitLink($funcNum, $url) {
             }
         <?php 
 			if (isset($_GET['func'])){
-			$func = @$_GET['func'];
 		?>
 				if(window.opener){
 					if(typeof window.opener.<?php echo $func; ?> == 'function') {
