From 35c7032723d85eee7318ff6c82f031fa2666b773 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Fri, 21 Jun 2019 17:30:44 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1609

---
 magick/threshold.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/magick/threshold.c b/magick/threshold.c
index a53e839e6c..4a8c6b2e72 100644
--- a/magick/threshold.c
+++ b/magick/threshold.c
@@ -202,7 +202,7 @@ MagickExport Image *AdaptiveThresholdImage(const Image *image,
   threshold_image=CloneImage(image,0,0,MagickTrue,exception);
   if (threshold_image == (Image *) NULL)
     return((Image *) NULL);
-  if (width == 0)
+  if ((width == 0) || (height == 0))
     return(threshold_image);
   if (SetImageStorageClass(threshold_image,DirectClass) == MagickFalse)
     {
