From 87afe070cd6866df7fe80f11b26ef75161de85e0 Mon Sep 17 00:00:00 2001
From: jeanlf <jeanlf@gpac.io>
Date: Thu, 8 Apr 2021 10:09:46 +0200
Subject: [PATCH] fixed #1734

---
 applications/mp4box/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/applications/mp4box/main.c b/applications/mp4box/main.c
index 3f30bfc8b..fc71e2ae2 100644
--- a/applications/mp4box/main.c
+++ b/applications/mp4box/main.c
@@ -3370,7 +3370,7 @@ GF_Err HintFile(GF_ISOFile *file, u32 MTUSize, u32 max_ptime, u32 rtp_rate, u32
 
 		streamType = 0;
 		esd = gf_isom_get_esd(file, i+1, 1);
-		if (esd) {
+		if (esd && esd->decoderConfig) {
 			streamType = esd->decoderConfig->streamType;
 			if (!prev_ocr) {
 				prev_ocr = esd->OCRESID;
