From c5d012a46ae22be9444326aa37969a3f75daa3ba Mon Sep 17 00:00:00 2001
From: Cristy <mikayla-grace@urban-warrior.org>
Date: Mon, 22 Apr 2019 12:43:09 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick6/issues/43

---
 MagickCore/blob.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MagickCore/blob.c b/MagickCore/blob.c
index 7de1023f69..70760a83b6 100644
--- a/MagickCore/blob.c
+++ b/MagickCore/blob.c
@@ -996,6 +996,7 @@ MagickExport void *DetachBlob(BlobInfo *blob_info)
   if (blob_info->mapped != MagickFalse)
     {
       (void) UnmapBlob(blob_info->data,blob_info->length);
+      blob_info->data=NULL;
       RelinquishMagickResource(MapResource,blob_info->length);
     }
   blob_info->mapped=MagickFalse;
