From 1e59b29e520d2beab73e8c78aacd5f1c0d76196d Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Tue, 18 Jun 2019 11:45:11 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1599

---
 coders/cut.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/coders/cut.c b/coders/cut.c
index 2ef35f4c45..070136f228 100644
--- a/coders/cut.c
+++ b/coders/cut.c
@@ -568,6 +568,7 @@ static Image *ReadCUTImage(const ImageInfo *image_info,ExceptionInfo *exception)
   BImgBuff=(unsigned char *) AcquireQuantumMemory((size_t) ldblk,
     sizeof(*BImgBuff));  /*Ldblk was set in the check phase*/
   if(BImgBuff==NULL) goto NoMemory;
+  (void) memset(BImgBuff,0,(size_t) ldblk*sizeof(*BImgBuff));
 
   offset=SeekBlob(image,6 /*sizeof(Header)*/,SEEK_SET);
   if (offset < 0)
