From 7dfa400ded53919d986c5d3d23446a09e0cf481b Mon Sep 17 00:00:00 2001
From: Alexander Sago <kallikrates.koios@gmail.com>
Date: Thu, 15 Jul 2021 06:34:08 -0400
Subject: [PATCH] Fix handling of APNG with 0 delay_den (#313)

---
 AUTHORS                  | 1 +
 lib/extras/codec_apng.cc | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/AUTHORS b/AUTHORS
index d74bca225..bc1ff2a7b 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -16,6 +16,7 @@ Cloudinary Ltd. <*@cloudinary.com>
 Google LLC <*@google.com>
 
 # Individuals:
+Alexander Sago <cagelight@gmail.com>
 Dirk Lemstra <dirk@lemstra.org>
 Jon Sneyers <jon@cloudinary.com>
 Pieter Wuille
diff --git a/lib/extras/codec_apng.cc b/lib/extras/codec_apng.cc
index 48b7653d8..bef59f636 100644
--- a/lib/extras/codec_apng.cc
+++ b/lib/extras/codec_apng.cc
@@ -342,6 +342,8 @@ Status DecodeImageAPNG(Span<const uint8_t> bytes, ThreadPool* pool,
           dop = chunk.p[32];
           bop = chunk.p[33];
 
+          if (!delay_den) delay_den = 100;
+
           if (w0 > cMaxPNGSize || h0 > cMaxPNGSize || x0 > cMaxPNGSize ||
               y0 > cMaxPNGSize || x0 + w0 > w || y0 + h0 > h || dop > 2 ||
               bop > 1) {
