From 87ceeae9b8ee415541d7d71de10675e699a76e5e Mon Sep 17 00:00:00 2001
From: Zachary Huff <zach.huff.386@gmail.com>
Date: Fri, 4 Sep 2020 11:42:55 -0400
Subject: [PATCH] Update profile write

---
 service/profile/profile.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/service/profile/profile.go b/service/profile/profile.go
index 2ba9ecb3..4a2ebac3 100644
--- a/service/profile/profile.go
+++ b/service/profile/profile.go
@@ -237,6 +237,10 @@ func (p *Profile) write() (pth string, err error) {
 
 		if strings.Contains(trimLine, "script-security") ||
 			strings.HasPrefix(trimLine, "log ") ||
+			strings.HasPrefix(trimLine, "log-append ") ||
+			strings.HasPrefix(trimLine, "syslog ") ||
+			strings.HasPrefix(trimLine, "management ") ||
+			strings.HasPrefix(trimLine, "plugin ") ||
 			strings.HasPrefix(trimLine, "up ") ||
 			strings.HasPrefix(trimLine, "down ") ||
 			strings.HasPrefix(trimLine, "route-pre-down ") ||
