From 01843366d6a7b96e22ad7bb67f3df7d9fd4d5d74 Mon Sep 17 00:00:00 2001
From: Dirk Lemstra <dirk@git.imagemagick.org>
Date: Mon, 15 May 2017 21:24:24 +0200
Subject: [PATCH] Fixed incorrect call to DestroyImage reported in #491.

---
 magick/image.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/magick/image.c b/magick/image.c
index 09cbb64456..24ec768b49 100644
--- a/magick/image.c
+++ b/magick/image.c
@@ -833,7 +833,7 @@ MagickExport Image *CloneImage(const Image *image,const size_t columns,
         sizeof(*clone_image->colormap));
       if (clone_image->colormap == (PixelPacket *) NULL)
         {
-          clone_image=DestroyImage(clone_image);
+          image=(Image *) RelinquishMagickMemory(image);
           ThrowImageException(ResourceLimitError,"MemoryAllocationFailed");
         }
       (void) CopyMagickMemory(clone_image->colormap,image->colormap,length*
