From 976da329ff03ecdfca3030eb6efe3c85e6db9999 Mon Sep 17 00:00:00 2001
From: Durran Jordan <durran@gmail.com>
Date: Thu, 4 Jun 2015 00:19:42 -0400
Subject: [PATCH] Use \A \z for checking regex on legal

---
 lib/bson/object_id.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/bson/object_id.rb b/lib/bson/object_id.rb
index d3931446..b8503f3b 100644
--- a/lib/bson/object_id.rb
+++ b/lib/bson/object_id.rb
@@ -282,7 +282,7 @@ def from_time(time, options = {})
       #
       # @since 2.0.0
       def legal?(string)
-        string.to_s =~ /^[0-9a-f]{24}$/i ? true : false
+        string.to_s =~ /\A[0-9a-f]{24}\z/i ? true : false
       end
 
       # Executes the provided block only if the size of the provided object is
