From 428c3d9adfb446e8492b1c2b7affb3d34072ff46 Mon Sep 17 00:00:00 2001
From: Alexandre Delaunay <delaunay.alexandre@gmail.com>
Date: Mon, 8 Nov 2021 11:27:21 +0100
Subject: [PATCH] die on insecure file request, fix #85

---
 front/send.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/front/send.php b/front/send.php
index 2dc8c86..937c971 100644
--- a/front/send.php
+++ b/front/send.php
@@ -54,6 +54,7 @@
       echo "Security attack !!!";
       Toolbox::logDebug("[Plugin barcode][security][sendfile] ".
                $_SESSION["glpiname"]." try to get a non standard file : ".$filename);
+      exit;
    }
 
    $file = $docDir.'/'.$filename;
