From 140ee8d2204b383279f854802b27bdb41c1d5d1a Mon Sep 17 00:00:00 2001
From: Thomas BERNARD <miniupnp@free.fr>
Date: Sat, 12 Apr 2014 10:05:42 +0200
Subject: [PATCH] minissdpd.c: Initialize pointers to NULL (fix)

---
 minissdpd/minissdpd.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/minissdpd/minissdpd.c b/minissdpd/minissdpd.c
index 2421a42e..5e79293e 100644
--- a/minissdpd/minissdpd.c
+++ b/minissdpd/minissdpd.c
@@ -658,6 +658,7 @@ void processRequest(struct reqelem * req)
 			syslog(LOG_ERR, "cannot allocate memory");
 			goto error;
 		}
+		memset(newserv, 0, sizeof(struct service));	/* set pointers to NULL */
 		if(containsForbiddenChars(p, l)) {
 			syslog(LOG_ERR, "bad request (st contains forbidden chars)");
 			goto error;
