From 5ea8129f74c80cbf1f68b9083c745cc8a685485d Mon Sep 17 00:00:00 2001
From: Elijah Fowler <j.elijah.fowler@gmail.com>
Date: Wed, 6 Sep 2017 11:31:51 -0500
Subject: [PATCH] Fix possible XSS bug in help dialog

---
 Web/Help.aspx.cs | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/Web/Help.aspx.cs b/Web/Help.aspx.cs
index b912b791..5df0a9e8 100644
--- a/Web/Help.aspx.cs
+++ b/Web/Help.aspx.cs
@@ -1,8 +1,9 @@
-using System;
-using System.Web.UI;
 using mojoPortal.Business.WebHelpers;
 using mojoPortal.Web.Framework;
 using Resources;
+using System;
+using System.Web;
+using System.Web.UI;
 
 namespace mojoPortal.Web.UI.Pages
 {
@@ -61,7 +62,7 @@ protected void ShowHelp()
 					Resource.HelpNoHelpAvailable;
 			}
 
-			litHelp.Text = helpText;
+			litHelp.Text = HttpUtility.HtmlDecode(SecurityHelper.RemoveMarkup(helpText));
 		}
 	}
-}
+}
\ No newline at end of file
