From c68402fa7e279d42886d5951d1ea8ac2facc1ea5 Mon Sep 17 00:00:00 2001
From: oy <Tom_Adams@web.de>
Date: Fri, 10 Apr 2020 18:58:44 +0200
Subject: [PATCH] changed a check

---
 src/engine/server/server.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/engine/server/server.cpp b/src/engine/server/server.cpp
index e3ac6cb22c..f01f7cc64a 100644
--- a/src/engine/server/server.cpp
+++ b/src/engine/server/server.cpp
@@ -496,8 +496,8 @@ int CServer::SendMsg(CMsgPacker *pMsg, int Flags, int ClientID)
 	if(!pMsg)
 		return -1;
 
-	// drop packet to dummy client
-	if(0 <= ClientID && ClientID < MAX_CLIENTS && GameServer()->IsClientBot(ClientID))
+	// drop invalid packet
+	if(ClientID != -1 && (ClientID < 0 || ClientID >= MAX_CLIENTS || m_aClients[ClientID].m_State == CClient::STATE_EMPTY || m_aClients[ClientID].m_Quitting))
 		return 0;
 
 	mem_zero(&Packet, sizeof(CNetChunk));
