From bcd4ce8cba1272fca52f2331c08f2e3ac19cbbef Mon Sep 17 00:00:00 2001
From: Madhura Jayaratne <madhura.cj@gmail.com>
Date: Sat, 27 Feb 2016 09:56:12 +1100
Subject: [PATCH] Fix XSS in normalization.js

Signed-off-by: Madhura Jayaratne <madhura.cj@gmail.com>
---
 js/normalization.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/normalization.js b/js/normalization.js
index 606ab899135..5e2b8802f66 100644
--- a/js/normalization.js
+++ b/js/normalization.js
@@ -82,7 +82,7 @@ function goTo2NFStep1() {
             $("#mainContent #extra").html(data.extra);
             $("#mainContent #newCols").html('');
             if (data.subText !== '') {
-                $('.tblFooters').html('<input type="submit" value="' + PMA_messages.strDone + '" onclick="processDependencies(\'' + data.primary_key + '\');">');
+                $('.tblFooters').html('<input type="submit" value="' + PMA_messages.strDone + '" onclick="processDependencies(\'' + escapeJsString(escapeHtml(data.primary_key)) + '\');">');
             } else {
                 if (normalizeto === '3nf') {
                     $("#mainContent #newCols").html(PMA_messages.strToNextStep);
