From 7f02eca670f29c00a1d2c305e96febae6ce5d37b Mon Sep 17 00:00:00 2001
From: plusvic <plusvic@gmail.com>
Date: Mon, 9 Jan 2017 13:17:39 +0100
Subject: [PATCH] Make sure that loop depth is greater than 0 before
 decrementing it.

---
 libyara/grammar.y | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/libyara/grammar.y b/libyara/grammar.y
index 446d37410..bf3c88b4d 100644
--- a/libyara/grammar.y
+++ b/libyara/grammar.y
@@ -1017,8 +1017,11 @@ expression
       }
     | _FOR_ for_expression error
       {
-        compiler->loop_depth--;
-        compiler->loop_identifier[compiler->loop_depth] = NULL;
+        if (compiler->loop_depth > 0)
+        {
+          compiler->loop_depth--;
+          compiler->loop_identifier[compiler->loop_depth] = NULL;
+        }
       }
     | _FOR_ for_expression _IDENTIFIER_ _IN_
       {
