From df50e967dbd27b13875625b7dd3189294619b071 Mon Sep 17 00:00:00 2001
From: Joas Schilling <coding@schilljs.com>
Date: Thu, 1 Sep 2016 10:17:15 +0200
Subject: [PATCH] Add PhantomJS to gitignore

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 63a34beb978d..0811a12f44ba 100644
--- a/.gitignore
+++ b/.gitignore
@@ -107,6 +107,7 @@ nbproject
 /build/lib/
 /build/jsdocs/
 /npm-debug.log
+/PhantomJS_*
 
 # puphpet
 puphpet
