From 880051eaa8ba32d1b257c87a7d8798a93561bfd3 Mon Sep 17 00:00:00 2001
From: David Chiles <david@davidchiles.org>
Date: Mon, 23 Jan 2017 13:21:49 -0800
Subject: [PATCH] Updated chatsecure

---
 ChatSecure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ChatSecure b/ChatSecure
index 5144dc6..a19dea4 160000
--- a/ChatSecure
+++ b/ChatSecure
@@ -1 +1 @@
-Subproject commit 5144dc67a1820b931fffa3ab561027a364d77e2e
+Subproject commit a19dea42bfe8d9076e2a4547e9b9a63f4981285b
