From 5e6be61e42448f503e75e287dc332b1ecbf2a665 Mon Sep 17 00:00:00 2001
From: "K. Reid Wightman" <wightman@digitalbond.com>
Date: Tue, 17 Oct 2017 22:03:49 -0400
Subject: [PATCH] Fix a null pointer dereference, if initial walloc() fails or
 wrealloc() fails

---
 src/cgi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cgi.c b/src/cgi.c
index 85b0cfc0..ee4e213e 100644
--- a/src/cgi.c
+++ b/src/cgi.c
@@ -137,7 +137,7 @@ PUBLIC bool cgiHandler(Webs *wp)
     if (strchr(wp->query, '=') == NULL) {
         query = sclone(wp->query);
         websDecodeUrl(query, query, strlen(query));
-        for (cp = stok(query, " ", &tok); cp != NULL; ) {
+        for (cp = stok(query, " ", &tok); cp != NULL && argp != NULL; ) {
             *(argp+n) = cp;
             trace(5, "ARG[%d] %s", n, argp[n-1]);
             n++;
