From 539851c66778f68a244633985f6f8d0df94ea3b3 Mon Sep 17 00:00:00 2001
From: objsys <eday@obj-sys.com>
Date: Wed, 12 Aug 2020 15:26:00 -0400
Subject: [PATCH] fixed missing return status test error

---
 util/cbor2json.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/util/cbor2json.c b/util/cbor2json.c
index 5b44edf..55a58c3 100644
--- a/util/cbor2json.c
+++ b/util/cbor2json.c
@@ -112,6 +112,7 @@ static int cbor2json (OSCTXT* pCborCtxt, OSCTXT* pJsonCtxt)
    case OSRTCBOR_UTF8STR: {
       OSUTF8CHAR* utf8str;
       ret = rtCborDecDynUTF8Str (pCborCtxt, ub, (char**)&utf8str);
+      if (0 != ret) return LOG_RTERR (pCborCtxt, ret);
 
       ret = rtJsonEncStringValue (pJsonCtxt, utf8str);
       rtxMemFreePtr (pCborCtxt, utf8str);
