From 809439e20e273ce0d25c1d04e111dcf6011eb566 Mon Sep 17 00:00:00 2001
From: Faisal Salman <f@faisalman.com>
Date: Fri, 12 Feb 2021 09:53:26 +0700
Subject: [PATCH] Fix potential ReDoS vulnerability as reported by Doyensec

---
 src/ua-parser.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ua-parser.js b/src/ua-parser.js
index e6a7114..63a9f68 100755
--- a/src/ua-parser.js
+++ b/src/ua-parser.js
@@ -620,7 +620,7 @@
             /android.+[;\/]\s*(Q[T|M][\d\w]+)\s+build/i                         // Verizon Tablet
             ], [MODEL, [VENDOR, 'Verizon'], [TYPE, TABLET]], [
 
-            /android.+[;\/]\s+(Barnes[&\s]+Noble\s+|BN[RT])(V?.*)\s+build/i     // Barnes & Noble Tablet
+            /android.+[;\/]\s+(Barnes[&\s]+Noble\s+|BN[RT])(\S(?:.*\S)?)\s+build/i     // Barnes & Noble Tablet
             ], [[VENDOR, 'Barnes & Noble'], MODEL, [TYPE, TABLET]], [
 
             /android.+[;\/]\s+(TM\d{3}.*\b)\s+build/i                           // Barnes & Noble Tablet
@@ -694,7 +694,7 @@
             /(android[\w\.\s\-]{0,9});.+build/i                                 // Generic Android Device
             ], [MODEL, [VENDOR, 'Generic']], [
 
-            /(phone)/i,
+            /(phone)/i
             ], [[TYPE, MOBILE]]
         ],
 
