From f83f911e19e27cbac1ccce7471d96642241dd759 Mon Sep 17 00:00:00 2001
From: Jonathan Claudius <jclaudius@mozilla.com>
Date: Thu, 1 Feb 2018 00:00:48 -0500
Subject: [PATCH] Add bug fix for #270068

---
 lib/rubygems/package.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/rubygems/package.rb b/lib/rubygems/package.rb
index 3ac5662b6820..9d78544f92ec 100644
--- a/lib/rubygems/package.rb
+++ b/lib/rubygems/package.rb
@@ -424,7 +424,7 @@ def install_location filename, destination_dir # :nodoc:
     destination = File.expand_path destination
 
     raise Gem::Package::PathError.new(destination, destination_dir) unless
-      destination.start_with? destination_dir
+      destination.start_with? destination_dir + '/'
 
     destination.untaint
     destination
