From 3b081dd10a1731f8cd981bebeac0e775fb217acf Mon Sep 17 00:00:00 2001
From: Chris Kelley <chris@organicbeemedia.com>
Date: Wed, 16 Dec 2020 11:23:34 -0600
Subject: [PATCH] escape title output

---
 includes/admin/metaboxes.php | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/includes/admin/metaboxes.php b/includes/admin/metaboxes.php
index 02853fd..209266c 100644
--- a/includes/admin/metaboxes.php
+++ b/includes/admin/metaboxes.php
@@ -972,7 +972,7 @@ public function config_tab( $post ) {
 
         </div>
         </div>
-        <?php 
+        <?php
 
         // Output an upgrade notice
         Envira_Gallery_Notice_Admin::get_instance()->display_inline_notice(
@@ -1164,7 +1164,7 @@ public function misc_tab( $post ) {
                             <label for="envira-config-title"><?php _e( 'Gallery Title', 'envira-gallery-lite' ); ?></label>
                         </th>
                         <td>
-                            <input id="envira-config-title" type="text" name="_envira_gallery[title]" value="<?php echo $this->get_config( 'title', $this->get_config_default( 'title' ) ); ?>" />
+                            <input id="envira-config-title" type="text" name="_envira_gallery[title]" value="<?php echo esc_html( $this->get_config( 'title', $this->get_config_default( 'title' ) ) ); ?>" />
                             <p class="description"><?php _e( 'Internal gallery title for identification in the admin.', 'envira-gallery-lite' ); ?></p>
                         </td>
                     </tr>
@@ -1321,7 +1321,7 @@ public function lite_videos_tab( $post ) { ?>
 		</div>
 
 		<?php
-				
+
     }
 
     /**
