From 2f267f89f957088197f4b1fc254632d1645b415d Mon Sep 17 00:00:00 2001
From: Gilles Boccon-Gibod <bok@bok.net>
Date: Sat, 9 Sep 2017 10:42:03 -0700
Subject: [PATCH] fix for #183

---
 Source/C++/Core/Ap4AtomSampleTable.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Source/C++/Core/Ap4AtomSampleTable.cpp b/Source/C++/Core/Ap4AtomSampleTable.cpp
index 77d11191..ec3b0806 100644
--- a/Source/C++/Core/Ap4AtomSampleTable.cpp
+++ b/Source/C++/Core/Ap4AtomSampleTable.cpp
@@ -140,8 +140,10 @@ AP4_AtomSampleTable::GetSample(AP4_Ordinal index,
     AP4_UI32 cts_offset = 0;
     AP4_UI64 dts        = 0;
     AP4_UI32 duration   = 0;
-    result = m_SttsAtom->GetDts(index, dts, &duration);
-    if (AP4_FAILED(result)) return result;
+    if (m_SttsAtom) {
+        result = m_SttsAtom->GetDts(index, dts, &duration);
+        if (AP4_FAILED(result)) return result;
+    }
     sample.SetDuration(duration);
     sample.SetDts(dts);
     if (m_CttsAtom == NULL) {
