From 8187d2d8fd010d2d6b1a3a8edd935beec404dddc Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Fri, 21 Jun 2019 18:49:58 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1610

---
 MagickCore/pixel-accessor.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MagickCore/pixel-accessor.h b/MagickCore/pixel-accessor.h
index 8af0e369cf..56fba6c978 100644
--- a/MagickCore/pixel-accessor.h
+++ b/MagickCore/pixel-accessor.h
@@ -111,7 +111,7 @@ static inline PixelTrait GetPixelCbTraits(const Image *magick_restrict image)
 static inline Quantum GetPixelChannel(const Image *magick_restrict image,
   const PixelChannel channel,const Quantum *magick_restrict pixel)
 {
-  if (image->channel_map[channel].traits == UndefinedPixelTrait)
+  if (image->channel_map[image->channel_map[channel].offset].traits == UndefinedPixelTrait)
     return((Quantum) 0);
   return(pixel[image->channel_map[channel].offset]);
 }
