From 545e0586320ae6f16204ba548cf533ef75dc2fbd Mon Sep 17 00:00:00 2001
From: Daniel Marzini <daniel.marzini@iubenda.com>
Date: Tue, 12 May 2020 11:35:16 +0200
Subject: [PATCH] fix pecl security bug onto the test script

---
 test.php | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/test.php b/test.php
index 63d6c98..1e6e832 100644
--- a/test.php
+++ b/test.php
@@ -49,7 +49,9 @@
 
 				<?php
 				if ( ! empty( $_POST['url'] ) )
-					$url = filter_var( $_POST['url'], FILTER_SANITIZE_URL );
+          if (substr($_POST['url'], 0, 4) == "http") {
+					   $url = filter_var( $_POST['url'], FILTER_SANITIZE_URL );
+          }
 				else
 					$url = '';
 
