From 5a7505ddec956fdc9e9c071ae5089865559174f1 Mon Sep 17 00:00:00 2001
From: Colin Verot <colin@verot.net>
Date: Wed, 4 Dec 2019 13:34:43 +0100
Subject: [PATCH] blacklist .phar extension

---
 src/class.upload.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/class.upload.php b/src/class.upload.php
index 6a66a89..71192c7 100755
--- a/src/class.upload.php
+++ b/src/class.upload.php
@@ -3065,7 +3065,7 @@ function process($server_path = null) {
                 }
                 // if the file is text based, or has a dangerous extension, we rename it as .txt
                 if ((((substr($this->file_src_mime, 0, 5) == 'text/' && $this->file_src_mime != 'text/rtf') || strpos($this->file_src_mime, 'javascript') !== false)  && (substr($file_src_name, -4) != '.txt'))
-                    || preg_match('/\.(php|php5|php4|php3|phtml|pl|py|cgi|asp|js)$/i', $this->file_src_name)
+                    || preg_match('/\.(php|php5|php4|php3|phtml|pl|py|cgi|asp|js|phar)$/i', $this->file_src_name)
                     || $this->file_force_extension && empty($file_src_name_ext)) {
                     $this->file_src_mime = 'text/plain';
                     if ($this->file_src_name_ext) $file_src_name_body = $file_src_name_body . '.' . $this->file_src_name_ext;
