From 824d17f424bf77d17af89c18c3dc807a3199567e Mon Sep 17 00:00:00 2001
From: Cerdic <cedric@yterium.com>
Date: Mon, 8 Apr 2019 13:57:54 +0200
Subject: [PATCH] sanitizer var_memotri avant de l'utiliser (G0uz)

---
 ecrire/inc_version.php | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/ecrire/inc_version.php b/ecrire/inc_version.php
index 357ad1175f3..46bb4dc4c04 100644
--- a/ecrire/inc_version.php
+++ b/ecrire/inc_version.php
@@ -547,7 +547,10 @@
 	if (!function_exists('session_set')) {
 		include_spip('inc/session');
 	}
-	session_set($t, _request($t));
+	$t = preg_replace(",\W,","_", $t);
+	if ($v = _request($t)) {
+		session_set($t, $v);
+	}
 }
 
 /**
