From 65d717415a271242b9a30a5330d4eabac1c1a837 Mon Sep 17 00:00:00 2001
From: Carsten Schmitz <carsten.schmitz@limesurvey.org>
Date: Fri, 26 Jun 2015 08:20:43 +0200
Subject: [PATCH] Fixed issue #9720: SQL injection vulnerability in data entry

---
 application/controllers/admin/dataentry.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/application/controllers/admin/dataentry.php b/application/controllers/admin/dataentry.php
index d6d623ccdb..35ed84c120 100644
--- a/application/controllers/admin/dataentry.php
+++ b/application/controllers/admin/dataentry.php
@@ -1716,7 +1716,7 @@ public function insert()
                         {
                             if (isset($usesleft) && $usesleft<=1)
                             {
-                                $utquery .= "SET usesleft=usesleft-1, completed='$submitdate'\n";
+                                $utquery .= "SET usesleft=usesleft-1, completed=".dbQuoteAll($submitdate);
                             }
                             else
                             {
