From 5aba27604d957e960d8069d85ccaf868f8a7b07a Mon Sep 17 00:00:00 2001
From: jeanlf <jeanlf@gpac.io>
Date: Mon, 4 Jan 2021 11:18:27 +0100
Subject: [PATCH] fixed #1661

---
 src/isomedia/isom_store.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/isomedia/isom_store.c b/src/isomedia/isom_store.c
index ee2b2cfaf2..b3aea7b693 100644
--- a/src/isomedia/isom_store.c
+++ b/src/isomedia/isom_store.c
@@ -102,6 +102,9 @@ void CleanWriters(GF_List *writers)
 {
 	while (gf_list_count(writers)) {
 		TrackWriter *writer = (TrackWriter*)gf_list_get(writers, 0);
+		//in case we have an error in the middle of file write, remove our created stco and stsc from sample table
+		gf_list_del_item(writer->stbl->child_boxes, writer->stco);
+		gf_list_del_item(writer->stbl->child_boxes, writer->stsc);
 		gf_isom_box_del(writer->stco);
 		gf_isom_box_del((GF_Box *)writer->stsc);
 		gf_free(writer);
