From 0010d28de1b15d51db3976080e26357fa7144436 Mon Sep 17 00:00:00 2001
From: Nicolas Vanhoren <nicolas.vanhoren@gmail.com>
Date: Tue, 19 Oct 2021 23:22:02 +0200
Subject: [PATCH] Fix #5

---
 src/forEachLimit.mjs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/forEachLimit.mjs b/src/forEachLimit.mjs
index 08841a7..e6c44fe 100644
--- a/src/forEachLimit.mjs
+++ b/src/forEachLimit.mjs
@@ -35,7 +35,7 @@ import mapLimit from './mapLimit.mjs'
  */
 async function forEachLimit (iterable, iteratee, concurrency) {
   await mapLimit(iterable, async (v, i, t) => {
-    iteratee(v, i, t)
+    await iteratee(v, i, t)
   }, concurrency)
 }
 
