From b173a352397877775c51c9a0e9d59eb6ce24c455 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 20 Feb 2016 09:52:34 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/131

---
 coders/mat.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/coders/mat.c b/coders/mat.c
index b81dc6880e..5091a98769 100644
--- a/coders/mat.c
+++ b/coders/mat.c
@@ -936,6 +936,7 @@ RestoreMSCWarning
   }
       }
     } while(z-- >= 2);
+    quantum_info=DestroyQuantumInfo(quantum_info);
 ExitLoop:
 
 
