From 99e5aa0050224289cfe64c9036f38ce2531bf633 Mon Sep 17 00:00:00 2001
From: Franck Danard <fdanard@sangoma.com>
Date: Tue, 3 Sep 2019 14:13:03 +0000
Subject: [PATCH] fix int options

---
 Contactmanager.class.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Contactmanager.class.php b/Contactmanager.class.php
index 414225a..b140778 100644
--- a/Contactmanager.class.php
+++ b/Contactmanager.class.php
@@ -975,7 +975,7 @@ public function myShowPage() {
 			break;
 			case "showgroup":
 			case "addgroup":
-				if ($action == "showgroup" && !empty((int) $_REQUEST['group'])) {
+				if ($action == "showgroup" && !empty($_REQUEST['group'])) {
 					$group = $this->getGroupByID((int) $_REQUEST['group']);
 					$entries = $this->getEntriesByGroupID((int) $_REQUEST['group']);
 				}
@@ -984,7 +984,7 @@ public function myShowPage() {
 			break;
 			case "showentry":
 			case "addentry":
-				if (!empty((int) $_REQUEST['group'])) {
+				if (!empty($_REQUEST['group'])) {
 					$group = $this->getGroupByID((int) $_REQUEST['group']);
 
 					if ($action == "showentry" && !empty($_REQUEST['entry'])) {
