From 2e1cbd0034cf0041f832ba81d07c24db886782d8 Mon Sep 17 00:00:00 2001
From: Hanno <hanno@gentoo.org>
Date: Sat, 14 Nov 2015 23:06:12 +0100
Subject: [PATCH] Fix use after free error.

---
 dbdimp.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/dbdimp.c b/dbdimp.c
index d507588..acdfee8 100644
--- a/dbdimp.c
+++ b/dbdimp.c
@@ -2085,10 +2085,6 @@ static int my_login(pTHX_ SV* dbh, imp_dbh_t *imp_dbh)
   }
   result = mysql_dr_connect(dbh, imp_dbh->pmysql, mysql_socket, host, port, user,
 			  password, dbname, imp_dbh) ? TRUE : FALSE;
-  if (fresh && !result) {
-      /* Prevent leaks, but do not free in case of a reconnect. See #97625 */
-      Safefree(imp_dbh->pmysql);
-  }
   return result;
 }
 
@@ -2142,9 +2138,12 @@ int dbd_db_login(SV* dbh, imp_dbh_t* imp_dbh, char* dbname, char* user,
 
   if (!my_login(aTHX_ dbh, imp_dbh))
   {
-    if(imp_dbh->pmysql)
+    if(imp_dbh->pmysql) {
         do_error(dbh, mysql_errno(imp_dbh->pmysql),
                 mysql_error(imp_dbh->pmysql) ,mysql_sqlstate(imp_dbh->pmysql));
+        Safefree(imp_dbh->pmysql);
+
+    }
     return FALSE;
   }
 
