From 6d55ba265637d6adf0ba7e9c9ef11187d1ec2f5b Mon Sep 17 00:00:00 2001
From: Stanislav Malyshev <stas@php.net>
Date: Mon, 5 Sep 2016 18:01:35 -0700
Subject: [PATCH] Fix bug #73007: add locale length check

---
 ext/intl/msgformat/msgformat_format.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ext/intl/msgformat/msgformat_format.c b/ext/intl/msgformat/msgformat_format.c
index 25c9619133d6..9b6df38ee398 100644
--- a/ext/intl/msgformat/msgformat_format.c
+++ b/ext/intl/msgformat/msgformat_format.c
@@ -117,6 +117,8 @@ PHP_FUNCTION( msgfmt_format_message )
 		RETURN_FALSE;
 	}
 
+	INTL_CHECK_LOCALE_LEN(slocale_len);
+
 	msgformat_data_init(&mfo->mf_data TSRMLS_CC);
 
 	if(pattern && pattern_len) {
