From 45acb4efbcaa57a20ac5127dc976538671fce018 Mon Sep 17 00:00:00 2001
From: Dev Sanghani <devsanghani@gmail.com>
Date: Wed, 10 Feb 2021 13:09:54 +0530
Subject: [PATCH] [Core] Check Start/TLS cert

---
 src/core/imap/MCIMAPSession.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/core/imap/MCIMAPSession.cpp b/src/core/imap/MCIMAPSession.cpp
index 734d74c89..38e73c20b 100644
--- a/src/core/imap/MCIMAPSession.cpp
+++ b/src/core/imap/MCIMAPSession.cpp
@@ -682,6 +682,13 @@ void IMAPSession::connect(ErrorCode * pError)
             * pError = ErrorTLSNotAvailable;
             goto close;
         }
+            
+        mIsCertificateValid = checkCertificate();
+        if (isCheckCertificateEnabled() && !mIsCertificateValid) {
+            * pError = ErrorCertificate;
+            goto close;
+        }
+            
         break;
 
         case ConnectionTypeTLS:
