From 664b1de4e47f298df90dd8972861ff76280aebf6 Mon Sep 17 00:00:00 2001
From: Pierre RAMBAUD <pierre.rambaud@prestashop.com>
Date: Mon, 23 Mar 2020 11:26:52 +0100
Subject: [PATCH] Escape exception message when rendering PrestaShopException

---
 classes/exception/PrestaShopException.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/classes/exception/PrestaShopException.php b/classes/exception/PrestaShopException.php
index 6d3ccd0a955d..1b16e52e57da 100644
--- a/classes/exception/PrestaShopException.php
+++ b/classes/exception/PrestaShopException.php
@@ -230,7 +230,7 @@ protected function getExtendedMessage($html = true)
 
         return sprintf(
             $format,
-            $this->getMessage(),
+            Tools::safeOutput($this->getMessage(), true),
             $this->getLine(),
             ltrim(str_replace(array(_PS_ROOT_DIR_, '\\'), array('', '/'), $this->getFile()), '/')
         );
