From 3e975c69cce21092c2ac9aa7a2f9dc40446f5ccd Mon Sep 17 00:00:00 2001
From: Kannan Goundan <kannan@cakoose.com>
Date: Tue, 3 Dec 2019 20:28:40 -0800
Subject: [PATCH] writeHead: Don't end() response

Backport of #265
---
 src/webjs.coffee | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/webjs.coffee b/src/webjs.coffee
index 42361c5..41cc5c8 100644
--- a/src/webjs.coffee
+++ b/src/webjs.coffee
@@ -45,9 +45,6 @@ fake_response = (req, res) ->
             try
                 res.write(r.join('\r\n'))
             catch x
-            try
-                res.end()
-            catch x
         res.setHeader = (k, v) -> headers[k] = v
 
 
