From 93b8ac558752d09e1c56d4f1bc82631316fa9c82 Mon Sep 17 00:00:00 2001
From: JerikoOne <jeriko.one@gmx.us>
Date: Sat, 7 Jul 2018 15:55:34 -0500
Subject: [PATCH] Ensure UID in fetch_uidl

---
 pop.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pop.c b/pop.c
index 4f3a592a8d..c52585abe7 100644
--- a/pop.c
+++ b/pop.c
@@ -202,6 +202,10 @@ static int fetch_uidl(char *line, void *data)
     endp++;
   memmove(line, endp, strlen(endp) + 1);
 
+  /* uid must be at least be 1 byte */
+  if (strlen(line) == 0)
+    return -1;
+
   for (i = 0; i < ctx->msgcount; i++)
     if (mutt_str_strcmp(line, ctx->hdrs[i]->data) == 0)
       break;
