From 47e8e6ceef979327614d0b8f0c76c6ecb18e09cf Mon Sep 17 00:00:00 2001
From: dirk <dirk@git.imagemagick.org>
Date: Wed, 19 Oct 2016 07:28:09 +0200
Subject: [PATCH] Fixed memory leak.

---
 coders/psd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/coders/psd.c b/coders/psd.c
index 47e85927f4..ee2f6c5298 100644
--- a/coders/psd.c
+++ b/coders/psd.c
@@ -1739,6 +1739,8 @@ ModuleExport MagickBooleanType ReadPSDLayers(Image *image,
             if (image->debug != MagickFalse)
               (void) LogMagickEvent(CoderEvent,GetMagickModule(),
                 "      layer data is empty");
+            if (layer_info[i].info != (StringInfo *) NULL)
+              layer_info[i].info=DestroyStringInfo(layer_info[i].info);
             continue;
           }
 
