From 0bfc0bf38a7758b27c5c446fec5e3b905e5a54ab Mon Sep 17 00:00:00 2001
From: Jakub Onderka <jakub.onderka@gmail.com>
Date: Fri, 4 Dec 2020 16:20:02 +0100
Subject: [PATCH] fix: [security] XSS in authkey comment field

---
 .../genericElements/SingleViews/Fields/genericField.ctp         | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/View/Elements/genericElements/SingleViews/Fields/genericField.ctp b/app/View/Elements/genericElements/SingleViews/Fields/genericField.ctp
index 748d59c778..6f7a8cb121 100644
--- a/app/View/Elements/genericElements/SingleViews/Fields/genericField.ctp
+++ b/app/View/Elements/genericElements/SingleViews/Fields/genericField.ctp
@@ -3,7 +3,7 @@ if (!empty($field['raw'])) {
     $string = $field['raw'];
 } else {
     $value = Hash::extract($data, $field['path']);
-    $string = empty($value[0]) ? '' : $value[0];
+    $string = empty($value[0]) ? '' : h($value[0]);
 }
 if (!empty($field['url'])) {
     if (!empty($field['url_vars'])) {
