From cebb67bd16a8296121201805332365ffccb29638 Mon Sep 17 00:00:00 2001
From: David Cooke <me@dave.lc>
Date: Fri, 26 Feb 2021 11:41:20 +0000
Subject: [PATCH] Merge pull request from GHSA-fw57-f7mq-9q85

---
 src/authentication/views.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/authentication/views.py b/src/authentication/views.py
index 5dcb9425..b0cfb329 100644
--- a/src/authentication/views.py
+++ b/src/authentication/views.py
@@ -150,7 +150,7 @@ def post(self, request, *args, **kwargs):
                     code.delete()
                     return self.issue_token(user)
 
-        return self.issue_token(user)
+        return FormattedResponse(status=HTTP_401_UNAUTHORIZED, d={'reason': 'login_failed'}, m='login_failed')
 
 
 class RegenerateBackupCodesView(APIView):
