From 26841d9c81f27ad23b2f6e4bd5eaec7f2f58dfe0 Mon Sep 17 00:00:00 2001
From: "Erwin Rommel H. Satingin" <winskie@gmail.com>
Date: Sun, 15 Jan 2017 16:31:00 +0800
Subject: [PATCH] Fixed fm_selected parameter exploit

---
 inc/files/files.ctrl.php | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/inc/files/files.ctrl.php b/inc/files/files.ctrl.php
index 66a473a5b8..e7ecda94e8 100644
--- a/inc/files/files.ctrl.php
+++ b/inc/files/files.ctrl.php
@@ -343,8 +343,15 @@
  * @global Filelist
  */
 $selected_Filelist = new Filelist( $fm_FileRoot, $ads_list_path );
+
+// Prevent directory traversal using '..'
+$re = '/\/?\.\.\/+/';
 foreach( $fm_selected as $l_source_path )
 {
+	if( preg_match( $re, $l_source_path ) )
+	{
+		debug_die( 'Invalid fm_selected parameter value' );
+	}
 	$selected_Filelist->add_by_subpath( urldecode($l_source_path), true );
 }
 
