From c8c6a0f123d5e35c173125365c97e2c0fc7eca42 Mon Sep 17 00:00:00 2001
From: Dirk Lemstra <dirk@git.imagemagick.org>
Date: Sat, 7 Jan 2017 16:56:49 +0100
Subject: [PATCH] Fix improper cast that could cause an overflow as
 demonstrated in #347.

---
 coders/psd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coders/psd.c b/coders/psd.c
index 9249e2e965..75ebd1d483 100644
--- a/coders/psd.c
+++ b/coders/psd.c
@@ -1691,7 +1691,7 @@ ModuleExport MagickBooleanType ReadPSDLayers(Image *image,
             /*
               Layer name.
             */
-            length=(MagickSizeType) ReadBlobByte(image);
+            length=(MagickSizeType) (unsigned char) ReadBlobByte(image);
             combined_length+=length+1;
             if (length > 0)
               (void) ReadBlob(image,(size_t) length++,layer_info[i].name);
