From 5da9f99b2d81713b1e36016b47ce656a33648bc7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?C=C3=A9dric=20Anne?= <cedric.anne@gmail.com>
Date: Thu, 14 Feb 2019 21:35:00 +0100
Subject: [PATCH] Fix password forget token check; fixes #5386

---
 inc/user.class.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/inc/user.class.php b/inc/user.class.php
index f5e86039fd7..b4ca2a9eced 100755
--- a/inc/user.class.php
+++ b/inc/user.class.php
@@ -4392,7 +4392,7 @@ static function showPasswordForgetChangeForm($token) {
          'FROM'   => self::getTable(),
          'WHERE'  => [
             'password_forget_token'       => $token,
-            new \QueryExpression('NOW() < ADDDATE(password_forget_token_date, INTERVAL 1 DAY))')
+            new \QueryExpression('NOW() < ADDDATE(' . $DB->quoteName('password_forget_token_date') . ', INTERVAL 1 DAY)')
          ]
       ]);
 
