From f0232a2a45dfd003c1faf6079497895df3ab0ee1 Mon Sep 17 00:00:00 2001
From: Dirk Lemstra <dirk@git.imagemagick.org>
Date: Mon, 15 May 2017 21:18:13 +0200
Subject: [PATCH] Fixed incorrect call to WriteBlob reported in #490.

---
 coders/png.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/coders/png.c b/coders/png.c
index c194d08585..cfd6844006 100644
--- a/coders/png.c
+++ b/coders/png.c
@@ -4782,10 +4782,11 @@ static Image *ReadOneJNGImage(MngInfo *mng_info,
           (void) LogMagickEvent(CoderEvent,GetMagickModule(),
             "    Copying JDAT chunk data to color_blob.");
 
-        (void) WriteBlob(color_image,length,chunk);
-
         if (length != 0)
-          chunk=(unsigned char *) RelinquishMagickMemory(chunk);
+          {
+            (void) WriteBlob(color_image,length,chunk);
+            chunk=(unsigned char *) RelinquishMagickMemory(chunk);
+          }
 
         continue;
       }
