From 0377b2152c0e59c8f35297f9a9b6ee335a62d963 Mon Sep 17 00:00:00 2001
From: FusionPBX <markjcrane@gmail.com>
Date: Wed, 3 Nov 2021 09:38:01 -0600
Subject: [PATCH] Force the fax_page_size to only preset allowed values.

---
 app/fax/fax_send.php | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/app/fax/fax_send.php b/app/fax/fax_send.php
index 9e1c651f4f..451db849d0 100644
--- a/app/fax/fax_send.php
+++ b/app/fax/fax_send.php
@@ -310,9 +310,13 @@ function fax_split_dtmf(&$fax_number, &$fax_dtmf){
 				$page_height = 14; //in
 				break;
 			case 'letter' :
+				$page_width = 8.5; //in
+				$page_height = 11; //in
+				break;
 			default	:
 				$page_width = 8.5; //in
 				$page_height = 11; //in
+				$fax_page_size = 'letter';
 		}
 
 		//set resolution
