From f398b5b45c019f873a54fe25c815dbaaf963728b Mon Sep 17 00:00:00 2001
From: Julio Montoya <gugli100@gmail.com>
Date: Fri, 14 May 2021 08:11:55 +0200
Subject: [PATCH] Add api_protect_course_script()

---
 main/plagiarism/compilatio/upload.php | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/main/plagiarism/compilatio/upload.php b/main/plagiarism/compilatio/upload.php
index c6e3e2da6c6..dbf8d3378cf 100644
--- a/main/plagiarism/compilatio/upload.php
+++ b/main/plagiarism/compilatio/upload.php
@@ -10,6 +10,8 @@
 
 api_set_more_memory_and_time_limits();
 
+api_protect_course_script();
+
 $courseId = api_get_course_int_id();
 $courseInfo = api_get_course_info();
 $compilatio = new Compilatio();
