From 73b6c35cede48ccb2e29dbe62c2d77488e5e70b7 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 16 Jul 2017 20:33:35 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/571

---
 coders/cals.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/coders/cals.c b/coders/cals.c
index 70e02132b9..de32652264 100644
--- a/coders/cals.c
+++ b/coders/cals.c
@@ -557,6 +557,7 @@ static MagickBooleanType WriteCALSImage(const ImageInfo *image_info,
   group4_image=CloneImage(image,0,0,MagickTrue,exception);
   if (group4_image == (Image *) NULL)
     {
+      write_info=DestroyImageInfo(write_info);
       (void) CloseBlob(image);
       return(MagickFalse);
     }
@@ -565,6 +566,7 @@ static MagickBooleanType WriteCALSImage(const ImageInfo *image_info,
   group4_image=DestroyImage(group4_image);
   if (group4 == (unsigned char *) NULL)
     {
+      write_info=DestroyImageInfo(write_info);
       (void) CloseBlob(image);
       return(MagickFalse);
     }
