From 2aced17651fb684959a6e04b1465a8329b3d5268 Mon Sep 17 00:00:00 2001
From: Arjun Shibu <arjunshibu1999@gmail.com>
Date: Thu, 17 Dec 2020 22:31:18 +0530
Subject: [PATCH] Security fix for Prototype Pollution

---
 src/index.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/index.js b/src/index.js
index c2f04ed..b92a850 100644
--- a/src/index.js
+++ b/src/index.js
@@ -46,6 +46,8 @@ function override(...rawArgs) {
       });
     } else {
       Object.keys(obj).forEach(key => {
+        if (key == '__proto__' || key == 'constructor' || key == 'prototype')
+          return
         src = target[key];
         val = obj[key];
         if (val === target) {
@@ -69,4 +71,4 @@ function override(...rawArgs) {
   return target;
 }
 
-module.exports = override;
\ No newline at end of file
+module.exports = override;
