From 307fa45d512641fcd105f0d57ca50c1437cd1b81 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Thu, 6 Jul 2017 19:20:57 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/542

---
 coders/png.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/coders/png.c b/coders/png.c
index dfe6843ac1..860d28ec41 100644
--- a/coders/png.c
+++ b/coders/png.c
@@ -5968,6 +5968,8 @@ static Image *ReadOneMNGImage(MngInfo* mng_info, const ImageInfo *image_info,
 
               for (i=(ssize_t) first_object; i <= (ssize_t) last_object; i++)
               {
+                if ((i < 0) || (i >= MNG_MAX_OBJECTS))
+                  continue;
                 if (mng_info->exists[i] && !mng_info->frozen[i] &&
                     (p-chunk) < (ssize_t) (length-8))
                   {
