From bfbd84f5d1308cdfcb08a1727821f7103be151ea Mon Sep 17 00:00:00 2001
From: Scott Nonnenberg <scott@signal.org>
Date: Fri, 11 May 2018 12:12:46 -0700
Subject: [PATCH] Update link handling

---
 js/views/message_view.js | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/js/views/message_view.js b/js/views/message_view.js
index 0cd3e06aeb..540be3ee01 100644
--- a/js/views/message_view.js
+++ b/js/views/message_view.js
@@ -15,6 +15,8 @@
 
   window.Whisper = window.Whisper || {};
 
+  const URL_REGEX = /(^|[\s\n]|<br\/?>)((?:https?|ftp):\/\/[-A-Z0-9\u00A0-\uD7FF\uE000-\uFDCF\uFDF0-\uFFFD+\u0026\u2019@#/%?=()~_|!:,.;]*[-A-Z0-9+\u0026@#/%=~()_|])/gi;
+
   const ErrorIconView = Whisper.View.extend({
     templateName: 'error-icon',
     className: 'error-icon-container',
@@ -508,7 +510,11 @@
 
       if (body.length > 0) {
         const escapedBody = body.html();
-        body.html(Signal.HTML.render(escapedBody));
+        body.html(
+          escapedBody
+            .replace(/\n/g, '<br>')
+            .replace(URL_REGEX, "$1<a href='$2' target='_blank'>$2</a>")
+        );
       }
 
       this.renderSent();
