From 198fffab4daf8aea88badd9c629350e5b26ec32f Mon Sep 17 00:00:00 2001
From: dirk <dirk@git.imagemagick.org>
Date: Fri, 15 Jan 2016 11:17:30 +0100
Subject: [PATCH] Added check for bit depth 1.

---
 coders/psd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/coders/psd.c b/coders/psd.c
index 5a5c633807..976d9076fb 100644
--- a/coders/psd.c
+++ b/coders/psd.c
@@ -1908,6 +1908,8 @@ static Image *ReadPSDImage(const ImageInfo *image_info,ExceptionInfo *exception)
           image->alpha_trait=UndefinedPixelTrait;
         }
     }
+  if ((image->depth == 1) && (image->storage_class != PseudoClass))
+    ThrowReaderException(CorruptImageError, "ImproperImageHeader");
   has_merged_image=MagickTrue;
   length=ReadBlobMSBLong(image);
   if (length != 0)
