From 999b5ad1d59760d7e450ceb541f55432fc74cd27 Mon Sep 17 00:00:00 2001
From: yurabakhtin <yurybakh@gmail.com>
Date: Mon, 3 Oct 2016 10:40:07 +0300
Subject: [PATCH] Fix object injection vulnerability

---
 htsrv/call_plugin.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/htsrv/call_plugin.php b/htsrv/call_plugin.php
index 919c8e7106..542df0d16f 100644
--- a/htsrv/call_plugin.php
+++ b/htsrv/call_plugin.php
@@ -36,7 +36,7 @@
 }
 else
 {	// Params given:
-	if( ( substr( $params, 0, 2 ) == 'a:' ) && ( ! preg_match( '/(;|{|})O:\+?[0-9]+:"/', $params ) ) )
+	if( ( substr( $params, 0, 2 ) == 'a:' ) && ( ! preg_match( '/(s:\d+:"[^"]*"|i:\d+)(;|{|})O:\+?[0-9]+:"/', $params ) ) )
 	{	// Allow to unserialize only arrays + check there is no Object in the array (We NEVER want to unserialize an object)
 		// (This may result in "false", but this means that unserializing failed)
 		$params = @unserialize( $params );
