From f38676b7b63bb1ec3a68d577fe23e6701f482aef Mon Sep 17 00:00:00 2001
From: FusionPBX <markjcrane@gmail.com>
Date: Fri, 5 Apr 2019 13:56:45 -0600
Subject: [PATCH] Update index_inc.php

---
 app/operator_panel/index_inc.php | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/app/operator_panel/index_inc.php b/app/operator_panel/index_inc.php
index cedfa93df38..95f6b114753 100644
--- a/app/operator_panel/index_inc.php
+++ b/app/operator_panel/index_inc.php
@@ -449,6 +449,7 @@
 }
 echo "<br><br>";
 
+/*
 if (if_group("superadmin") && isset($_GET['debug'])) {
 	echo '$activity<br>';
 	echo "<textarea style='width: 100%; height: 600px; overflow: scroll;' onfocus='refresh_stop();' onblur='refresh_start();'>";
@@ -461,5 +462,6 @@
 	print_r($_SESSION);
 	echo "</textarea>";
 }
+*/
 
 ?>
