From 7ebd30462bd405c4c0570b93a6d48710e6c3db19 Mon Sep 17 00:00:00 2001
From: Sami Mazouz <sychocouldy@gmail.com>
Date: Fri, 22 Jan 2021 19:53:11 +0100
Subject: [PATCH] Fix evaluation of post content by m.trust() (#24)

---
 js/src/forum/addStickyExcerpt.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/js/src/forum/addStickyExcerpt.js b/js/src/forum/addStickyExcerpt.js
index 817e626..05e6954 100644
--- a/js/src/forum/addStickyExcerpt.js
+++ b/js/src/forum/addStickyExcerpt.js
@@ -21,7 +21,8 @@ export default function addStickyControl() {
       if (firstPost) {
         const excerpt = truncate(firstPost.contentPlain(), 175);
 
-        items.add('excerpt', m.trust(excerpt), -100);
+        // Wrapping in <div> because ItemList entries need to be vnodes
+        items.add('excerpt', <div>{excerpt}</div>, -100);
       }
     }
   });
