From 9b00093288237f5073c403f2c4f62acfdfa8ed46 Mon Sep 17 00:00:00 2001
From: Abhinav Singh <mailsforabhinav@gmail.com>
Date: Sun, 10 Jan 2021 22:00:14 +0530
Subject: [PATCH] Fix basic auth condition

---
 proxy/http/proxy/auth.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/proxy/http/proxy/auth.py b/proxy/http/proxy/auth.py
index 263ec9bd0..d1ac9a861 100644
--- a/proxy/http/proxy/auth.py
+++ b/proxy/http/proxy/auth.py
@@ -35,8 +35,8 @@ def before_upstream_connection(
                 raise ProxyAuthenticationFailed()
             parts = request.headers[b'proxy-authorization'][1].split()
             if len(parts) != 2 \
-                    and parts[0].lower() != b'basic' \
-                    and parts[1] != self.flags.auth_code:
+                    or parts[0].lower() != b'basic' \
+                    or parts[1] != self.flags.auth_code:
                 raise ProxyAuthenticationFailed()
         return request
 
