From a77d8d97f5a7bced0468f0b08798c83fb67427bc Mon Sep 17 00:00:00 2001
From: Cristy <mikayla-grace@urban-warrior.org>
Date: Thu, 18 Apr 2019 19:55:30 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1552

---
 MagickCore/feature.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MagickCore/feature.c b/MagickCore/feature.c
index a0205ba158..a17a65b22b 100644
--- a/MagickCore/feature.c
+++ b/MagickCore/feature.c
@@ -2288,7 +2288,7 @@ MagickExport Image *MeanShiftImage(const Image *image,const size_t width,
               }
           }
         }
-        gamma=1.0/count;
+        gamma=PerceptibleReciprocal(count);
         mean_location.x=gamma*sum_location.x;
         mean_location.y=gamma*sum_location.y;
         mean_pixel.red=gamma*sum_pixel.red;
