From 7c68fde2d6fc65e362f20863d868c112a90a9b49 Mon Sep 17 00:00:00 2001
From: Stian Soiland-Reyes <soiland-reyes@cs.manchester.ac.uk>
Date: Wed, 3 Feb 2016 01:03:20 +0000
Subject: [PATCH] Prevent deserialization of Handler

---
 src/bsh/XThis.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/bsh/XThis.java b/src/bsh/XThis.java
index 6f68b431..f169fdc2 100644
--- a/src/bsh/XThis.java
+++ b/src/bsh/XThis.java
@@ -118,6 +118,10 @@ public Object getInterface( Class [] ca )
 	*/
 	class Handler implements InvocationHandler
 	{
+		private Object readResolve() throws ObjectStreamException {
+			throw new NotSerializableException();
+		}
+
 		public Object invoke( Object proxy, Method method, Object[] args )
 			throws Throwable
 		{
