From f6463ca9588579633bbaed9460899d892aa3c64a Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 9 Jul 2017 09:06:00 -0400
Subject: [PATCH] Zero pixel buffer

---
 coders/jpeg.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/coders/jpeg.c b/coders/jpeg.c
index 2d1be58a92..c3a813a6e5 100644
--- a/coders/jpeg.c
+++ b/coders/jpeg.c
@@ -1318,6 +1318,8 @@ static Image *ReadJPEGImage(const ImageInfo *image_info,
       ThrowReaderException(ResourceLimitError,"MemoryAllocationFailed");
     }
   jpeg_pixels=(JSAMPLE *) GetVirtualMemoryBlob(memory_info);
+  (void) ResetMagickMemory(jpeg_pixels,0,image->columns*
+    jpeg_info.output_components*sizeof(*jpeg_pixels));
   /*
     Convert JPEG pixels to pixel packets.
   */
