From 9d5d1a9f9aae2c8baee48551185da5de0cdb62c2 Mon Sep 17 00:00:00 2001
From: snipe <snipe@snipe.net>
Date: Mon, 13 Dec 2021 12:03:03 -0800
Subject: [PATCH] Added escape to assigned_to API response

Signed-off-by: snipe <snipe@snipe.net>
---
 app/Http/Transformers/AssetsTransformer.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/Http/Transformers/AssetsTransformer.php b/app/Http/Transformers/AssetsTransformer.php
index 0da314002f5..e9abca65e13 100644
--- a/app/Http/Transformers/AssetsTransformer.php
+++ b/app/Http/Transformers/AssetsTransformer.php
@@ -170,7 +170,7 @@ public function transformAssignedTo($asset)
         }
         return $asset->assigned ? [
             'id' => $asset->assigned->id,
-            'name' => $asset->assigned->display_name,
+            'name' => e($asset->assigned->display_name),
             'type' => $asset->assignedType()
         ] : null;
     }
