From 3f89610f721120ded3ff491cb9cd99d9927c7582 Mon Sep 17 00:00:00 2001
From: snow <>
Date: Wed, 17 Nov 2021 15:36:11 +0800
Subject: [PATCH] fix:fix xss #113

Signed-off-by: snow <>
---
 admin/globals.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/admin/globals.php b/admin/globals.php
index 388e9c41a..ca5a65b35 100755
--- a/admin/globals.php
+++ b/admin/globals.php
@@ -20,7 +20,7 @@
 $sta_cache = $CACHE->readCache('sta');
 $user_cache = $CACHE->readCache('user');
 $action = isset($_GET['action']) ? addslashes($_GET['action']) : '';
-$admin_path_code = isset($_GET['s']) ? addslashes($_GET['s']) : '';
+$admin_path_code = isset($_GET['s']) ? addslashes(htmlClean($_GET['s'])) : '';
 
 if ($action == 'login') {
 	if (defined('ADMIN_PATH_CODE') && $admin_path_code !== ADMIN_PATH_CODE) {
