From 62acb7753b0e3c0d3ab3c15057b0a65222313334 Mon Sep 17 00:00:00 2001
From: Jason Summers <jason1@pobox.com>
Date: Sat, 13 Mar 2021 07:35:56 -0500
Subject: [PATCH] pict,macrsrc: Fixed a bug that could cause division by 0
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Found by F. Çelik.
---
 src/fmtutil.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/fmtutil.c b/src/fmtutil.c
index f7e9ac1b..ab728413 100644
--- a/src/fmtutil.c
+++ b/src/fmtutil.c
@@ -1618,7 +1618,9 @@ void fmtutil_macbitmap_read_pixmap_only_fields(deark *c, dbuf *f, struct fmtutil
 	de_dbg(c, "pixel type=%d, bits/pixel=%d, components/pixel=%d, bits/comp=%d",
 		(int)bi->pixeltype, (int)bi->pixelsize, (int)bi->cmpcount, (int)bi->cmpsize);
 
-	bi->pdwidth = (bi->rowbytes*8)/bi->pixelsize;
+	if(bi->pixelsize>0) {
+		bi->pdwidth = (bi->rowbytes*8)/bi->pixelsize;
+	}
 	if(bi->pdwidth < bi->npwidth) {
 		bi->pdwidth = bi->npwidth;
 	}
