From 22192de5367fa0cee985917f092be4060b7c00b0 Mon Sep 17 00:00:00 2001
From: Gilles Boccon-Gibod <bok@bok.net>
Date: Sat, 9 Sep 2017 10:56:33 -0700
Subject: [PATCH] fix for #185

---
 Source/C++/Core/Ap4HdlrAtom.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Source/C++/Core/Ap4HdlrAtom.cpp b/Source/C++/Core/Ap4HdlrAtom.cpp
index 34aefca9..42c117c5 100644
--- a/Source/C++/Core/Ap4HdlrAtom.cpp
+++ b/Source/C++/Core/Ap4HdlrAtom.cpp
@@ -80,9 +80,10 @@ AP4_HdlrAtom::AP4_HdlrAtom(AP4_UI32        size,
     stream.ReadUI32(m_Reserved[2]);
     
     // read the name unless it is empty
-    int name_size = size-(AP4_FULL_ATOM_HEADER_SIZE+20);
-    if (name_size == 0) return;
+    if (size < AP4_FULL_ATOM_HEADER_SIZE+20) return;
+    AP4_UI32 name_size = size-(AP4_FULL_ATOM_HEADER_SIZE+20);
     char* name = new char[name_size+1];
+    if (name == NULL) return;
     stream.Read(name, name_size);
     name[name_size] = '\0'; // force a null termination
     // handle a special case: the Quicktime files have a pascal
