From 83fd44ef8921a8dcf394a012e44901ab08596bdc Mon Sep 17 00:00:00 2001
From: Julia Nguyen <tojulianguyen@gmail.com>
Date: Thu, 18 Nov 2021 18:03:30 -0500
Subject: [PATCH] [#2052] Sanitize string content in renderContent

---
 client/app/utils/index.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/client/app/utils/index.js b/client/app/utils/index.js
index 7bd7b0390..b0015344d 100644
--- a/client/app/utils/index.js
+++ b/client/app/utils/index.js
@@ -1,6 +1,7 @@
 // @flow
 import axios from 'axios';
 import renderHTML from 'react-render-html';
+import { sanitize } from 'dompurify';
 
 const randomString = () => Math.random()
   .toString(36)
@@ -34,7 +35,7 @@ const getPusher = () => {
 
 const renderContent = (content: string | any) => {
   if (typeof content === 'string') {
-    return renderHTML(content);
+    return renderHTML(sanitize(content));
   }
   return content;
 };
