From 30610bda60cbfc058f90a1c0a17d0e8f4516221a Mon Sep 17 00:00:00 2001
From: Luca Deri <deri@ntop.org>
Date: Sun, 17 Jun 2018 15:24:25 +0200
Subject: [PATCH] Added code to randomize values returned by rand()

---
 src/HTTPserver.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/HTTPserver.cpp b/src/HTTPserver.cpp
index f954a4b942c..68cffcfb5ba 100644
--- a/src/HTTPserver.cpp
+++ b/src/HTTPserver.cpp
@@ -864,7 +864,12 @@ HTTPserver::HTTPserver(const char *_docs_dir, const char *_scripts_dir) {
   bool use_http = true;
   struct stat statsBuf;
   int stat_rc;
+  struct timeval tv;
 
+  /* Randomize data */
+  gettimeofday(&tv, NULL);
+  srand(tv.tv_sec + tv.tv_usec);
+  
   static char *http_options[] = {
     (char*)"listening_ports", ports,
     (char*)"enable_directory_listing", (char*)"no",
