From 895a131d2eb7e447757a35d5731c7d647823ea8b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Mon, 20 Jun 2016 10:06:42 +0200
Subject: [PATCH] Escape HTML when rendering AJAX error
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 js/ajax.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/js/ajax.js b/js/ajax.js
index ba121c81a84..bac829d9f09 100644
--- a/js/ajax.js
+++ b/js/ajax.js
@@ -787,8 +787,8 @@ $(document).ajaxError(function (event, request, settings) {
         PMA_ajaxShowMessage(
             '<div class="error">' +
             PMA_messages.strErrorProcessingRequest +
-            '<div>' + errorCode + '</div>' +
-            '<div>' + errorText + '</div>' +
+            '<div>' + escapeHtml(errorCode) + '</div>' +
+            '<div>' + escapeHtml(errorText) + '</div>' +
             '</div>',
             false
         );
