From 37c34d089aa19f30d11203bb0c7f85b486424372 Mon Sep 17 00:00:00 2001
From: Madhura Jayaratne <madhura.cj@gmail.com>
Date: Thu, 25 Feb 2016 21:51:27 +1100
Subject: [PATCH] Escape selectors

Signed-off-by: Madhura Jayaratne <madhura.cj@gmail.com>
---
 js/normalization.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/js/normalization.js b/js/normalization.js
index 50ad3ed39f3..606ab899135 100644
--- a/js/normalization.js
+++ b/js/normalization.js
@@ -128,7 +128,7 @@ function goToStep4()
             $("#mainContent #newCols").html('');
             $('.tblFooters').html('');
             for(var pk in primary_key) {
-                $("#extra input[value='" + primary_key[pk] + "']").attr("disabled","disabled");
+                $("#extra input[value='" + escapeJsString(primary_key[pk]) + "']").attr("disabled","disabled");
             }
         }
     );
@@ -153,7 +153,7 @@ function goToStep3()
             $('.tblFooters').html('');
             primary_key = $.parseJSON(data.primary_key);
             for(var pk in primary_key) {
-                $("#extra input[value='" + primary_key[pk] + "']").attr("disabled","disabled");
+                $("#extra input[value='" + escapeJsString(primary_key[pk]) + "']").attr("disabled","disabled");
             }
         }
     );
