From 36149c1ff99973840a809bb865f23e1b23d6df00 Mon Sep 17 00:00:00 2001
From: Julio Montoya <gugli100@gmail.com>
Date: Thu, 13 May 2021 12:58:16 +0200
Subject: [PATCH] Add int casting

---
 main/plagiarism/compilatio/upload.php | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/main/plagiarism/compilatio/upload.php b/main/plagiarism/compilatio/upload.php
index 04d87405909..c6e3e2da6c6 100644
--- a/main/plagiarism/compilatio/upload.php
+++ b/main/plagiarism/compilatio/upload.php
@@ -88,7 +88,16 @@
 
 function sendDocument($documentId, $courseInfo)
 {
-    $courseId = $courseInfo['real_id'];
+    if (empty($courseInfo)) {
+        return false;
+    }
+
+    $courseId = $courseInfo['real_id'] ?? 0;
+    $documentId = (int) $documentId;
+
+    if (empty($courseId) || empty($documentId)) {
+        return false;
+    }
 
     compilatioUpdateWorkDocument($documentId, $courseId);
     $workTable = Database::get_course_table(TABLE_STUDENT_PUBLICATION);
