From 9e927affe3a021175f354af5fa01d22657c20585 Mon Sep 17 00:00:00 2001
From: JerikoOne <jeriko.one@gmx.us>
Date: Tue, 3 Jul 2018 17:22:12 -0500
Subject: [PATCH] Add alloc fail check in nntp_fetch_headers

---
 nntp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/nntp.c b/nntp.c
index 032b6e141d..a0ffa609cb 100644
--- a/nntp.c
+++ b/nntp.c
@@ -1288,6 +1288,8 @@ static int nntp_fetch_headers(struct Context *ctx, void *hc, anum_t first,
   fc.last = last;
   fc.restore = restore;
   fc.messages = mutt_mem_calloc(last - first + 1, sizeof(unsigned char));
+  if (fc.messages == NULL)
+	  return -1;
 #ifdef USE_HCACHE
   fc.hc = hc;
 #endif
