From da72e0fddaeceee44fbbd7241e07e5d53d1eee64 Mon Sep 17 00:00:00 2001
From: Charlene Auger <root@OCS-srv-trunk2-ocs.factorfx.com>
Date: Wed, 8 Jul 2020 08:52:09 +0000
Subject: [PATCH] Fix CVE-2020-14947

---
 require/commandLine/CommandLine.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/require/commandLine/CommandLine.php b/require/commandLine/CommandLine.php
index 0124b3e10..57e4f9f4f 100644
--- a/require/commandLine/CommandLine.php
+++ b/require/commandLine/CommandLine.php
@@ -32,7 +32,7 @@ public function get_mib_oid($file) {
         $champs = array('SNMP_MIB_DIRECTORY' => 'SNMP_MIB_DIRECTORY');
         $values = look_config_default_values($champs);
         $cmd = "snmptranslate -Tz -m ".$values['tvalue']['SNMP_MIB_DIRECTORY']."/".$file;
-        $result_cmd = shell_exec($cmd);
+        $result_cmd = shell_exec(escapeshellcmd($cmd));
         $result_cmd = preg_split("/\r\n|\n|\r/", $result_cmd);
         $result_cmd = str_replace('"', "", $result_cmd);
 
