From 13273e019d737f8eb4d7b1fe1eb665185dddfb5a Mon Sep 17 00:00:00 2001
From: Taylor Finley <taylorfinley@gmail.com>
Date: Wed, 29 May 2019 12:06:59 -1000
Subject: [PATCH] Sanitizing user input

---
 RelativePath.Example1.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/RelativePath.Example1.php b/RelativePath.Example1.php
index 323b45f..92057c3 100644
--- a/RelativePath.Example1.php
+++ b/RelativePath.Example1.php
@@ -82,12 +82,13 @@ function execTime($function, $title="", $iterations = 100000) {
 <form method="post">
 <p>Path:<br />
 <input type="text" size="120" name="path"
-	value="<?php echo stripslashes($_POST['path']); ?>" /></p>
+	value="<?php echo htmlspecialchars(stripslashes($_POST['path']), ENT_QUOTES); ?>" /></p>
 <p><input type="submit" /></p>
 </form>
 <dl>
 <?php
 foreach ($paths as $path) {
+    $path = htmlspecialchars($path, ENT_QUOTES);
 	echo "<dt><pre>Path '$path' becomes:</pre></dt>\n";
 	echo "<dd><pre>";
 	echo "'" . RelativePath::getRelativePath($path) . "'\n";
