From cf43beedb05131937ef46f365ab0a0c6fa6ac618 Mon Sep 17 00:00:00 2001
From: psychobunny <psycho.bunny@hotmail.com>
Date: Thu, 20 Aug 2020 01:11:57 -0400
Subject: [PATCH] fix: CSRF issues

---
 library.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/library.js b/library.js
index 7786830a..7960b486 100644
--- a/library.js
+++ b/library.js
@@ -248,8 +248,8 @@
 		});
 
 		app.get('/comments/get/:id/:pagination?', middleware.applyCSRF, Comments.getCommentData);
-		app.post('/comments/reply', Comments.replyToComment);
-		app.post('/comments/publish', Comments.publishArticle);
+		app.post('/comments/reply', middleware.applyCSRF, Comments.replyToComment);
+		app.post('/comments/publish', middleware.applyCSRF, Comments.publishArticle);
 
 		app.get('/admin/blog-comments', middleware.admin.buildHeader, renderAdmin);
 		app.get('/api/admin/blog-comments', renderAdmin);
