From 559906731f153c9b3a632c2839ed11669b76d593 Mon Sep 17 00:00:00 2001
From: Nik Okuntseff <support@anuko.com>
Date: Tue, 12 Oct 2021 20:01:36 +0000
Subject: [PATCH] Added a check for passed in date to time.php.

---
 initialize.php | 2 +-
 time.php       | 6 ++++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/initialize.php b/initialize.php
index 10550736..e0b38f4d 100644
--- a/initialize.php
+++ b/initialize.php
@@ -12,7 +12,7 @@
 // Disable displaying errors on screen.
 ini_set('display_errors', 'Off');
 
-define("APP_VERSION", "1.19.29.5598");
+define("APP_VERSION", "1.19.30.5599");
 define("APP_DIR", dirname(__FILE__));
 define("LIBRARY_DIR", APP_DIR."/WEB-INF/lib");
 define("TEMPLATE_DIR", APP_DIR."/WEB-INF/templates");
diff --git a/time.php b/time.php
index 1bee1129..61e771f8 100644
--- a/time.php
+++ b/time.php
@@ -32,6 +32,12 @@
     exit();
   }
 }
+// If we are passed in a date, make sure it is in correct format.
+$date = $request->getParameter('date');
+if ($date && !ttValidDate($date)) {
+  header('Location: access_denied.php');
+  exit();
+}
 // End of access checks.
 
 // Determine user for whom we display this page.
