From e652d56ac60eadfc26489ab83927af13a9b9d8ce Mon Sep 17 00:00:00 2001
From: Morgan-Phoenix <73711602+Morgan-Phoenix@users.noreply.github.com>
Date: Sat, 6 Nov 2021 19:34:45 +0530
Subject: [PATCH] Fixed GHSA-35m5-8cvj-8783

---
 enrocrypt/hashing.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/enrocrypt/hashing.py b/enrocrypt/hashing.py
index 275d2e3..14328f0 100644
--- a/enrocrypt/hashing.py
+++ b/enrocrypt/hashing.py
@@ -66,11 +66,6 @@ def SHA244(self,data:str):
         hash = str(sha.digest())
         return self.__Salt(hash,salt=self.salt)
 
-    def MD5(self,data:str):
-        sha = hashlib.md5(bytes(data.encode()))
-        hash = str(sha.digest())
-        return self.__Salt(hash,salt=self.salt)
-
     def SHA384(self,data:str):
         sha = hashlib.sha384(bytes(data.encode()))
         hash = str(sha.digest())
