From 47338393f1f79558f6144213409f09f81d7c4837 Mon Sep 17 00:00:00 2001
From: Cobus van Eeden <35851496+cobusve@users.noreply.github.com>
Date: Mon, 7 Dec 2020 11:48:51 -0800
Subject: [PATCH] add assert for addition overflow on queue creation (#225)

---
 queue.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/queue.c b/queue.c
index d2e27e55a5..b01dfd11ff 100644
--- a/queue.c
+++ b/queue.c
@@ -397,6 +397,9 @@ BaseType_t xQueueGenericReset( QueueHandle_t xQueue,
         /* Check for multiplication overflow. */
         configASSERT( ( uxItemSize == 0 ) || ( uxQueueLength == ( xQueueSizeInBytes / uxItemSize ) ) );
 
+        /* Check for addition overflow. */
+        configASSERT( ( sizeof( Queue_t ) + xQueueSizeInBytes ) >  xQueueSizeInBytes );
+
         /* Allocate the queue and storage area.  Justification for MISRA
          * deviation as follows:  pvPortMalloc() always ensures returned memory
          * blocks are aligned per the requirements of the MCU stack.  In this case
