From e7879e27a37119fad4faf133a9f70bdcdc75d760 Mon Sep 17 00:00:00 2001
From: Russ Allbery <rra@debian.org>
Date: Sun, 22 Mar 2020 21:17:58 -0700
Subject: [PATCH] Add CVE for buffer overflow security fix

---
 NEWS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/NEWS b/NEWS
index bd2d73c..1834aa3 100644
--- a/NEWS
+++ b/NEWS
@@ -5,6 +5,7 @@ pam-krb5 4.9 (unreleased)
     SECURITY: All previous versions of this module could overflow the
     buffer provided by the underlying Kerberos library for the response to
     a prompt by writing a single nul character past the end of the buffer.
+    (CVE-2020-10595)
 
     Support use_pkinit with MIT Kerberos.  (Debian Bug#871699)
 
