From 97c9f438a9b3454d085895f4d1f66389fd22a0fb Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 24 Jan 2016 19:09:18 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/96

---
 coders/meta.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/coders/meta.c b/coders/meta.c
index eaff13c3e5..fb95a94caa 100644
--- a/coders/meta.c
+++ b/coders/meta.c
@@ -460,7 +460,7 @@ static ssize_t parse8BIM(Image *ifile, Image *ofile)
 
                     next=0;
                     outputlen += len;
-                    while (len--)
+                    while (len-- > 0)
                       (void) WriteBlobByte(ofile,(unsigned char) token[next++]);
 
                     if (outputlen & 1)
@@ -491,7 +491,7 @@ static ssize_t parse8BIM(Image *ifile, Image *ofile)
                     outputlen += 5;
                     next=0;
                     outputlen += len;
-                    while (len--)
+                    while (len-- > 0)
                       (void) WriteBlobByte(ofile,(unsigned char) token[next++]);
                   }
               }
