From 41c4e0214c286f28830cca54423b5db57e7c0ce4 Mon Sep 17 00:00:00 2001
From: Madhura Jayaratne <madhura.cj@gmail.com>
Date: Mon, 29 Feb 2016 12:26:44 +1100
Subject: [PATCH] Fix XSS in zoom search

Signed-off-by: Madhura Jayaratne <madhura.cj@gmail.com>
---
 libraries/controllers/TableSearchController.class.php | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libraries/controllers/TableSearchController.class.php b/libraries/controllers/TableSearchController.class.php
index 12de6d2367a..ddd6d56c4ba 100644
--- a/libraries/controllers/TableSearchController.class.php
+++ b/libraries/controllers/TableSearchController.class.php
@@ -471,7 +471,11 @@ public function changeTableInfoAction()
             return;
         }
         $key = array_search($field, $this->_columnNames);
-        $properties = $this->getColumnProperties($_REQUEST['it'], $key);
+        $search_index = 0;
+        if (PMA_isValid($_REQUEST['it'], 'integer')) {
+            $search_index = $_REQUEST['it'];
+        }
+        $properties = $this->getColumnProperties($search_index, $key);
         $this->response->addJSON(
             'field_type', htmlspecialchars($properties['type'])
         );
