From 746240bd13b62b5956fc34389cfbdc09e1e67775 Mon Sep 17 00:00:00 2001
From: Madhura Jayaratne <madhura.cj@gmail.com>
Date: Thu, 25 Feb 2016 20:08:38 +1100
Subject: [PATCH] Fix XSS in normalization

Signed-off-by: Madhura Jayaratne <madhura.cj@gmail.com>
---
 normalization.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/normalization.php b/normalization.php
index 4cb95691810..069dabbc0fd 100644
--- a/normalization.php
+++ b/normalization.php
@@ -72,7 +72,7 @@
 $scripts->addFile('normalization.js');
 $scripts->addFile('jquery/jquery.uitablefilter.js');
 $normalForm = '1nf';
-if (isset($_REQUEST['normalizeTo'])) {
+if (PMA_isValid($_REQUEST['normalizeTo'], array('1nf', '2nf', '3nf'))) {
     $normalForm = $_REQUEST['normalizeTo'];
 }
 if (isset($_REQUEST['createNewTables2NF'])) {
