From bec8e371f170a12e11cd222ffc7a6e1ae9942638 Mon Sep 17 00:00:00 2001
From: ssnau <korige@gmail.com>
Date: Thu, 18 Mar 2021 01:26:35 +0800
Subject: [PATCH] fix a vulnerability issue if a provided port is not a number

---
 index.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/index.js b/index.js
index e5f2d6d..8043f1f 100644
--- a/index.js
+++ b/index.js
@@ -5,6 +5,7 @@ var notEmpty = function(x) {return x};
 
 module.exports = function killport(port) {
   return (new Promise(function(resolve, reject) {
+    if (!/^\d+$/.test(port)) throw new Error('port must be a number.');
     var cmd = 'lsof -i:' + port; 
     cp.exec(cmd, function(err, stdout, stderr){
       // do not check `err`, if no process found
