From 7bb1b4a4dd23c885f9db9f577dfe79ecc5433109 Mon Sep 17 00:00:00 2001
From: jeanlf <jeanlf@gpac.io>
Date: Mon, 30 Aug 2021 16:15:32 +0200
Subject: [PATCH] fixed #1893

---
 src/media_tools/av_parsers.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/media_tools/av_parsers.c b/src/media_tools/av_parsers.c
index 3ce9a9015..4ea0db8a9 100644
--- a/src/media_tools/av_parsers.c
+++ b/src/media_tools/av_parsers.c
@@ -3980,10 +3980,8 @@ void gf_av1_reset_state(AV1State *state, Bool is_destroy)
 		gf_list_del(l1);
 		gf_list_del(l2);
 		if (state->bs) {
-			if (gf_bs_get_position(state->bs)) {
-				u32 size;
-				gf_bs_get_content_no_truncate(state->bs, &state->frame_obus, &size, &state->frame_obus_alloc);
-			}
+			u32 size;
+			gf_bs_get_content_no_truncate(state->bs, &state->frame_obus, &size, &state->frame_obus_alloc);
 			gf_bs_del(state->bs);
 		}
 		state->bs = NULL;
