From 76aa8a7b49379dbee48e8096d681114be235196f Mon Sep 17 00:00:00 2001
From: parker02311 <35582919+parker02311@users.noreply.github.com>
Date: Mon, 18 Oct 2021 12:33:23 -1000
Subject: [PATCH] V1.0.2

---
 BOT/lib/cogs/website.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/BOT/lib/cogs/website.py b/BOT/lib/cogs/website.py
index bcb449b..bc69f0f 100644
--- a/BOT/lib/cogs/website.py
+++ b/BOT/lib/cogs/website.py
@@ -59,6 +59,7 @@ async def status():
 
 
 @app.route("/v1/products", methods=["GET"])
+@require_apikey
 async def products():
     dbresponse = getproducts()
     results = {}
