From 4ec444f4eab88cf4bec664fafcf9cab50bc5ff6a Mon Sep 17 00:00:00 2001
From: dirk <dirk@git.imagemagick.org>
Date: Wed, 19 Oct 2016 07:29:36 +0200
Subject: [PATCH] Fixed memory leak.

---
 coders/psd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/coders/psd.c b/coders/psd.c
index 2e587fed16..e936a7d669 100644
--- a/coders/psd.c
+++ b/coders/psd.c
@@ -1719,6 +1719,8 @@ ModuleExport MagickBooleanType ReadPSDLayers(Image *image,
             if (image->debug != MagickFalse)
               (void) LogMagickEvent(CoderEvent,GetMagickModule(),
                 "      layer data is empty");
+            if (layer_info[i].info != (StringInfo *) NULL)
+              layer_info[i].info=DestroyStringInfo(layer_info[i].info);
             continue;
           }
 
