From 8c35502217c1879cb8257c617007282eee3fe1cc Mon Sep 17 00:00:00 2001
From: Dirk Lemstra <dirk@git.imagemagick.org>
Date: Sat, 18 Nov 2017 10:23:27 +0100
Subject: [PATCH] Added missing return to avoid use after free.

---
 Magick++/lib/Image.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Magick++/lib/Image.cpp b/Magick++/lib/Image.cpp
index f4fc34a689..15d98f2d95 100644
--- a/Magick++/lib/Image.cpp
+++ b/Magick++/lib/Image.cpp
@@ -5087,6 +5087,7 @@ void Magick::Image::read(MagickCore::Image *image,
       if (!quiet())
         throwExceptionExplicit(MagickCore::ImageWarning,
           "No image was loaded.");
+      return;
     }
   ThrowImageException;
 }
