From f68a98a9d385838a1c73ec960a14102949940a64 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Thu, 31 Aug 2017 10:27:46 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/714

---
 coders/psd.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/coders/psd.c b/coders/psd.c
index 88227678ef..6c026d2b14 100644
--- a/coders/psd.c
+++ b/coders/psd.c
@@ -1688,6 +1688,12 @@ static MagickBooleanType ReadPSDLayersInternal(Image *image,
                 {
                   size_t blend_source=ReadBlobLong(image);
                   size_t blend_dest=ReadBlobLong(image);
+                  if (EOFBlob(image) != MagickFalse)
+                    {
+                      layer_info=DestroyLayerInfo(layer_info,number_layers);
+                      ThrowBinaryException(CorruptImageError,
+                        "InsufficientImageDataInFile",image->filename);
+                    }
                   if (image->debug != MagickFalse)
                     (void) LogMagickEvent(CoderEvent,GetMagickModule(),
                       "        source(%x), dest(%x)",(unsigned int)
