From e269ea0d3bc88417163c18431b1df38a9be92bfc Mon Sep 17 00:00:00 2001
From: Kowlin <boxedpp@gmail.com>
Date: Wed, 12 Aug 2020 00:40:06 +0200
Subject: [PATCH] Added consume all to streams.

---
 redbot/cogs/streams/streams.py | 22 +++++++++++++---------
 1 file changed, 13 insertions(+), 9 deletions(-)

diff --git a/redbot/cogs/streams/streams.py b/redbot/cogs/streams/streams.py
index 5570afc9e53..1aa582e8675 100644
--- a/redbot/cogs/streams/streams.py
+++ b/redbot/cogs/streams/streams.py
@@ -497,14 +497,13 @@ async def message(self, ctx: commands.Context):
 
     @message.command(name="mention")
     @commands.guild_only()
-    async def with_mention(self, ctx: commands.Context, message: str = None):
+    async def with_mention(self, ctx: commands.Context, *, message: str = None):
         """Set stream alert message when mentions are enabled.
 
         Use `{mention}` in the message to insert the selected mentions.
+        Use `{stream}` in the message to insert the channel or user name.
 
-        Use `{stream.name}` in the message to insert the channel or user name.
-
-        For example: `[p]streamset message mention "{mention}, {stream.name} is live!"`
+        For example: `[p]streamset message mention "{mention}, {stream} is live!"`
         """
         if message is not None:
             guild = ctx.guild
@@ -515,12 +514,12 @@ async def with_mention(self, ctx: commands.Context, message: str = None):
 
     @message.command(name="nomention")
     @commands.guild_only()
-    async def without_mention(self, ctx: commands.Context, message: str = None):
+    async def without_mention(self, ctx: commands.Context, *, message: str = None):
         """Set stream alert message when mentions are disabled.
 
-        Use `{stream.name}` in the message to insert the channel or user name.
+        Use `{stream}` in the message to insert the channel or user name.
 
-        For example: `[p]streamset message nomention "{stream.name} is live!"`
+        For example: `[p]streamset message nomention "{stream} is live!"`
         """
         if message is not None:
             guild = ctx.guild
@@ -720,7 +719,10 @@ async def check_streams(self):
                                 channel.guild
                             ).live_message_mention()
                             if alert_msg:
-                                content = alert_msg.format(mention=mention_str, stream=stream)
+                                content = alert_msg  # Stop bad things from happening here...
+                                content = content.replace("{stream.name}", str(stream.name))  # Backwards compatability
+                                content = content.replace("{stream}", str(stream.name))
+                                content = content.replace("{mention}", mention_str)
                             else:
                                 content = _("{mention}, {stream} is live!").format(
                                     mention=mention_str,
@@ -733,7 +735,9 @@ async def check_streams(self):
                                 channel.guild
                             ).live_message_nomention()
                             if alert_msg:
-                                content = alert_msg.format(stream=stream)
+                                content = alert_msg  # Stop bad things from happening here...
+                                content = content.replace("{stream.name}", str(stream.name))  # Backwards compatability
+                                content = content.replace("{stream}", str(stream.name))
                             else:
                                 content = _("{stream} is live!").format(
                                     stream=escape(
