From 86ef1a4e143ea5bb0c7b309574127cc39d4faa74 Mon Sep 17 00:00:00 2001
From: Andrew Bastin <andrewbastin.k@gmail.com>
Date: Wed, 5 Jan 2022 09:18:16 +0530
Subject: [PATCH] fix: apply sandbox to HTML preview iframe

---
 .../components/lenses/renderers/HTMLLensRenderer.vue             | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packages/hoppscotch-app/components/lenses/renderers/HTMLLensRenderer.vue b/packages/hoppscotch-app/components/lenses/renderers/HTMLLensRenderer.vue
index 3b43da26da..2740aefaeb 100644
--- a/packages/hoppscotch-app/components/lenses/renderers/HTMLLensRenderer.vue
+++ b/packages/hoppscotch-app/components/lenses/renderers/HTMLLensRenderer.vue
@@ -49,6 +49,7 @@
       class="covers-response"
       src="about:blank"
       loading="lazy"
+      sandbox=""
     ></iframe>
   </div>
 </template>
