From 5edadcfb83bf27107578830801817f9e6d0ad941 Mon Sep 17 00:00:00 2001
From: Bron Gondwana <brong@fastmail.fm>
Date: Fri, 11 Aug 2017 11:58:24 +1000
Subject: [PATCH] dlist: don't allow overwrite of arbitrary files

---
 imap/dlist.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/imap/dlist.c b/imap/dlist.c
index efe38b70c4..aa1024c14d 100644
--- a/imap/dlist.c
+++ b/imap/dlist.c
@@ -141,16 +141,10 @@ EXPORTED const char *dlist_reserve_path(const char *part, int isarchive,
                                         const struct message_guid *guid)
 {
     static char buf[MAX_MAILBOX_PATH];
-    const char *base;
 
-    /* part can be either a configured partition name, or a path */
-    if (strchr(part, '/')) {
-        base = part;
-    }
-    else {
-        base = isarchive ? config_archivepartitiondir(part)
-                         : config_partitiondir(part);
-    }
+    /* part must be a configured partition name on this server */
+    const char *base = isarchive ? config_archivepartitiondir(part)
+                                 : config_partitiondir(part);
 
     /* we expect to have a base at this point, so let's assert that */
     assert(base != NULL);
