From 36df83a97a7f140fdb008b727a94f882847c6a6f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Wed, 22 Jun 2016 11:00:14 +0200
Subject: [PATCH] Escape saved search name
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 libraries/DbQbe.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libraries/DbQbe.php b/libraries/DbQbe.php
index 37f268e36e6..d9d1f3ca21f 100644
--- a/libraries/DbQbe.php
+++ b/libraries/DbQbe.php
@@ -1900,7 +1900,7 @@ private function _getSavedSearchesField()
         }
         $html_output .= '</select>';
         $html_output .= '<input type="text" name="searchName" id="searchName" '
-            . 'value="' . $currentSearchName . '" />';
+            . 'value="' . htmlspecialchars($currentSearchName) . '" />';
         $html_output .= '<input type="hidden" name="action" id="action" value="" />';
         $html_output .= '<input type="submit" name="saveSearch" id="saveSearch" '
             . 'value="' . __('Create bookmark') . '" />';
