From 614a257295bdcdeda347086761062ac7658b6830 Mon Sep 17 00:00:00 2001
From: Cristy <mikayla-grace@urban-warrior.org>
Date: Mon, 22 Apr 2019 12:42:50 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick6/issues/43

---
 magick/blob.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/magick/blob.c b/magick/blob.c
index 520a727de9..cdc6932039 100644
--- a/magick/blob.c
+++ b/magick/blob.c
@@ -700,6 +700,7 @@ MagickExport unsigned char *DetachBlob(BlobInfo *blob_info)
   if (blob_info->mapped != MagickFalse)
     {
       (void) UnmapBlob(blob_info->data,blob_info->length);
+      blob_info->data=NULL;
       RelinquishMagickResource(MapResource,blob_info->length);
     }
   blob_info->mapped=MagickFalse;
