From ce1924193862d58cb43d3d4d9dada710a8361b89 Mon Sep 17 00:00:00 2001
From: Gordon Williams <gw@pur3.co.uk>
Date: Mon, 21 May 2018 15:29:33 +0100
Subject: [PATCH] fix jsvGetString regression

---
 src/jsvar.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/jsvar.c b/src/jsvar.c
index 7cd34f357..88d49b3ad 100644
--- a/src/jsvar.c
+++ b/src/jsvar.c
@@ -1190,7 +1190,7 @@ size_t jsvGetString(const JsVar *v, char *str, size_t len) {
      * want to pad the entire buffer with zeros */
     len--;
     int l = 0;
-    while (*s && l<len) {
+    while (s[l] && l<len) {
       str[l] = s[l];
       l++;
     }
