From 36a29280448097f34ce9c94606195f2ac643fed1 Mon Sep 17 00:00:00 2001
From: JerikoOne <jeriko.one@gmx.us>
Date: Sat, 7 Jul 2018 14:15:56 -0500
Subject: [PATCH] Handle NO response without message properly

---
 imap/command.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imap/command.c b/imap/command.c
index 859f98727c..74d12df18e 100644
--- a/imap/command.c
+++ b/imap/command.c
@@ -892,7 +892,7 @@ static int cmd_handle_untagged(struct ImapData *idata)
     mutt_debug(2, "Handling untagged NO\n");
 
     /* Display the warning message from the server */
-    mutt_error("%s", s + 3);
+    mutt_error("%s", s + 2);
   }
 
   return 0;
