From aca42efa01917cc0fe8cfdb2927a6399ca1742f2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Mon, 25 Jan 2016 12:21:48 +0100
Subject: [PATCH] Escape javascript variable content
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 templates/header_location.phtml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/templates/header_location.phtml b/templates/header_location.phtml
index 6e8246569d9..e1dabdfa565 100644
--- a/templates/header_location.phtml
+++ b/templates/header_location.phtml
@@ -25,8 +25,8 @@ $uri = isset($uri) ? $uri : null;
 <body>
 <script type="text/javascript">
     //<![CDATA[
-    document.write('<p><a href="<?php echo htmlspecialchars($uri) ?>"><?php echo __('Go') ?></a></p>');
+    document.write('<p><a href="<?php echo PMA_escapeJsString(htmlspecialchars($uri)) ?>"><?php echo __('Go') ?></a></p>');
     //]]>
 </script>
 </body>
-</html>
\ No newline at end of file
+</html>
