From cd3738dea038dbd12e603ad48cd7373ae0440f65 Mon Sep 17 00:00:00 2001
From: jeanlf <jeanlf@gpac.io>
Date: Fri, 9 Apr 2021 11:12:47 +0200
Subject: [PATCH] fixed #1737

---
 src/media_tools/isom_hinter.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/media_tools/isom_hinter.c b/src/media_tools/isom_hinter.c
index 6aeb9d172..6c4da454a 100644
--- a/src/media_tools/isom_hinter.c
+++ b/src/media_tools/isom_hinter.c
@@ -402,8 +402,9 @@ GF_RTPHinter *gf_hinter_track_new(GF_ISOFile *file, u32 TrackNum,
 				else if (gf_isom_has_sync_shadows(file, TrackNum) || gf_isom_has_sample_dependency(file, TrackNum)) {
 					flags |= GP_RTP_PCK_SYSTEMS_CAROUSEL;
 				}
-				gf_odf_desc_del((GF_Descriptor*)esd);
 			}
+			if (esd)
+				gf_odf_desc_del((GF_Descriptor*)esd);
 			break;
 		case GF_ISOM_SUBTYPE_3GP_H263:
 			hintType = GF_RTP_PAYT_H263;
