From ad0f966e9a997b17d853a6972ea283d4dcd70fa8 Mon Sep 17 00:00:00 2001
From: "C. Michael Pilato" <cmpilato@collab.net>
Date: Thu, 26 Mar 2020 14:11:41 -0400
Subject: [PATCH] issue #211: escape CVS subdir last-modified file name

---
 lib/viewvc.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/viewvc.py b/lib/viewvc.py
index f6d07f36..b32941e5 100644
--- a/lib/viewvc.py
+++ b/lib/viewvc.py
@@ -2412,7 +2412,7 @@ def view_directory(request):
       if request.roottype == 'cvs' and file.rev is not None:
         row.rev = None
         if cfg.options.show_logs:
-          row.log_file = file.newest_file
+          row.log_file = request.server.escape(file.newest_file)
           row.log_rev = file.rev
 
       if request.roottype == 'svn':
