From a0e0d7de20f5d1712f1793f925496def4383c610 Mon Sep 17 00:00:00 2001
From: Demis Bellot <demis.bellot@gmail.com>
Date: Fri, 17 Aug 2018 11:05:59 -0400
Subject: [PATCH] HtmlEncode Raw URL in HtmlFormat snapshot

---
 src/ServiceStack/Formats/HtmlFormat.cs | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/ServiceStack/Formats/HtmlFormat.cs b/src/ServiceStack/Formats/HtmlFormat.cs
index e62df99e576..482955e38fc 100644
--- a/src/ServiceStack/Formats/HtmlFormat.cs
+++ b/src/ServiceStack/Formats/HtmlFormat.cs
@@ -90,12 +90,13 @@ public async Task SerializeToStreamAsync(IRequest req, object response, Stream o
             {
                 // Serialize then escape any potential script tags to avoid XSS when displaying as HTML
                 var json = JsonDataContractSerializer.Instance.SerializeToString(dto) ?? "null";
-                json = json.Replace("<", "&lt;").Replace(">", "&gt;");
+                json = json.HtmlEncode();
 
                 var url = req.ResolveAbsoluteUrl()
                     .Replace("format=html", "")
                     .Replace("format=shtm", "")
-                    .TrimEnd('?', '&');
+                    .TrimEnd('?', '&')
+                    .HtmlEncode();
 
                 url += url.Contains("?") ? "&" : "?";
 
