From b522d2d857d2f75b659936b59b0da9df1682c256 Mon Sep 17 00:00:00 2001
From: Cristy <mikayla-grace@urban-warrior.org>
Date: Thu, 18 Apr 2019 19:55:44 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1552

---
 magick/feature.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/magick/feature.c b/magick/feature.c
index db33743d2f..434c5c5c16 100644
--- a/magick/feature.c
+++ b/magick/feature.c
@@ -2313,7 +2313,7 @@ MagickExport Image *MeanShiftImage(const Image *image,const size_t width,
               }
           }
         }
-        gamma=1.0/count;
+        gamma=PerceptibleReciprocal(count);
         mean_location.x=gamma*sum_location.x;
         mean_location.y=gamma*sum_location.y;
         mean_pixel.red=gamma*sum_pixel.red;
