From c9aa9f80828594eacbe3affe16c43d623562e5d8 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 8 Jul 2017 18:30:09 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/550

---
 coders/png.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/coders/png.c b/coders/png.c
index dc8be52207..080709c047 100644
--- a/coders/png.c
+++ b/coders/png.c
@@ -4377,7 +4377,13 @@ static Image *ReadOneJNGImage(MngInfo *mng_info,
         type[0],type[1],type[2],type[3],(double) length);
 
     if (length > PNG_UINT_31_MAX || count == 0)
-      ThrowReaderException(CorruptImageError,"CorruptImage");
+      {
+        if (color_image != (Image *) NULL)
+          color_image=DestroyImage(color_image);
+        if (color_image_info != (Image *) NULL)
+          color_image_info=DestroyImageInfo(color_image_info);
+        ThrowReaderException(CorruptImageError,"CorruptImage");
+      }
 
     p=NULL;
     chunk=(unsigned char *) NULL;
