From 6bd33275368edfa71ae117de895488584678ac79 Mon Sep 17 00:00:00 2001
From: Bron Gondwana <brong@fastmail.fm>
Date: Thu, 31 Aug 2017 12:43:09 +1000
Subject: [PATCH] mboxlist: fix uninitialised memory use where pattern is
 "Other Users"

---
 imap/mboxlist.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imap/mboxlist.c b/imap/mboxlist.c
index 413cdcebe5..dbee06c0f5 100644
--- a/imap/mboxlist.c
+++ b/imap/mboxlist.c
@@ -2851,7 +2851,7 @@ static int mboxlist_do_find(struct find_rock *rock, const strarray_t *patterns)
         if (len) len--; // trailing separator
 
         if (!strncmp(rock->namespace->prefix[NAMESPACE_USER], commonpat, MIN(len, prefixlen))) {
-            if (prefixlen < len) {
+            if (prefixlen <= len) {
                 /* we match all users */
                 strlcpy(domainpat+domainlen, "user.", sizeof(domainpat)-domainlen);
             }
