From 4d42ca1b5849518762d110f34b6ce2e03d6df9ec Mon Sep 17 00:00:00 2001
From: Juliusz Chroboczek <jch@pps.univ-paris-diderot.fr>
Date: Sun, 6 Nov 2016 19:34:34 +0100
Subject: [PATCH] Rephrase README.

---
 README | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/README b/README
index 33d803d..3c61ff9 100644
--- a/README
+++ b/README
@@ -1,11 +1,7 @@
-Polipo is obsolete
-==================
+Polipo is no longer maintained
+==============================
 
-When it was first written, Polipo was probably the best HTTP proxy
-available.  Since then, the web protocols have changed, and Polipo hasn't
-kept up -- it has no support for HTTP/2.0 or the WebSocket protocol.
-
-Polipo will no longer be maintained.
+Polipo is no longer maintained.  Sorry.
 
 
 Original Polipo README
