From 86a43ae47b3dd844947f40a2ffcf1a36e53dbba6 Mon Sep 17 00:00:00 2001
From: Johan Cwiklinski <jcwiklinski@teclib.com>
Date: Tue, 12 Mar 2019 10:59:54 +0100
Subject: [PATCH] Password token date was not removed

---
 inc/user.class.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/inc/user.class.php b/inc/user.class.php
index b96f845cc7d..4f482e3adaf 100755
--- a/inc/user.class.php
+++ b/inc/user.class.php
@@ -4515,7 +4515,7 @@ public function updateForgottenPassword(array $input) {
                }
                $input2 = [
                   'password_forget_token'      => '',
-                  'password_forget_token_date' => null,
+                  'password_forget_token_date' => 'NULL',
                   'id'                         => $this->fields['id']
                ];
                $this->update($input2);
