From cdb383749ef7b68a38891440af8cc23e0115306d Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Tue, 18 Jun 2019 11:44:39 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1599

---
 coders/cut.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/coders/cut.c b/coders/cut.c
index 69f236276b..c4f2768cbe 100644
--- a/coders/cut.c
+++ b/coders/cut.c
@@ -552,6 +552,7 @@ static Image *ReadCUTImage(const ImageInfo *image_info,ExceptionInfo *exception)
   BImgBuff=(unsigned char *) AcquireQuantumMemory((size_t) ldblk,
     sizeof(*BImgBuff));  /*Ldblk was set in the check phase*/
   if(BImgBuff==NULL) goto NoMemory;
+  (void) memset(BImgBuff,0,(size_t) ldblk*sizeof(*BImgBuff));
 
   offset=SeekBlob(image,6 /*sizeof(Header)*/,SEEK_SET);
   if (offset < 0)
