From b5ed738f8060266bf4ae521f7e3ed145aa4498a3 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 30 Jan 2016 08:49:15 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/110

---
 magick/quantum.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/magick/quantum.c b/magick/quantum.c
index 1065c2a80d..d7d2acb41e 100644
--- a/magick/quantum.c
+++ b/magick/quantum.c
@@ -682,7 +682,7 @@ MagickExport MagickBooleanType SetQuantumDepth(const Image *image,
     DestroyQuantumPixels(quantum_info);
   quantum=(quantum_info->pad+6)*(quantum_info->depth+7)/8;
   extent=image->columns*quantum;
-  if (quantum != (extent/image->columns))
+  if ((image->columns != 0) && (quantum != (extent/image->columns)))
     return(MagickFalse);
   return(AcquireQuantumPixels(quantum_info,extent));
 }
