From 87580d767868360d2fed503980129504da84b63e Mon Sep 17 00:00:00 2001
From: Hans-Christian Esperer <hc@hcesperer.org>
Date: Sat, 9 Jan 2016 19:58:58 +0100
Subject: [PATCH] Do not copy more bytes than were allocated

---
 modules/transport/xmlrpc/xmlrpclib.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules/transport/xmlrpc/xmlrpclib.c b/modules/transport/xmlrpc/xmlrpclib.c
index b67d215233..5fcde9e474 100644
--- a/modules/transport/xmlrpc/xmlrpclib.c
+++ b/modules/transport/xmlrpc/xmlrpclib.c
@@ -777,8 +777,9 @@ void xmlrpc_char_encode(char *outbuffer, const char *s1)
 			s->append_char(s, c);
 		}
 	}
+	s->append_char(s, 0);
 
-	memcpy(outbuffer, s->str, XMLRPC_BUFSIZE);
+	strncpy(outbuffer, s->str, XMLRPC_BUFSIZE);
 }
 
 static void xmlrpc_append_char_encode(mowgli_string_t *s, const char *s1)
