From 2c33634da7129a5aefcc262d2fec2e72224404e5 Mon Sep 17 00:00:00 2001
From: doowb <brian.woodward@gmail.com>
Date: Wed, 7 Feb 2018 11:43:20 -0500
Subject: [PATCH] exclude __proto__

---
 index.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/index.js b/index.js
index 11031ee..a7b8fc5 100644
--- a/index.js
+++ b/index.js
@@ -30,7 +30,7 @@ module.exports = function mergeDeep(orig, objects) {
 
 function merge(target, obj) {
   for (var key in obj) {
-    if (!hasOwn(obj, key)) {
+    if (key === '__proto__' || !hasOwn(obj, key)) {
       continue;
     }
 
