From 6d60bc8ed010c4eda05d6645c64849f415f68d65 Mon Sep 17 00:00:00 2001
From: Antonin Steinhauser <steinhauser@d3s.mff.cuni.cz>
Date: Fri, 26 Oct 2018 16:55:55 -0700
Subject: [PATCH] Context-sensitive XSS bugfix.

---
 app/helpers/tags_helper.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/helpers/tags_helper.rb b/app/helpers/tags_helper.rb
index 266060c6ea..397977682f 100755
--- a/app/helpers/tags_helper.rb
+++ b/app/helpers/tags_helper.rb
@@ -17,7 +17,7 @@ def tags_for_index(model)
       elsif !query.include?(hashtag)
         query += " #{hashtag}"
       end
-      out << link_to_function(tag, "crm.search_tagged('#{query}', '#{model.class.to_s.tableize}')", title: tag)
+      out << link_to_function(tag, "crm.search_tagged('#{escape_javascript(query)}', '#{model.class.to_s.tableize}')", title: tag)
     end
   end
 
