From b899d654d931f3bf3cfbbdd71e0d1a0f3a16d04c Mon Sep 17 00:00:00 2001
From: yurabakhtin <yurybakh@gmail.com>
Date: Wed, 9 Aug 2017 12:27:27 +0300
Subject: [PATCH] Format install params

---
 install/_functions_install.php | 26 +++++++++++++++++++-------
 1 file changed, 19 insertions(+), 7 deletions(-)

diff --git a/install/_functions_install.php b/install/_functions_install.php
index 27d76ae342..c4c2e16055 100644
--- a/install/_functions_install.php
+++ b/install/_functions_install.php
@@ -1791,6 +1791,18 @@ function check_quick_install_request()
 }
 
 
+/**
+ * Format an install param like DB config and base url
+ *
+ * @return string
+ */
+function format_install_param( $value )
+{
+	$value = str_replace( array( "'",  "\$" ), array( "\'", "\\$" ), $value );
+	return preg_replace( "#([\\\\]*)(\\\\\\\')#", "\\\\\\\\\\\\\\\\\'", $value );
+}
+
+
 /**
  * Update file /conf/_basic_config.php
  *
@@ -1891,13 +1903,13 @@ function update_basic_config_file( $params = array() )
 			),
 			array(
 				"\$db_config = array(\n"
-					."\t'user'     => '".str_replace( array( "'", "\$" ), array( "\'", "\\$" ), $params['db_user'] )."',\$1"
-					."\t'password' => '".str_replace( array( "'", "\$" ), array( "\'", "\\$" ), $params['db_password'] )."',\$2"
-					."\t'name'     => '".str_replace( array( "'", "\$" ), array( "\'", "\\$" ), $params['db_name'] )."',\$3"
-					."\t'host'     => '".str_replace( array( "'", "\$" ), array( "\'", "\\$" ), $params['db_host'] )."',\$4",
-				"tableprefix = '".str_replace( "'", "\'", $params['db_tableprefix'] )."';",
-				"baseurl = '".str_replace( "'", "\'", $params['baseurl'] )."';",
-				"admin_email = '".str_replace( "'", "\'", $params['admin_email'] )."';",
+					."\t'user'     => '".format_install_param( $params['db_user'] )."',\$1"
+					."\t'password' => '".format_install_param( $params['db_password'] )."',\$2"
+					."\t'name'     => '".format_install_param( $params['db_name'] )."',\$3"
+					."\t'host'     => '".format_install_param( $params['db_host'] )."',\$4",
+				"tableprefix = '".format_install_param( $params['db_tableprefix'] )."';",
+				"baseurl = '".format_install_param( $params['baseurl'] )."';",
+				"admin_email = '".format_install_param( $params['admin_email'] )."';",
 				'config_is_done = 1;',
 			), $conf );
 
