From 90406972f108c4da71f998601b06abdc2ac6f06e Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 1 Oct 2016 11:18:08 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/281

---
 MagickCore/attribute.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MagickCore/attribute.c b/MagickCore/attribute.c
index d9f088ad75..f6510b7bf3 100644
--- a/MagickCore/attribute.c
+++ b/MagickCore/attribute.c
@@ -1264,7 +1264,7 @@ MagickExport MagickBooleanType SetImageType(Image *image,const ImageType type,
           status=QuantizeImage(quantize_info,image,exception);
           quantize_info=DestroyQuantizeInfo(quantize_info);
         }
-      image->colors=2;
+      status=AcquireImageColormap(image,2,exception);
       image->alpha_trait=UndefinedPixelTrait;
       break;
     }
