From 46e3aabbf8d59a1bdebdbb65acb9b9e0484577d3 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 25 Jun 2017 09:23:22 -0400
Subject: [PATCH] ...

---
 coders/png.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/coders/png.c b/coders/png.c
index 1f75090cac..9efa1d0a2e 100644
--- a/coders/png.c
+++ b/coders/png.c
@@ -4305,6 +4305,8 @@ static Image *ReadOneJNGImage(MngInfo *mng_info,
             c;
 
           c=ReadBlobByte(image);
+          if (c == EOF)
+            break;
           chunk[i]=(unsigned char) c;
         }
 
@@ -5164,6 +5166,8 @@ static Image *ReadOneMNGImage(MngInfo* mng_info, const ImageInfo *image_info,
                 c;
 
               c=ReadBlobByte(image);
+              if (c == EOF)
+                break;
               chunk[i]=(unsigned char) c;
             }
 
