From 1d65ad6786282d23ba1865f56e2fd19188e7c26a Mon Sep 17 00:00:00 2001
From: Michael Yoshitaka Erlewine <mitcho@mitcho.com>
Date: Sun, 20 Mar 2016 23:41:37 +0800
Subject: [PATCH] escape url after add_query_arg

---
 shibboleth.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shibboleth.php b/shibboleth.php
index 4ecddc9..f03b5ef 100644
--- a/shibboleth.php
+++ b/shibboleth.php
@@ -460,7 +460,7 @@ function shibboleth_update_user_data($user_id, $force_update = false) {
 function shibboleth_login_form() {
 	$login_url = add_query_arg('action', 'shibboleth');
 	$login_url = remove_query_arg('reauth', $login_url);
-	echo '<p id="shibboleth_login"><a href="' . $login_url . '">' . __('Login with Shibboleth', 'shibboleth') . '</a></p>';
+	echo '<p id="shibboleth_login"><a href="' . esc_url($login_url) . '">' . __('Login with Shibboleth', 'shibboleth') . '</a></p>';
 }
 add_action('login_form', 'shibboleth_login_form');
 
