From 12786867d6faaceaec62c7c2cb5b0e2dc074d71a Mon Sep 17 00:00:00 2001
From: Kunpei Sakai <namusyaka@gmail.com>
Date: Thu, 31 May 2018 01:05:27 +0900
Subject: [PATCH] escape invalid query params, fixes #1428

---
 lib/sinatra/base.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/sinatra/base.rb b/lib/sinatra/base.rb
index 6843b4d375..727078d815 100644
--- a/lib/sinatra/base.rb
+++ b/lib/sinatra/base.rb
@@ -78,7 +78,7 @@ def unlink?
     def params
       super
     rescue Rack::Utils::ParameterTypeError, Rack::Utils::InvalidParameterError => e
-      raise BadRequest, "Invalid query parameters: #{e.message}"
+      raise BadRequest, "Invalid query parameters: #{Rack::Utils.escape_html(e.message)}"
     end
 
     private
