From 325027012783f108602f2a258565cdd8f61e9a8d Mon Sep 17 00:00:00 2001
From: "Shota Fukumori (sora_h)" <sorah@cookpad.com>
Date: Fri, 30 Oct 2015 12:38:01 +0900
Subject: [PATCH] Use array to pass arguments :lock:

---
 eventapp/lib/gcloud.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/eventapp/lib/gcloud.rb b/eventapp/lib/gcloud.rb
index c2895e9..c225e02 100644
--- a/eventapp/lib/gcloud.rb
+++ b/eventapp/lib/gcloud.rb
@@ -13,7 +13,7 @@ def self.valid_ip_address(project_id, zone_name, instance_name)
   end
 
   def self.server_info(project_id, zone_name, instance_name)
-    jsonText = IO.popen("gcloud compute --project #{project_id} instances list --zones #{zone_name} --format json") do |io|
+    jsonText = IO.popen([*%w(gcloud compute --project), project_id, %w(instances list --zones), zone_name, %w(--format json)]) do |io|
       io.read()
     end
     serverInfoList = JSON.parse(jsonText) rescue nil
