From 86f35221c8a57cdd7557ce731a56b90db216c8e0 Mon Sep 17 00:00:00 2001
From: wyclif <wyclif@openmrs.org>
Date: Fri, 16 Jun 2017 15:13:12 -0400
Subject: [PATCH] Added privilege check when previewing an uploaded html form

---
 .../web/controller/HtmlFormFromFileController.java              | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/omod/src/main/java/org/openmrs/module/htmlformentry/web/controller/HtmlFormFromFileController.java b/omod/src/main/java/org/openmrs/module/htmlformentry/web/controller/HtmlFormFromFileController.java
index a132a610b..6aebfade2 100644
--- a/omod/src/main/java/org/openmrs/module/htmlformentry/web/controller/HtmlFormFromFileController.java
+++ b/omod/src/main/java/org/openmrs/module/htmlformentry/web/controller/HtmlFormFromFileController.java
@@ -44,6 +44,8 @@ public void handleRequest(Model model, @RequestParam(value = "filePath", require
 	                          @RequestParam(value = "patientId", required = false) Integer pId,
 	                          @RequestParam(value = "isFileUpload", required = false) boolean isFileUpload,
 	                          HttpServletRequest request) throws Exception {
+
+        Context.requirePrivilege("Manage Forms");
 		
 		if (log.isDebugEnabled())
 			log.debug("In reference data...");
