From 70849ef375016a1061490c8c4744046dbfc3e679 Mon Sep 17 00:00:00 2001
From: Ken <kenny2minecraft@gmail.com>
Date: Sat, 29 Aug 2020 00:04:19 +0800
Subject: [PATCH] Disallow MW-invalid usernames from being picked

---
 ScratchLogin.common.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ScratchLogin.common.php b/ScratchLogin.common.php
index 67d3c15..b746039 100644
--- a/ScratchLogin.common.php
+++ b/ScratchLogin.common.php
@@ -37,6 +37,7 @@ function topVerifCommenter($req_comment) {
 	$comments = verifComments();
 
 	$matching_comments = array_filter($comments, function(&$comment) use($req_comment) {
+		if (preg_match('/^_+|_+$|__+/', $comment['author']['username'])) return false;
 		return stristr($comment['content'], $req_comment);
 	});
 	if (empty($matching_comments)) {
@@ -144,4 +145,4 @@ function doCodeReset($out, $request, $returnto) {
 		generateNewCodeForSession($request->getSession());
 		$out->addWikiMsg('scratchlogin-code-reset', $returnto);
 	}
-}
\ No newline at end of file
+}
