From 1e6a3ace073c9ec9c71e439c111d23c6e66cb6ae Mon Sep 17 00:00:00 2001
From: Cristy <mikayla-grace@urban-warrior.org>
Date: Sun, 20 Jan 2019 11:11:52 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1450

---
 coders/pcd.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/coders/pcd.c b/coders/pcd.c
index 8eb8ac6526..7e48c138c6 100644
--- a/coders/pcd.c
+++ b/coders/pcd.c
@@ -204,6 +204,8 @@ static MagickBooleanType DecodeImage(Image *image,unsigned char *luma,
     if (pcd_table[i] == (PCDTable *) NULL)
       {
         buffer=(unsigned char *) RelinquishMagickMemory(buffer);
+        for (j=0; j < i; j++)
+          pcd_table[j]=(PCDTable *) RelinquishMagickMemory(pcd_table[j]);
         ThrowBinaryException(ResourceLimitError,"MemoryAllocationFailed",
           image->filename);
       }
@@ -215,6 +217,8 @@ static MagickBooleanType DecodeImage(Image *image,unsigned char *luma,
       if (r->length > 16)
         {
           buffer=(unsigned char *) RelinquishMagickMemory(buffer);
+          for (j=0; j <= i; j++)
+            pcd_table[j]=(PCDTable *) RelinquishMagickMemory(pcd_table[j]);
           return(MagickFalse);
         }
       PCDGetBits(16);
