From a0cef507a90adaac7dbbe9c404f09a50bdefb348 Mon Sep 17 00:00:00 2001
From: Samuel Holmes <sam@edge.app>
Date: Tue, 3 Aug 2021 09:31:10 -0700
Subject: [PATCH] Patch: Filter out blacklisted keys in og property name

---
 index.js | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/index.js b/index.js
index 50a6020..be75277 100644
--- a/index.js
+++ b/index.js
@@ -10,6 +10,11 @@ var shorthandProperties = {
 	"audio": "audio:url"
 }
 
+var keyBlacklist = [
+	'__proto__',
+	'constructor',
+	'prototype'
+]
 
 exports = module.exports = function(url, cb, options){
   var userAgent = (options || {}).userAgent || 'NodeOpenGraphCrawler (https://github.com/samholmes/node-open-graph)'
@@ -122,6 +127,8 @@ exports.parse = function($, options){
 		while (keys.length > 1) {
 			key = keys.shift();
 
+			if (keyBlacklist.includes(key)) continue
+
 			if (Array.isArray(ptr[key])) {
 				// the last index of ptr[key] should become
 				// the object we are examining.
