From 41bd3645bdb616e1248b2167ca83636a2653f781 Mon Sep 17 00:00:00 2001
From: Ali Abid <you@example.comgit>
Date: Tue, 14 Dec 2021 21:01:55 +0000
Subject: [PATCH] secure path hotfix

---
 gradio/networking.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/gradio/networking.py b/gradio/networking.py
index 0a972816..18017249 100644
--- a/gradio/networking.py
+++ b/gradio/networking.py
@@ -377,15 +377,14 @@ def interpret():
 @app.route("/file/<path:path>", methods=["GET"])
 @login_check
 def file(path):
-    path = secure_filename(path)
     if app.interface.encrypt and isinstance(app.interface.examples, str) and path.startswith(app.interface.examples):
-        with open(os.path.join(app.cwd, path), "rb") as encrypted_file:
+        with open(safe_join(app.cwd, path), "rb") as encrypted_file:
             encrypted_data = encrypted_file.read()
         file_data = encryptor.decrypt(
             app.interface.encryption_key, encrypted_data)
         return send_file(io.BytesIO(file_data), attachment_filename=os.path.basename(path))
     else:
-        return send_file(os.path.join(app.cwd, path))
+        return send_file(safe_join(app.cwd, path))
 
 
 @app.route("/api/queue/push/", methods=["POST"])
