From d23beebe7b1179fb75db1e85fbca3100e49593d9 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 24 Dec 2016 11:44:42 -0500
Subject: [PATCH] ...

---
 coders/wpg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coders/wpg.c b/coders/wpg.c
index 7cd9162931..82c3b7e29e 100644
--- a/coders/wpg.c
+++ b/coders/wpg.c
@@ -803,7 +803,7 @@ static Image *ExtractPostscript(Image *image,const ImageInfo *image_info,
   if(exception->severity != UndefinedException) goto FINISH_UNL;
   if(magic_info->name == (char *) NULL) goto FINISH_UNL;
 
-  (void) strncpy(clone_info->magick,magic_info->name,MaxTextExtent);
+  (void) strncpy(clone_info->magick,magic_info->name,MaxTextExtent-1);
 
     /* Read nested image */
   /*FormatString(clone_info->filename,"%s:%s",magic_info->name,postscript_file);*/
