From c93bc100ec93fc78940fbdca9b6b009101858309 Mon Sep 17 00:00:00 2001
From: Nivash Srinivasa Ragavan <balanivash@gmail.com>
Date: Tue, 3 Feb 2015 23:15:21 -0600
Subject: [PATCH] Update index.php

---
 index.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/index.php b/index.php
index 7903f95..09ae3db 100755
--- a/index.php
+++ b/index.php
@@ -144,7 +144,7 @@
 ///Case 2 : request for a user profile page
 else if(isset($_GET['user'])) {
 	$publicPageRequest = true;
-	$userProfileId = $_GET['user'];
+	$userProfileId = safe_html(escape($_GET['user']));
 	//This is just to prevent parsing a NULL url when someone misplaces the code for User profile parser
 	$pageFullPath = "home";
 }
