From 8682b1e4e328432ba692bed52326e84439cec9e4 Mon Sep 17 00:00:00 2001
From: Andrey Sitnik <andrey@sitnik.ru>
Date: Thu, 8 Apr 2021 02:57:25 -0400
Subject: [PATCH] Fix unsafe regexp

---
 lib/previous-map.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/previous-map.js b/lib/previous-map.js
index 1a5bd1cf3..ff474a2bf 100644
--- a/lib/previous-map.js
+++ b/lib/previous-map.js
@@ -54,7 +54,7 @@ class PreviousMap {
   }
 
   loadAnnotation(css) {
-    let annotations = css.match(/\/\*\s*# sourceMappingURL=.*\s*\*\//gm)
+    let annotations = css.match(/\/\*\s*# sourceMappingURL=.*\*\//gm)
 
     if (annotations && annotations.length > 0) {
       // Locate the last sourceMappingURL to avoid picking up
