From 5c911ddeec4493f40f42e2f137800ed4692004a4 Mon Sep 17 00:00:00 2001
From: Tiago Daniel Jacobs <tiago.jacobs@gmail.com>
Date: Fri, 13 Nov 2020 05:59:31 +0000
Subject: [PATCH] Sanitize fullName parameter of join API. Fixes #10818

---
 .../org/bigbluebutton/web/controllers/ApiController.groovy      | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy b/bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy
index 54803a19edb..38610fde2cd 100755
--- a/bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy
+++ b/bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy
@@ -245,6 +245,8 @@ class ApiController {
     // Do we have a name for the user joining? If none, complain.
     if (!StringUtils.isEmpty(params.fullName)) {
       params.fullName = StringUtils.strip(params.fullName);
+      // remove control characters ( sanitize )
+      params.fullName = params.fullName.replaceAll("\\p{Cntrl}", "");
       if (StringUtils.isEmpty(params.fullName)) {
         errors.missingParamError("fullName");
       }
