From b32b410c9191c3c5db888514c29d7921f124d883 Mon Sep 17 00:00:00 2001
From: daniel <daniel@daniel-DC1A-S>
Date: Mon, 26 Aug 2019 18:28:37 -0300
Subject: [PATCH] secury fix to not allow recreate the config file

---
 install/checkConfiguration.php | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/install/checkConfiguration.php b/install/checkConfiguration.php
index 91ee8ad9f7..56c45c62c4 100644
--- a/install/checkConfiguration.php
+++ b/install/checkConfiguration.php
@@ -1,4 +1,8 @@
 <?php
+if (file_exists("../videos/configuration.php")) {
+    error_log("Can not create configuration again: ".  json_encode($_SERVER));
+    exit;
+}
 
 $installationVersion = "7.5";
 
