From 659864b393961bf224bba1efc03b7dcbed7de533 Mon Sep 17 00:00:00 2001
From: Angel Lai <catfan.me@gmail.com>
Date: Fri, 11 Oct 2019 23:50:40 +0800
Subject: [PATCH] [fix] Fix columnQuote for security issue reported by Snyk

---
 src/Medoo.php | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Medoo.php b/src/Medoo.php
index 6d68f4c..478e0a8 100644
--- a/src/Medoo.php
+++ b/src/Medoo.php
@@ -500,6 +500,11 @@ protected function typeMap($value, $type)
 
 	protected function columnQuote($string)
 	{
+		if (!preg_match('/^[a-zA-Z0-9]+(\.?[a-zA-Z0-9]+)?$/i', $string))
+		{
+			throw new InvalidArgumentException("Incorrect column name \"$string\"");
+		}
+
 		if (strpos($string, '.') !== false)
 		{
 			return '"' . $this->prefix . str_replace('.', '"."', $string) . '"';
