From 33e7199575631ba1f74cba6b16b10c820bec59af Mon Sep 17 00:00:00 2001
From: Artur Heinze <artur@agentejo.com>
Date: Mon, 14 Sep 2020 20:23:06 +0200
Subject: [PATCH] don't allow callable strings for $func (MongoLite)

---
 lib/MongoLite/Database.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/MongoLite/Database.php b/lib/MongoLite/Database.php
index fb47438cc..08141f405 100644
--- a/lib/MongoLite/Database.php
+++ b/lib/MongoLite/Database.php
@@ -429,7 +429,7 @@ private static function evaluate($func, $a, $b) {
             case '$func' :
             case '$fn' :
             case '$f' :
-                if (! \is_callable($b))
+                if (\is_string($b) || !\is_callable($b))
                     throw new \InvalidArgumentException('Function should be callable');
                 $r = $b($a);
                 break;
