From 224d16f315f264f2b1c952ce59705c97fd6882f0 Mon Sep 17 00:00:00 2001
From: Pierre RAMBAUD <pierre.rambaud86@gmail.com>
Date: Wed, 29 Jul 2020 16:14:12 +0200
Subject: [PATCH] Make sure files are downloaded when reaching the upload
 directory

---
 classes/Tools.php | 5 +++++
 upload/.htaccess  | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/classes/Tools.php b/classes/Tools.php
index a82a5c720f38..cf8208e3e572 100644
--- a/classes/Tools.php
+++ b/classes/Tools.php
@@ -2646,6 +2646,11 @@ public static function generateHtaccess($path = null, $rewrite_settings = null,
 	<FilesMatch \"\.(ttf|ttc|otf|eot|woff|woff2|svg)$\">
 		Header set Access-Control-Allow-Origin \"*\"
 	</FilesMatch>
+
+    <FilesMatch \"\.pdf$\">
+      Header set Content-Disposition \"Attachment\"
+      Header set X-Content-Type-Options \"nosniff\"
+    </FilesMatch>
 </IfModule>\n\n");
         fwrite($write_fd, "<Files composer.lock>
     # Apache 2.2
diff --git a/upload/.htaccess b/upload/.htaccess
index 39ef42e7157a..235cff7e4750 100644
--- a/upload/.htaccess
+++ b/upload/.htaccess
@@ -5,3 +5,8 @@
 <IfModule mod_php5.c>
     php_flag engine off
 </IfModule>
+
+<IfModule mod_headers.c>
+  Header set Content-Disposition "Attachment"
+  Header set X-Content-Type-Options "nosniff"
+</IfModule>
