From 7647a7b53ceab31524cfcfb3beb8435af0a30fc1 Mon Sep 17 00:00:00 2001
From: Francisco Mancardi <francisco.mancardi@gmail.com>
Date: Mon, 30 Dec 2019 09:59:16 +0100
Subject: [PATCH] fix: security #8829

---
 lib/testcases/tcCompareVersions.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/testcases/tcCompareVersions.php b/lib/testcases/tcCompareVersions.php
index 327cf10739..0cf2a3ebca 100644
--- a/lib/testcases/tcCompareVersions.php
+++ b/lib/testcases/tcCompareVersions.php
@@ -85,7 +85,8 @@ function init_args()
 	
 
   $args->tcase_id = isset($_REQUEST['testcase_id']) ? $_REQUEST['testcase_id'] : 0;
-
+  $args->tcase_id = intval($args->tcase_id);
+ 
   $key2set = array('compare_selected_versions' => 0,'version_left' => '','version_right' => '');
   foreach($key2set as $tk => $value)
   {
