From f8c318d462270b03e77f082e2a3a32867cacd3c6 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 24 Jan 2016 19:10:07 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/96

---
 coders/meta.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/coders/meta.c b/coders/meta.c
index 53fde4c0d0..96ad6c8f19 100644
--- a/coders/meta.c
+++ b/coders/meta.c
@@ -461,7 +461,7 @@ static ssize_t parse8BIM(Image *ifile, Image *ofile)
 
                     next=0;
                     outputlen += len;
-                    while (len--)
+                    while (len-- > 0)
                       (void) WriteBlobByte(ofile,(unsigned char) token[next++]);
 
                     if (outputlen & 1)
@@ -492,7 +492,7 @@ static ssize_t parse8BIM(Image *ifile, Image *ofile)
                     outputlen += 5;
                     next=0;
                     outputlen += len;
-                    while (len--)
+                    while (len-- > 0)
                       (void) WriteBlobByte(ofile,(unsigned char) token[next++]);
                   }
               }
