From 03d1222ab9c2ce779cdf01bdb96cdd69cbdcfeda Mon Sep 17 00:00:00 2001
From: Gilles Boccon-Gibod <bok@bok.net>
Date: Sat, 9 Sep 2017 16:12:31 -0700
Subject: [PATCH] fix for #190

---
 Source/C++/Core/Ap4SampleEntry.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Source/C++/Core/Ap4SampleEntry.cpp b/Source/C++/Core/Ap4SampleEntry.cpp
index d29d08a9..6bdf6f8f 100644
--- a/Source/C++/Core/Ap4SampleEntry.cpp
+++ b/Source/C++/Core/Ap4SampleEntry.cpp
@@ -772,13 +772,13 @@ AP4_VisualSampleEntry::ReadFields(AP4_ByteStream& stream)
     stream.ReadUI32(m_Reserved3);
     stream.ReadUI16(m_FrameCount);
 
-    char compressor_name[33];
+    AP4_UI08 compressor_name[33];
     compressor_name[32] = 0;
     stream.Read(compressor_name, 32);
-    int name_length = compressor_name[0];
+    AP4_UI08 name_length = compressor_name[0];
     if (name_length < 32) {
         compressor_name[name_length+1] = 0; // force null termination
-        m_CompressorName = &compressor_name[1];
+        m_CompressorName = (const char*)(&compressor_name[1]);
     }
 
     stream.ReadUI16(m_Depth);
