From 66a28b9275551384fdab45f3591a82d6b59952cb Mon Sep 17 00:00:00 2001
From: Jonathan Claudius <jclaudius@mozilla.com>
Date: Thu, 1 Feb 2018 23:04:33 -0500
Subject: [PATCH] Fix 289313

---
 lib/rubygems/server.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/rubygems/server.rb b/lib/rubygems/server.rb
index 93b3af36f8a4..31a6d0234ef8 100644
--- a/lib/rubygems/server.rb
+++ b/lib/rubygems/server.rb
@@ -632,7 +632,7 @@ def root(req, res)
         "only_one_executable" => (executables && executables.size == 1),
         "full_name"           => spec.full_name,
         "has_deps"            => !deps.empty?,
-        "homepage"            => spec.homepage,
+        "homepage"            => (URI.parse(spec.homepage).is_a?(URI::HTTP) || URI.parse(spec.homepage).is_a?(URI::HTTPS)) ? spec.homepage : ".",
         "name"                => spec.name,
         "rdoc_installed"      => Gem::RDoc.new(spec).rdoc_installed?,
         "ri_installed"        => Gem::RDoc.new(spec).ri_installed?,
