From 30e29ed2cdaec8684d60a6d12ea07b611c765a12 Mon Sep 17 00:00:00 2001
From: KMcB <cavrois.cedric@hotmail.fr>
Date: Tue, 5 May 2020 17:04:29 +0200
Subject: [PATCH] expose ignoreIncludes on mjml2html

---
 packages/mjml-core/src/index.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/packages/mjml-core/src/index.js b/packages/mjml-core/src/index.js
index 9bc466aa2..039b956fe 100644
--- a/packages/mjml-core/src/index.js
+++ b/packages/mjml-core/src/index.js
@@ -82,6 +82,7 @@ export default function mjml2html(mjml, options = {}) {
     keepComments,
     minify = false,
     minifyOptions = {},
+    ignoreIncludes = false,
     juiceOptions = {},
     juicePreserveTags = null,
     skeleton = defaultSkeleton,
@@ -102,6 +103,7 @@ export default function mjml2html(mjml, options = {}) {
       filePath,
       actualPath,
       preprocessors,
+      ignoreIncludes,
     })
   }
 
