From 9ab536235bafc2b42c3c17d7ce26f1cc64482a81 Mon Sep 17 00:00:00 2001
From: Flame442 <34169552+Flame442@users.noreply.github.com>
Date: Sun, 9 Aug 2020 19:11:15 -0400
Subject: [PATCH] Remove an unnecessary `.format`

---
 redbot/cogs/trivia/trivia.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/redbot/cogs/trivia/trivia.py b/redbot/cogs/trivia/trivia.py
index b428605fd6d..4f80edd6e01 100644
--- a/redbot/cogs/trivia/trivia.py
+++ b/redbot/cogs/trivia/trivia.py
@@ -539,7 +539,7 @@ def _get_leaderboard(data: dict, key: str, top: int):
             )
             padding = [" " * (len(h) - len(f)) for h, f in zip(headers, fields)]
             fields = tuple(f + padding[i] for i, f in enumerate(fields))
-            lines.append(" | ".join(fields).format(member=member, **m_data))
+            lines.append(" | ".join(fields))
             if rank == top:
                 break
         return "\n".join(lines)
