From b9150b7194f8cfb1178ca3674a0b3c86b7cd92f5 Mon Sep 17 00:00:00 2001
From: mjansen <mjansen@databay.de>
Date: Tue, 11 Dec 2018 08:41:36 +0100
Subject: [PATCH] Test: String handling in cloze question

---
 .../TestQuestionPool/classes/class.assClozeTestGUI.php    | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Modules/TestQuestionPool/classes/class.assClozeTestGUI.php b/Modules/TestQuestionPool/classes/class.assClozeTestGUI.php
index b4ebd09e08dd..01af25a924f5 100755
--- a/Modules/TestQuestionPool/classes/class.assClozeTestGUI.php
+++ b/Modules/TestQuestionPool/classes/class.assClozeTestGUI.php
@@ -1547,7 +1547,7 @@ public function getAggregatedAnswersView($relevant_answers)
 				foreach($gap->getItems($this->object->getShuffler()) as $gap_item)
 				{
 					$aggregate = $aggregation[$i];
-					$html .= '<li>' . $gap_item->getAnswerText() . ' - ' . ($aggregate[$j] ? $aggregate[$j] : 0) . '</li>';
+					$html .= '<li>' . ilUtil::prepareFormOutput($gap_item->getAnswerText()) . ' - ' . ($aggregate[$j] ? $aggregate[$j] : 0) . '</li>';
 					$j++;
 				}
 				$html .= '</ul>';
@@ -1573,8 +1573,8 @@ public function getAggregatedAnswersView($relevant_answers)
 						$show_mover = ' style="display: none;" ';
 					}
 
-					$html .= '<li>' . $answer . ' - ' . $count
-						. '&nbsp;<button class="clone_fields_add btn btn-link" ' . $show_mover . ' data-answer="'.$answer.'" name="add_gap_'.$i.'_0">
+					$html .= '<li>' . ilUtil::prepareFormOutput($answer) . ' - ' . $count
+						. '&nbsp;<button class="clone_fields_add btn btn-link" ' . $show_mover . ' data-answer="'.ilUtil::prepareFormOutput($answer).'" name="add_gap_'.$i.'_0">
 						<span class="sr-only"></span><span class="glyphicon glyphicon-plus"></span></button>
 						</li>';
 				}
@@ -1591,7 +1591,7 @@ public function getAggregatedAnswersView($relevant_answers)
 					$aggregate = (array)$aggregation[$i];
 					foreach($aggregate as $answer => $count)
 					{
-						$html .= '<li>' . $answer . ' - ' . $count . '</li>';
+						$html .= '<li>' . ilUtil::prepareFormOutput($answer) . ' - ' . $count . '</li>';
 					}
 					$j++;
 				}
