From 256041d8cab2fdc50678f4d49d916fea25cdb4fd Mon Sep 17 00:00:00 2001
From: RemRem <remrem@users.noreply.github.com>
Date: Mon, 11 Dec 2017 23:43:22 +0100
Subject: [PATCH] fix form email security issue

---
 admin/preferences.php | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/admin/preferences.php b/admin/preferences.php
index 3c67d83..a1b5d6c 100755
--- a/admin/preferences.php
+++ b/admin/preferences.php
@@ -174,7 +174,7 @@ function validate_form_preferences()
     $errors = array();
     $token = (string)filter_input(INPUT_POST, 'token');
     $author = (string)filter_input(INPUT_POST, 'auteur');
-    $email = (string)filter_input(INPUT_POST, 'email');
+    $email = (string)filter_input(INPUT_POST, 'email', FILTER_VALIDATE_EMAIL);
     $root = (string)filter_input(INPUT_POST, 'racine');
     $username = (string)filter_input(INPUT_POST, 'identifiant');
     $password = (string)filter_input(INPUT_POST, 'mdp');
@@ -186,11 +186,11 @@ function validate_form_preferences()
     if (!strlen(trim($author))) {
         $errors[] = $GLOBALS['lang']['err_prefs_auteur'];
     }
-    if ($GLOBALS['require_email'] == 1) {
-        if (!preg_match('#^[\w.+~\'*-]+@[\w.-]+\.[a-zA-Z]{2,6}$#i', trim($email))) {
+    // if ($GLOBALS['require_email'] == 1) {
+        if ($email === null || !preg_match('#^[\w.+~\'*-]+@[\w.-]+\.[a-zA-Z]{2,6}$#i', trim($email))) {
             $errors[] = $GLOBALS['lang']['err_prefs_email'] ;
         }
-    }
+    // }
     if (!preg_match('#^(https?://).*/$#', $root)) {
         $errors[] = $GLOBALS['lang']['err_prefs_racine_slash'];
     }
