From f663dfb8431c97d95682a2b533cca1c8233d21b4 Mon Sep 17 00:00:00 2001
From: Cristy <mikayla-grace@urban-warrior.org>
Date: Sun, 14 Apr 2019 11:49:45 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1546

---
 coders/xwd.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/coders/xwd.c b/coders/xwd.c
index 186b3f1ca2..94a2907c93 100644
--- a/coders/xwd.c
+++ b/coders/xwd.c
@@ -243,6 +243,9 @@ static Image *ReadXWDImage(const ImageInfo *image_info,ExceptionInfo *exception)
     ThrowReaderException(CorruptImageError,"ImproperImageHeader");
   if ((header.bits_per_pixel == 0) || (header.bits_per_pixel > 32))
     ThrowReaderException(CorruptImageError,"ImproperImageHeader");
+  if ((header.bitmap_bit_order != MSBFirst) &&
+      (header.bitmap_bit_order != LSBFirst))
+    ThrowReaderException(CorruptImageError,"ImproperImageHeader");
   if (((header.bitmap_pad % 8) != 0) || (header.bitmap_pad > 32))
     ThrowReaderException(CorruptImageError,"ImproperImageHeader");
   if (header.bitmap_unit > 32)
