From 210fdff7a65370c2ae24e022a2b35da8c423cc5f Mon Sep 17 00:00:00 2001
From: xzyfer <xzyfer@gmail.com>
Date: Thu, 21 Jun 2018 20:10:55 +1000
Subject: [PATCH] Remove legacy workaround for parsing modulo operator

Fixes #2659
---
 src/parser.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/parser.cpp b/src/parser.cpp
index ee51d56b6a..3053aaf4fb 100644
--- a/src/parser.cpp
+++ b/src/parser.cpp
@@ -1723,10 +1723,6 @@ namespace Sass {
     if (lex< variable >())
     { return SASS_MEMORY_NEW(Variable, pstate, Util::normalize_underscores(lexed)); }
 
-    // Special case handling for `%` proceeding an interpolant.
-    if (lex< sequence< exactly<'%'>, optional< percentage > > >())
-    { return SASS_MEMORY_NEW(String_Constant, pstate, lexed); }
-
     css_error("Invalid CSS", " after ", ": expected expression (e.g. 1px, bold), was ");
 
     // unreachable statement
