From 25b77a2ff8deabe8e8894002b9a5647eaec27b0a Mon Sep 17 00:00:00 2001
From: Daniel Hensby <dhensby@silverstripe.com>
Date: Thu, 25 May 2017 13:07:53 +0100
Subject: [PATCH] [SS-2017-017] FIX SVG uploads disabled by default

---
 filesystem/File.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/filesystem/File.php b/filesystem/File.php
index f6aca1d1f46..5750414ac49 100644
--- a/filesystem/File.php
+++ b/filesystem/File.php
@@ -129,7 +129,7 @@ class File extends DataObject {
 		'','ace','arc','arj','asf','au','avi','bmp','bz2','cab','cda','css','csv','dmg','doc','docx','dotx','dotm',
 		'flv','gif','gpx','gz','hqx','ico','jar','jpeg','jpg','js','kml', 'm4a','m4v',
 		'mid','midi','mkv','mov','mp3','mp4','mpa','mpeg','mpg','ogg','ogv','pages','pcx','pdf','pkg',
-		'png','pps','ppt','pptx','potx','potm','ra','ram','rm','rtf','sit','sitx', 'svg', 'tar','tgz','tif','tiff',
+		'png','pps','ppt','pptx','potx','potm','ra','ram','rm','rtf','sit','sitx', 'tar','tgz','tif','tiff',
 		'txt','wav','webm','wma','wmv','xls','xlsx','xltx','xltm','zip','zipx',
 	);
 
