From 0f35fe14acf0d0d3b55e9337bc5e2b9640ff2372 Mon Sep 17 00:00:00 2001
From: Marek Hulan <mhulan@redhat.com>
Date: Wed, 10 Aug 2016 09:45:29 +0200
Subject: [PATCH] Fixes #16024 - escape labels of form fields

---
 app/helpers/form_helper.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/helpers/form_helper.rb b/app/helpers/form_helper.rb
index 0db9635a046..2e7d451fb2e 100644
--- a/app/helpers/form_helper.rb
+++ b/app/helpers/form_helper.rb
@@ -283,7 +283,7 @@ def add_label options, f, attr
     label = options[:label] == :none ? '' : options.delete(:label)
     label ||= ((clazz = f.object.class).respond_to?(:gettext_translation_for_attribute_name) &&
         s_(clazz.gettext_translation_for_attribute_name attr)) if f
-    label = label.present? ? label_tag(attr, "#{label}#{required_mark}".html_safe, :class => label_size + " control-label") : ''
+    label = label.present? ? label_tag(attr, "#{label}#{required_mark}", :class => label_size + " control-label") : ''
     label
   end
 
