From 22d2f7a1ec93300c34e8c42d14ec39d51e610b5c Mon Sep 17 00:00:00 2001
From: Colton Myers <colton.myers@gmail.com>
Date: Wed, 25 Mar 2015 16:01:20 -0600
Subject: [PATCH] Use cachedir

---
 salt/modules/chef.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/salt/modules/chef.py b/salt/modules/chef.py
index 9c6d928fc3b9..40d01f10dc1e 100644
--- a/salt/modules/chef.py
+++ b/salt/modules/chef.py
@@ -29,7 +29,10 @@ def _default_logfile(exe_name):
     Retrieve the logfile name
     '''
     if salt.utils.is_windows():
-        logfile_tmp = tempfile.NamedTemporaryFile(dir=os.environ['TMP'],
+        tmp_dir = os.path.join(__opts__['cachedir'], 'tmp')
+        if not os.path.isdir(tmp_dir):
+            os.mkdir(tmp_dir)
+        logfile_tmp = tempfile.NamedTemporaryFile(dir=tmp_dir,
                                                   prefix=exe_name,
                                                   suffix='.log',
                                                   delete=False)
