From 53c4137bd924b954432c6c59da7572c4c5ffa901 Mon Sep 17 00:00:00 2001
From: Bron Gondwana <brong@fastmail.fm>
Date: Fri, 11 Aug 2017 11:58:37 +1000
Subject: [PATCH] imapd: check for isadmin BEFORE parsing sync lines

---
 imap/imapd.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/imap/imapd.c b/imap/imapd.c
index 49b3f0e800..9ac96d2bfe 100644
--- a/imap/imapd.c
+++ b/imap/imapd.c
@@ -2062,6 +2062,8 @@ static void cmdloop(void)
                 snmp_increment(SCAN_COUNT, 1);
             }
             else if (!strcmp(cmd.s, "Syncapply")) {
+                if (!imapd_userisadmin) goto badcmd;
+
                 struct dlist *kl = sync_parseline(imapd_in);
 
                 if (kl) {
@@ -2071,6 +2073,8 @@ static void cmdloop(void)
                 else goto extraargs;
             }
             else if (!strcmp(cmd.s, "Syncget")) {
+                if (!imapd_userisadmin) goto badcmd;
+
                 struct dlist *kl = sync_parseline(imapd_in);
 
                 if (kl) {
@@ -2080,6 +2084,8 @@ static void cmdloop(void)
                 else goto extraargs;
             }
             else if (!strcmp(cmd.s, "Syncrestart")) {
+                if (!imapd_userisadmin) goto badcmd;
+
                 if (c == '\r') c = prot_getc(imapd_in);
                 if (c != '\n') goto extraargs;
 
@@ -2087,6 +2093,8 @@ static void cmdloop(void)
                 cmd_syncrestart(tag.s, &reserve_list, 1);
             }
             else if (!strcmp(cmd.s, "Syncrestore")) {
+                if (!imapd_userisadmin) goto badcmd;
+
                 struct dlist *kl = sync_parseline(imapd_in);
 
                 if (kl) {
