From 3627aaf7d289315b614a584afb388f04abfb5bbf Mon Sep 17 00:00:00 2001
From: akallabeth <akallabeth@posteo.net>
Date: Tue, 31 Mar 2020 14:32:01 +0200
Subject: [PATCH] Fixed #6011: Bounds check in rdp_read_font_capability_set

---
 libfreerdp/core/capabilities.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libfreerdp/core/capabilities.c b/libfreerdp/core/capabilities.c
index be0c2d5a6b4..93ba8e19d3b 100644
--- a/libfreerdp/core/capabilities.c
+++ b/libfreerdp/core/capabilities.c
@@ -1397,10 +1397,10 @@ static BOOL rdp_print_input_capability_set(wStream* s, UINT16 length)
 static BOOL rdp_read_font_capability_set(wStream* s, UINT16 length, rdpSettings* settings)
 {
 	WINPR_UNUSED(settings);
-	if (length > 4)
+	if (length > 5)
 		Stream_Seek_UINT16(s); /* fontSupportFlags (2 bytes) */
 
-	if (length > 6)
+	if (length > 7)
 		Stream_Seek_UINT16(s); /* pad2Octets (2 bytes) */
 
 	return TRUE;
