From d5559407ce29f4371e5df9c1cbde65455fe5854c Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Mon, 17 Jul 2017 07:35:30 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/572

---
 coders/inline.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/coders/inline.c b/coders/inline.c
index 318624e1c5..4bf2ced92e 100644
--- a/coders/inline.c
+++ b/coders/inline.c
@@ -316,7 +316,10 @@ static MagickBooleanType WriteINLINEImage(const ImageInfo *image_info,
   magick_info=GetMagickInfo(write_info->magick,exception);
   if ((magick_info == (const MagickInfo *) NULL) ||
       (GetMagickMimeType(magick_info) == (const char *) NULL))
-    ThrowWriterException(CorruptImageError,"ImageTypeNotSupported");
+    {
+      write_info=DestroyImageInfo(write_info);
+      ThrowWriterException(CorruptImageError,"ImageTypeNotSupported");
+    }
   (void) CopyMagickString(image->filename,write_info->filename,
     MagickPathExtent);
   blob_length=2048;
