From cc55f44a4a90147a007dee1aefa1cb529e23798b Mon Sep 17 00:00:00 2001
From: Madhura Jayaratne <madhura.cj@gmail.com>
Date: Mon, 29 Feb 2016 11:12:07 +1100
Subject: [PATCH] Fix XSS in Central columns page

Signed-off-by: Madhura Jayaratne <madhura.cj@gmail.com>
---
 db_central_columns.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/db_central_columns.php b/db_central_columns.php
index 2e9aac73bbf..4c78eba255c 100644
--- a/db_central_columns.php
+++ b/db_central_columns.php
@@ -92,7 +92,7 @@
 } else {
     $total_rows = PMA_getCentralColumnsCount($db);
 }
-if (isset($_REQUEST['pos'])) {
+if (PMA_isValid($_REQUEST['pos'], 'integer')) {
     $pos = $_REQUEST['pos'];
 } else {
     $pos = 0;
