From e8b58d03534ade8d83c2d1946a8350a23952531e Mon Sep 17 00:00:00 2001
From: John McLear <john@mclear.co.uk>
Date: Wed, 1 Apr 2020 18:05:33 +0000
Subject: [PATCH] go go gadget mysql fix for 2877

---
 mysql_db.js  | 4 ++--
 package.json | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/mysql_db.js b/mysql_db.js
index bf19f85..806c670 100644
--- a/mysql_db.js
+++ b/mysql_db.js
@@ -129,7 +129,7 @@ exports.database.prototype.init = function(callback)
 
 exports.database.prototype.get = function (key, callback)
 {
-  this.db.query("SELECT `value` FROM `store` WHERE  `key` = ?", [key], function(err,results)
+  this.db.query("SELECT `value` FROM `store` WHERE BINARY `key` = ?", [key], function(err,results)
   {
     var value = null;
     
@@ -146,7 +146,7 @@ exports.database.prototype.get = function (key, callback)
 
 exports.database.prototype.findKeys = function (key, notKey, callback)
 {
-  var query="SELECT `key` FROM `store` WHERE  `key` LIKE ?"
+  var query="SELECT `key` FROM `store` WHERE BINARY `key` LIKE ?"
     , params=[]
   ;
   
diff --git a/package.json b/package.json
index 29ee791..0045cdd 100644
--- a/package.json
+++ b/package.json
@@ -43,7 +43,7 @@
     "url": "https://github.com/ether/ueberDB.git"
   },
   "main": "./CloneAndAtomicLayer",
-  "version": "0.4.3",
+  "version": "0.4.4",
   "gitHead": "00cb8c896433922111c12fe9d2a45087fdb11d59",
   "bugs": {
     "url": "https://github.com/ether/ueberDB/issues"
