From ea5b1f377e196a4492e05ff070eba8b30b7372c4 Mon Sep 17 00:00:00 2001
From: Guy Mograbi <guy.mograbi@gmail.com>
Date: Wed, 24 Mar 2021 07:00:50 -0700
Subject: [PATCH] fix kill by port

---
 index.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/index.js b/index.js
index a827f37..e19507a 100644
--- a/index.js
+++ b/index.js
@@ -5,7 +5,7 @@ const exec = require('child_process').execSync
 exports.killByPort = function (port) {
   var processId = null
   try {
-    processId = exec(`lsof -t -i:${port}`)
+    processId = exec(`lsof -t -i:${parseInt(port, 10)}`)
   } catch (e) {
 
   }
