From 632e61961553a5cdd4c12ad7218e914455dbaa6b Mon Sep 17 00:00:00 2001
From: Pierre RAMBAUD <pierre.rambaud86@gmail.com>
Date: Wed, 18 Mar 2020 12:13:09 +0100
Subject: [PATCH] Check that the url is a real url accepting only http & https

---
 src/Form/Type/CustomUrlType.php | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Form/Type/CustomUrlType.php b/src/Form/Type/CustomUrlType.php
index da78efe..943f38f 100644
--- a/src/Form/Type/CustomUrlType.php
+++ b/src/Form/Type/CustomUrlType.php
@@ -29,6 +29,7 @@
 use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Validator\Constraints as Assert;
 
 class CustomUrlType extends TranslatorAwareType
 {
@@ -45,6 +46,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ->add('url', TextType::class, [
                 'label' => $this->trans('URL', 'Modules.Linklist.Admin'),
                 'required' => true,
+                'constraints' => [new Assert\Url()],
             ])
         ;
     }
