From ad55822430a03fe075221b543efb434567e9e431 Mon Sep 17 00:00:00 2001
From: pancake <pancake@nopcode.org>
Date: Thu, 23 Feb 2017 19:17:28 +0100
Subject: [PATCH] Fix #6836 - oob write in dex

---
 libr/bin/p/bin_dex.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libr/bin/p/bin_dex.c b/libr/bin/p/bin_dex.c
index 601b28a5d71..9a91dae9c69 100644
--- a/libr/bin/p/bin_dex.c
+++ b/libr/bin/p/bin_dex.c
@@ -309,8 +309,8 @@ static void dex_parse_debug_item(RBinFile *binfile, RBinDexObj *bin,
 	ut64 source_file_idx = c->source_file;
 	RList *params, *debug_positions, *emitted_debug_locals = NULL; 
 	bool keep = true;
-	if (argReg >= regsz) {
-		//return; // this return breaks tests
+	if (argReg > regsz) {
+		return; // this return breaks tests
 	}
 	p4 = r_uleb128 (p4, p4_end - p4, &line_start);
 	p4 = r_uleb128 (p4, p4_end - p4, &parameters_size);
