From b6ae2f9e0ab13343c0281732d479757a8e8979c7 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 30 Jan 2016 10:20:32 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/102

---
 coders/wpg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coders/wpg.c b/coders/wpg.c
index 46d843d421..ea987fd31a 100644
--- a/coders/wpg.c
+++ b/coders/wpg.c
@@ -325,7 +325,7 @@ static void InsertRow(unsigned char *p,ssize_t y,Image *image, int bpp)
         if (q == (PixelPacket *) NULL)
           break;
         indexes=GetAuthenticIndexQueue(image);
-        for (x=0; x < ((ssize_t) image->columns-1); x+=2)
+        for (x=0; x < ((ssize_t) image->columns-1); x+=4)
         {
             index=ConstrainColormapIndex(image,(*p >> 6) & 0x3);
             SetPixelIndex(indexes+x,index);
