From f3b483e8b054c50149912523b4773687e18afe25 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 14 Feb 2016 09:24:32 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/131

---
 coders/mat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coders/mat.c b/coders/mat.c
index b451c40d77..b81dc6880e 100644
--- a/coders/mat.c
+++ b/coders/mat.c
@@ -875,7 +875,7 @@ RestoreMSCWarning
       }
 
   /* ----- Load raster data ----- */
-    BImgBuff = (unsigned char *) AcquireQuantumMemory((size_t) (ldblk),sizeof(unsigned char));    /* Ldblk was set in the check phase */
+    BImgBuff = (unsigned char *) AcquireQuantumMemory((size_t) (ldblk),sizeof(double));    /* Ldblk was set in the check phase */
     if (BImgBuff == NULL)
       ThrowReaderException(ResourceLimitError,"MemoryAllocationFailed");
 
