From 04178de2247e353fc095846784b9a10fefdbf890 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Wed, 9 Aug 2017 08:01:38 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/662

---
 coders/mat.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/coders/mat.c b/coders/mat.c
index e3b80f14b9..fc8da2b144 100644
--- a/coders/mat.c
+++ b/coders/mat.c
@@ -1330,8 +1330,6 @@ RestoreMSCWarning
   if (quantum_info != (QuantumInfo *) NULL)
     quantum_info=DestroyQuantumInfo(quantum_info);
 END_OF_READING:
-  if (clone_info)
-    clone_info=DestroyImageInfo(clone_info);
   CloseBlob(image);
 
 
@@ -1349,6 +1347,8 @@ RestoreMSCWarning
         Image *tmp=p;
         if ((p->rows == 0) || (p->columns == 0)) {
           p=p->previous;
+          if (tmp == image2)
+            image2=(Image *) NULL;
           DeleteImageFromList(&tmp);
         } else {
           image=p;
