From 4043718264095cde6623c2cbe8c644541036d7bf Mon Sep 17 00:00:00 2001
From: "Daniel Knoppel (Phusion)" <daniel@phusion.nl>
Date: Wed, 11 Oct 2017 15:55:07 +0200
Subject: [PATCH] Disable unused feature.

---
 src/agent/Core/SpawningKit/Spawner.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/agent/Core/SpawningKit/Spawner.h b/src/agent/Core/SpawningKit/Spawner.h
index e65034ae9..99dedb852 100644
--- a/src/agent/Core/SpawningKit/Spawner.h
+++ b/src/agent/Core/SpawningKit/Spawner.h
@@ -721,7 +721,6 @@ class Spawner {
 		prepareChroot(info, options);
 		info.userSwitching = prepareUserSwitching(options);
 		prepareSwitchingWorkingDirectory(info, options);
-		inferApplicationInfo(info);
 		return info;
 	}
 
@@ -775,6 +774,7 @@ class Spawner {
 		assert(info.appRootPathsInsideChroot.back() == info.appRootInsideChroot);
 	}
 
+#ifdef false
 	void inferApplicationInfo(SpawnPreparationInfo &info) const {
 		info.codeRevision = readFromRevisionFile(info);
 		if (info.codeRevision.empty()) {
@@ -817,6 +817,7 @@ class Spawner {
 			return string();
 		}
 	}
+#endif
 
 	bool shouldLoadShellEnvvars(const Options &options, const SpawnPreparationInfo &preparation) const {
 		if (options.loadShellEnvvars) {
