From a85b6420c19ace35134f896e094e1971d8c7954b Mon Sep 17 00:00:00 2001
From: James Cole <james@firefly-iii.org>
Date: Fri, 1 Oct 2021 11:33:45 +0200
Subject: [PATCH] Make sure temp file is closed (and thus deleted)

---
 app/Helpers/Attachments/AttachmentHelper.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/app/Helpers/Attachments/AttachmentHelper.php b/app/Helpers/Attachments/AttachmentHelper.php
index d442d461376..b988220f21b 100644
--- a/app/Helpers/Attachments/AttachmentHelper.php
+++ b/app/Helpers/Attachments/AttachmentHelper.php
@@ -171,6 +171,7 @@ public function saveAttachmentFromApi(Attachment $attachment, string $content):
         $allowedMime = config('firefly.allowedMimes');
         if (!in_array($mime, $allowedMime, true)) {
             Log::error(sprintf('Mime type %s is not allowed for API file upload.', $mime));
+            fclose($resource);
 
             return false;
         }
