From e27b65b3bf8defa34db58919496056caf3850cd4 Mon Sep 17 00:00:00 2001
From: JerikoOne <jeriko.one@gmx.us>
Date: Sat, 7 Jul 2018 15:35:56 -0500
Subject: [PATCH] imap_quote_string make room for quotes

---
 imap/util.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/imap/util.c b/imap/util.c
index 57802b254d..fccfdbf453 100644
--- a/imap/util.c
+++ b/imap/util.c
@@ -814,8 +814,8 @@ void imap_quote_string(char *dest, size_t dlen, const char *src, bool quote_back
   const char *s = src;
 
   *pt++ = '"';
-  /* save room for trailing quote-char */
-  dlen -= 2;
+  /* save room for quote-chars */
+  dlen -= 3;
 
   for (; *s && dlen; s++)
   {
