From b5c6a1b407b7f8b952bded2789593b68b1876211 Mon Sep 17 00:00:00 2001
From: Etienne Samson <samson.etienne@gmail.com>
Date: Wed, 21 Dec 2016 21:24:33 +0100
Subject: [PATCH] http: check certificate validity before clobbering the error
 variable

---
 src/transports/http.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/transports/http.c b/src/transports/http.c
index f0efd956a8f..b4418f69ab3 100644
--- a/src/transports/http.c
+++ b/src/transports/http.c
@@ -602,13 +602,12 @@ static int http_connect(http_subtransport *t)
 	if ((!error || error == GIT_ECERTIFICATE) && t->owner->certificate_check_cb != NULL &&
 	    git_stream_is_encrypted(t->io)) {
 		git_cert *cert;
-		int is_valid;
+		int is_valid = (error == GIT_OK);
 
 		if ((error = git_stream_certificate(&cert, t->io)) < 0)
 			return error;
 
 		giterr_clear();
-		is_valid = error != GIT_ECERTIFICATE;
 		error = t->owner->certificate_check_cb(cert, is_valid, t->connection_data.host, t->owner->message_cb_payload);
 
 		if (error < 0) {
