From ab1283e8366c97a155d4e9ae58628a248458ea32 Mon Sep 17 00:00:00 2001
From: Madhura Jayaratne <madhura.cj@gmail.com>
Date: Mon, 29 Feb 2016 11:08:38 +1100
Subject: [PATCH] Fix XSS in User accounts page

Signed-off-by: Madhura Jayaratne <madhura.cj@gmail.com>
---
 libraries/server_privileges.lib.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libraries/server_privileges.lib.php b/libraries/server_privileges.lib.php
index 9ad749a177f..a60c999875e 100644
--- a/libraries/server_privileges.lib.php
+++ b/libraries/server_privileges.lib.php
@@ -3499,7 +3499,7 @@ function PMA_getUsersOverview($result, $db_rights, $pmaThemeImage, $text_dir)
         __('Export'), 'b_tblexport.png', 'export'
     );
     $html_output .= '<input type="hidden" name="initial" '
-        . 'value="' . (isset($_GET['initial']) ? $_GET['initial'] : '') . '" />';
+        . 'value="' . (isset($_GET['initial']) ? htmlspecialchars($_GET['initial']) : '') . '" />';
     $html_output .= '</div>'
         . '<div class="clear_both" style="clear:both"></div>';
 
