From 269320e0e0e06a1785e1a1556da769a34280b7e6 Mon Sep 17 00:00:00 2001
From: Peter Ivanov <peter@microweber.com>
Date: Fri, 22 May 2020 10:12:18 +0300
Subject: [PATCH] update

---
 userfiles/modules/users/controller.php | 1 -
 1 file changed, 1 deletion(-)

diff --git a/userfiles/modules/users/controller.php b/userfiles/modules/users/controller.php
index 540ea87125..a4abe2dafc 100644
--- a/userfiles/modules/users/controller.php
+++ b/userfiles/modules/users/controller.php
@@ -1,3 +1,2 @@
 <?php
 
-dd(User::all());
\ No newline at end of file
