From ee70db46f81afa582c96b887b73bcd2a86feda00 Mon Sep 17 00:00:00 2001
From: Rob Peters <ROb@DVBControl.com>
Date: Fri, 29 Mar 2019 15:04:04 +0100
Subject: [PATCH] Fixed possible SQL Injection Vulnerability (Thanks to Fabio
 Carretto!)

---
 main/WebServer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main/WebServer.cpp b/main/WebServer.cpp
index 6c538faf23..ece2155fd2 100644
--- a/main/WebServer.cpp
+++ b/main/WebServer.cpp
@@ -10772,7 +10772,7 @@ namespace http {
 				return;
 			}
 			std::vector<std::vector<std::string> > result;
-			result = m_sql.safe_queryBlob("SELECT Image FROM Floorplans WHERE ID=%s", idx.c_str());
+			result = m_sql.safe_queryBlob("SELECT Image FROM Floorplans WHERE ID=%d", atol(idx.c_str()));
 			if (result.empty())
 				return;
 			reply::set_content(&rep, result[0][0].begin(), result[0][0].end());
