From a9a0a8e9561c3487854c7cae42565d9652ec858b Mon Sep 17 00:00:00 2001
From: Pavan Kumar Sunkara <pavan.sss1991@gmail.com>
Date: Tue, 21 Sep 2021 16:19:42 +0530
Subject: [PATCH] Fix CVE-2021-3820

---
 lib/methods.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/methods.js b/lib/methods.js
index 00f83ce..450e63f 100644
--- a/lib/methods.js
+++ b/lib/methods.js
@@ -61,7 +61,7 @@ inflect.camelize = function (lower_case_and_underscored_word, first_letter_in_up
 inflect.underscore = function (camel_cased_word) {
   var self;
   self = util.string.gsub(camel_cased_word, /\./, '/');
-  self = util.string.gsub(self, /([A-Z]+)([A-Z][a-z])/, '$1_$2');
+  self = util.string.gsub(self, /([A-Z])([A-Z][a-z])/, '$1_$2');
   self = util.string.gsub(self, /([a-z\d])([A-Z])/, '$1_$2');
   self = util.string.gsub(self, /-/, '_');
   return self.toLowerCase();
@@ -230,5 +230,5 @@ inflect.tableize = function (class_name) {
 //
 //     "business".classify()       // => "Busines"
 inflect.classify = function (table_name) {
-  return inflect.camelize(inflect.singularize(util.string.gsub(table_name, /.*\./, '')));
+  return inflect.camelize(inflect.singularize(util.string.gsub(table_name, /^.*\./, '')));
 };
