From 75ea92decc16a5daf7f618f85dc621d1b534b5e1 Mon Sep 17 00:00:00 2001
From: nao-pon <hypweb@gmail.com>
Date: Mon, 31 May 2021 20:50:39 +0900
Subject: [PATCH] [VD:abstract] add `'phar:*' => 'text/x-php'` into
 'staticMineMap'

rel. #3295
---
 php/elFinderVolumeDriver.class.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/php/elFinderVolumeDriver.class.php b/php/elFinderVolumeDriver.class.php
index 3dd3389a3..b4e364a39 100644
--- a/php/elFinderVolumeDriver.class.php
+++ b/php/elFinderVolumeDriver.class.php
@@ -281,6 +281,7 @@ abstract class elFinderVolumeDriver
             'php5:*' => 'text/x-php',
             'php7:*' => 'text/x-php',
             'phtml:*' => 'text/x-php',
+            'phar:*' => 'text/x-php',
             'cgi:*' => 'text/x-httpd-cgi',
             'pl:*' => 'text/x-perl',
             'asp:*' => 'text/x-asap',
