From 56ccb03af4dddebc2b2defb348b6558783d5757e Mon Sep 17 00:00:00 2001
From: FXG387 <anton.myshenin@motorolasolutions.com>
Date: Sun, 4 Mar 2018 07:24:21 +0100
Subject: [PATCH] security update

---
 index.js     | 2 +-
 package.json | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/index.js b/index.js
index d8626fb..e30fb44 100644
--- a/index.js
+++ b/index.js
@@ -7,7 +7,7 @@ function getValueIgnoringKeyCase(obj, lookedKey) {
 module.exports.parse = (event, spotText) => {
     const boundary = getValueIgnoringKeyCase(event.headers, 'Content-Type').split('=')[1];
     const body = (event.isBase64Encoded ? Buffer.from(event.body, 'base64').toString('binary') : event.body)
-        .split(new RegExp(boundary))
+        .split(boundary)
         .filter(item => item.match(/Content-Disposition/))
         .map((item) => {
             if (item.match(/filename/)) {
diff --git a/package.json b/package.json
index 0db7dee..be8d531 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "aws-lambda-multipart-parser",
-  "version": "0.1.1",
+  "version": "0.1.2",
   "description": "Parser of multipart/form-data requests for AWS Lambda",
   "main": "index.js",
   "scripts": {
