From ad5a630174f08d279504bc51ebba8772fd71b86b Mon Sep 17 00:00:00 2001
From: Tatu Saloranta <tatu.saloranta@iki.fi>
Date: Fri, 18 Sep 2020 10:17:24 -0700
Subject: [PATCH] Add cve id for #2798

---
 release-notes/VERSION-2.x | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/release-notes/VERSION-2.x b/release-notes/VERSION-2.x
index 55fd50aec7..6a13e3ff79 100644
--- a/release-notes/VERSION-2.x
+++ b/release-notes/VERSION-2.x
@@ -11,7 +11,7 @@ Project: jackson-databind
 
 2.9.10.6 (24-Aug-2020)
 
-#2798: Block one more gadget type (xxx, CVE-xxxx-xxx)
+#2798: Block one more gadget type (com.pastdev.httpcomponents, CVE-2020-24750)
  (reported by Al1ex@knownsec)
 #2814: Block one more gadget type (Anteros-DBCP, CVE-2020-24616)
  (reported by ChenZhaojun)
