From 2119afbe74ee0c914c1d5c4c859c594c08b0ad42 Mon Sep 17 00:00:00 2001
From: Rob Peters <ROb@DVBControl.com>
Date: Fri, 29 Mar 2019 15:23:06 +0100
Subject: [PATCH] Do not allow enters/returns in arguments (thanks to Fabio
 Carretto)

---
 main/Helper.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/main/Helper.cpp b/main/Helper.cpp
index dddc1e65dd..3f120f628a 100644
--- a/main/Helper.cpp
+++ b/main/Helper.cpp
@@ -1014,6 +1014,8 @@ const char *szInsecureArgumentOptions[] = {
 	"$",
 	"<",
 	">",
+	"\n",
+	"\r",
 	NULL
 };
 
