From 25f15dd0fedc53688b25fd053ebbdffe3d5c1c56 Mon Sep 17 00:00:00 2001
From: Mathias Buus <mathiasbuus@gmail.com>
Date: Wed, 19 May 2021 19:59:39 +0200
Subject: [PATCH] do trim on encodingLength as well

---
 index.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/index.js b/index.js
index b6a52c6..9b9b4de 100644
--- a/index.js
+++ b/index.js
@@ -78,7 +78,7 @@ name.decode.bytes = 0
 
 name.encodingLength = function (n) {
   if (n === '.') return 1
-  return Buffer.byteLength(n) + 2
+  return Buffer.byteLength(n.replace(/^\.|\.$/gm, '')) + 2
 }
 
 const string = {}
