From b3645b4c6718cef6de7003f41aafe7bfcc0395d1 Mon Sep 17 00:00:00 2001
From: sphinx <sphinx@kronos>
Date: Tue, 25 Jul 2017 15:44:07 +0200
Subject: [PATCH] Escaping shell arguments

---
 components/filemanager/class.filemanager.php | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/components/filemanager/class.filemanager.php b/components/filemanager/class.filemanager.php
index a04d7790..ca2a0ba9 100755
--- a/components/filemanager/class.filemanager.php
+++ b/components/filemanager/class.filemanager.php
@@ -57,7 +57,7 @@ public function __construct($get, $post, $files)
             $this->rel_path .= "/";
         }
         if (!empty($get['query'])) {
-            $this->query = $get['query'];
+            $this->query = escapeshellarg($get['query']);
         }
         if (!empty($get['options'])) {
             $this->foptions = $get['options'];
@@ -71,10 +71,10 @@ public function __construct($get, $post, $files)
         }
         // Search
         if (!empty($post['search_string'])) {
-            $this->search_string = $post['search_string'];
+            $this->search_string = escapeshellarg($post['search_string']);
         }
         if (!empty($post['search_file_type'])) {
-            $this->search_file_type = $post['search_file_type'];
+            $this->search_file_type = escapeshellarg($post['search_file_type']);
         }
         // Create
         if (!empty($get['type'])) {
