From 03925b4707520e04ebbf5184445a1393def47344 Mon Sep 17 00:00:00 2001
From: Pierre RAMBAUD <pierre.rambaud86@gmail.com>
Date: Mon, 23 Mar 2020 11:47:22 +0100
Subject: [PATCH] Escape url_name parameter

---
 ps_facetedsearch.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ps_facetedsearch.php b/ps_facetedsearch.php
index b2bb71757..6b45f3ca6 100644
--- a/ps_facetedsearch.php
+++ b/ps_facetedsearch.php
@@ -1204,7 +1204,7 @@ public function checkLinksRewrite($params)
                 $params['errors'][] = Tools::displayError(
                     $this->trans(
                         '"%s" is not a valid url',
-                        [$urlNameLang],
+                        [Tools::safeOutput($urlNameLang, true)],
                         'Modules.Facetedsearch.Admin'
                     )
                 );
