From a7759f410b773a1dd57b0e1fb28112e1cd8b97bc Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Fri, 21 Jun 2019 16:51:38 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1608

---
 MagickCore/threshold.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/MagickCore/threshold.c b/MagickCore/threshold.c
index a3bb96a214..f9412bb52e 100644
--- a/MagickCore/threshold.c
+++ b/MagickCore/threshold.c
@@ -212,6 +212,8 @@ MagickExport Image *AdaptiveThresholdImage(const Image *image,
   threshold_image=CloneImage(image,0,0,MagickTrue,exception);
   if (threshold_image == (Image *) NULL)
     return((Image *) NULL);
+  if (width == 0)
+    return(threshold_image);
   status=SetImageStorageClass(threshold_image,DirectClass,exception);
   if (status == MagickFalse)
     {
