From b60d1ed0af37c50b91a40937825b4c61e8458095 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 30 Jan 2016 10:21:15 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/102

---
 coders/wpg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coders/wpg.c b/coders/wpg.c
index 4e4f8f1e89..e35a97c4d8 100644
--- a/coders/wpg.c
+++ b/coders/wpg.c
@@ -317,7 +317,7 @@ static void InsertRow(Image *image,unsigned char *p,ssize_t y,int bpp,
         q=QueueAuthenticPixels(image,0,y,image->columns,1,exception);
         if (q == (Quantum *) NULL)
           break;
-        for (x=0; x < ((ssize_t) image->columns-1); x+=2)
+        for (x=0; x < ((ssize_t) image->columns-1); x+=4)
         {
             index=ConstrainColormapIndex(image,(*p >> 6) & 0x3,exception);
             SetPixelIndex(image,index,q);
