From fe5155cf047252d1c4cb91602048bfa682af0ea7 Mon Sep 17 00:00:00 2001
From: jeanlf <jeanlf@gpac.io>
Date: Mon, 10 May 2021 11:18:59 +0200
Subject: [PATCH] fixed #1783 (fuzz)

---
 src/isomedia/isom_intern.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/isomedia/isom_intern.c b/src/isomedia/isom_intern.c
index e7c41d69b..74ee92f76 100644
--- a/src/isomedia/isom_intern.c
+++ b/src/isomedia/isom_intern.c
@@ -497,6 +497,7 @@ static GF_Err gf_isom_parse_movie_boxes_internal(GF_ISOFile *mov, u32 *boxType,
 					if (pos<0) pos=0;
 					gf_list_insert(mov->TopBoxes, brand, pos);
 				}
+				gf_isom_box_del(a);
 			}
 			break;
 
