From 37213e91d525c95df262712cba80d1497e395a58 Mon Sep 17 00:00:00 2001
From: Philip Gatzka <p.gatzka@shopware.com>
Date: Fri, 22 Oct 2021 16:06:21 +0200
Subject: [PATCH] SW-26367 - Add new CSP directive to .htaccess

---
 .htaccess      | 6 ++++++
 UPGRADE-5.7.md | 8 ++++++++
 2 files changed, 14 insertions(+)

diff --git a/.htaccess b/.htaccess
index d0268ab9394..8af59ab41ea 100644
--- a/.htaccess
+++ b/.htaccess
@@ -96,6 +96,12 @@ DirectoryIndex shopware.php
     </IfModule>
 </FilesMatch>
 
+<IfModule mod_headers.c>
+    <FilesMatch "\.(?i:svg)$">
+        Header set Content-Security-Policy "script-src 'none'"
+    </FilesMatch>
+</IfModule>
+
 # Disables auto directory index
 <IfModule mod_autoindex.c>
     Options -Indexes
diff --git a/UPGRADE-5.7.md b/UPGRADE-5.7.md
index ad66c724e1c..0321bdbfe48 100644
--- a/UPGRADE-5.7.md
+++ b/UPGRADE-5.7.md
@@ -2,6 +2,14 @@
 
 This changelog references changes done in Shopware 5.7 patch versions.
 
+## 5.7.6
+
+[View all changes from v5.7.5...v5.7.6](https://github.com/shopware/shopware/compare/v5.7.5...v5.7.6)
+
+### Additions
+
+* Added a new CSP directive to the default `.htaccess`
+
 ## 5.7.4
 
 [View all changes from v5.7.3...v5.7.4](https://github.com/shopware/shopware/compare/v5.7.3...v5.7.4)
