From f9d69eadd2b16b00d1a1f9f286122123f8e547dd Mon Sep 17 00:00:00 2001
From: Dirk Lemstra <dirk@git.imagemagick.org>
Date: Mon, 27 Nov 2017 21:18:43 +0100
Subject: [PATCH] Fixed possible credentials leaking reported by Alex Birsan.

---
 rsvg-io.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rsvg-io.c b/rsvg-io.c
index 7452b73..6e68b6f 100644
--- a/rsvg-io.c
+++ b/rsvg-io.c
@@ -130,7 +130,7 @@ _rsvg_io_get_file_path (const gchar * filename,
 {
     gchar *absolute_filename;
 
-    if (g_file_test (filename, G_FILE_TEST_EXISTS) || g_path_is_absolute (filename)) {
+    if (g_path_is_absolute (filename)) {
         absolute_filename = g_strdup (filename);
     } else {
         gchar *tmpcdir;
