From 51b0ae01709adc1e4a9245e158ef17b85a110960 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Thu, 26 Jan 2017 11:22:42 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/362

---
 coders/mat.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/coders/mat.c b/coders/mat.c
index e1623a82e8..cf2190bf37 100644
--- a/coders/mat.c
+++ b/coders/mat.c
@@ -1123,6 +1123,7 @@ RestoreMSCWarning
     BImgBuff = (unsigned char *) AcquireQuantumMemory((size_t) (ldblk),sizeof(double));    /* Ldblk was set in the check phase */
     if (BImgBuff == NULL)
       ThrowReaderException(ResourceLimitError,"MemoryAllocationFailed");
+    (void) ResetMagickMemory(BImgBuff,0,ldblk*sizeof(double));
 
     MinVal = 0;
     MaxVal = 0;
