From a33f7498f9052b50e8fe8c8422a11ba84474cb42 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 9 Jul 2017 08:10:10 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/551

---
 coders/dcm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coders/dcm.c b/coders/dcm.c
index 08d497f002..55377c4f9c 100644
--- a/coders/dcm.c
+++ b/coders/dcm.c
@@ -3270,7 +3270,7 @@ static Image *ReadDCMImage(const ImageInfo *image_info,ExceptionInfo *exception)
           if ((quantum != 0) && (length != 0))
             {
               if (length > GetBlobSize(image))
-                ThrowReaderException(CorruptImageError,
+                ThrowDCMException(CorruptImageError,
                   "InsufficientImageDataInFile");
               if (~length >= 1)
                 data=(unsigned char *) AcquireQuantumMemory(length+1,quantum*
