From 7d21d4223bdbe0306593309132b4263d7087d13b Mon Sep 17 00:00:00 2001
From: Isaac Bennetch <bennetch@gmail.com>
Date: Fri, 20 Sep 2019 18:02:38 -0400
Subject: [PATCH] Add ChangeLog entries for:

* 427fbed55d3154d96ecfc1c7784d49eaa3c04161
* 6e1bbc54c266c210e47e40946f4a6f314c8a36a7

Signed-off-by: Isaac Bennetch <bennetch@gmail.com>
---
 ChangeLog | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index 12c23c2dcf8..70d52881e2b 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -38,6 +38,8 @@ phpMyAdmin - ChangeLog
 - issue #15440 Fix page number is not being updated in the URL after saving a designer's page
 - issue        Fix reloading a designer's page
 - issue        Fix designer full screen mode button and text stuck when exiting full-screen mode
+- issue        Reduced possibility of causing heavy server traffic between the database and web servers
+- issue        Fix a situation where a server could be deleted while an administator is using the setup script
 
 4.9.0.1 (2019-06-04)
 - issue #14478 phpMyAdmin no longer streams the export data
