From 701ab8129ba9ea64f569daedca9a8603abad740f Mon Sep 17 00:00:00 2001
From: dequis <dx@dxzone.com.ar>
Date: Sun, 13 Nov 2016 16:52:43 -0300
Subject: [PATCH] imcb_file_send_start: handle ft attempts from non-existing
 users

---
 protocols/bee_ft.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/protocols/bee_ft.c b/protocols/bee_ft.c
index 27fd4eac..916b2e88 100644
--- a/protocols/bee_ft.c
+++ b/protocols/bee_ft.c
@@ -30,7 +30,7 @@ file_transfer_t *imcb_file_send_start(struct im_connection *ic, char *handle, ch
 	bee_t *bee = ic->bee;
 	bee_user_t *bu = bee_user_by_handle(bee, ic, handle);
 
-	if (bee->ui->ft_in_start) {
+	if (bee->ui->ft_in_start && bu) {
 		return bee->ui->ft_in_start(bee, bu, file_name, file_size);
 	} else {
 		return NULL;
