From 4f6af7f69f29df8555a18b9bb7b646906b45924d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jaime=20Pe=CC=81rez=20Crespo?= <jaime.perez@uninett.no>
Date: Fri, 2 Mar 2018 15:30:38 +0100
Subject: [PATCH] Be strict when checking return values.

---
 src/SAML2/HTTPRedirect.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/SAML2/HTTPRedirect.php b/src/SAML2/HTTPRedirect.php
index 733c6d0fe..ec40a15a9 100644
--- a/src/SAML2/HTTPRedirect.php
+++ b/src/SAML2/HTTPRedirect.php
@@ -226,7 +226,7 @@ public static function validateSignature(array $data, XMLSecurityKey $key)
             $key = Utils::castKey($key, $sigAlg);
         }
 
-        if (!$key->verifySignature($query, $signature)) {
+        if ($key->verifySignature($query, $signature) !== 1) {
             throw new \Exception('Unable to validate signature on query string.');
         }
     }
