From 768f60da87a3fa0b5561da5ade9309577c176d04 Mon Sep 17 00:00:00 2001
From: Aidan Holm <aidanholm@gmail.com>
Date: Thu, 13 Feb 2020 21:31:55 +0800
Subject: [PATCH] Disable lua bytecode loading

---
 crawl-ref/source/clua.cc | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/crawl-ref/source/clua.cc b/crawl-ref/source/clua.cc
index 7a4afa91aca..d336a278dc0 100644
--- a/crawl-ref/source/clua.cc
+++ b/crawl-ref/source/clua.cc
@@ -776,10 +776,27 @@ void CLua::init_lua()
     setregistry("__clua");
 }
 
+static int lua_loadstring(lua_State *ls)
+{
+    const auto lua = luaL_checkstring(ls, 1);
+    if (lua[0] == 0x1b)
+        abort();
+    lua_settop(ls, 0);
+    if (luaL_loadstring(ls, lua))
+    {
+        lua_pushnil(ls);
+        lua_insert(ls, 1);
+    }
+    return lua_gettop(ls);
+}
+
 void CLua::init_libraries()
 {
     lua_stack_cleaner clean(state());
 
+    lua_pushcfunction(_state, lua_loadstring);
+    lua_setglobal(_state, "loadstring");
+
     // Open Crawl bindings
     cluaopen_kills(_state);
     cluaopen_you(_state);
