From 1c7873b4f46f236c7cfed4fb280a897cb582e991 Mon Sep 17 00:00:00 2001
From: Laurent Destailleur <eldy@destailleur.fr>
Date: Fri, 1 May 2020 15:53:56 +0200
Subject: [PATCH] FIX Sanitizing menu parameter

---
 htdocs/core/get_menudiv.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/htdocs/core/get_menudiv.php b/htdocs/core/get_menudiv.php
index edebf8a31b45..af64d0659311 100644
--- a/htdocs/core/get_menudiv.php
+++ b/htdocs/core/get_menudiv.php
@@ -184,8 +184,8 @@
 }
 
 // Load the menu manager (only if not already done)
-$file_menu=$conf->standard_menu;
-if (GETPOST('menu')) $file_menu=GETPOST('menu');     // example: menu=eldy_menu.php
+$file_menu = $conf->standard_menu;
+if (GETPOST('menu', 'aZ09')) $file_menu = GETPOST('menu', 'aZ09');     // example: menu=eldy_menu.php
 if (! class_exists('MenuManager'))
 {
 	$menufound=0;
