From 2f0f2e8c355b75fa4407d23f85feea037d2bc4b6 Mon Sep 17 00:00:00 2001
From: "Alexander A. Klimov" <alexander.klimov@icinga.com>
Date: Mon, 8 Jun 2020 15:14:29 +0200
Subject: [PATCH] prepare-dirs: combine mkdir and chmod

---
 etc/initsystem/prepare-dirs.cmake | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/etc/initsystem/prepare-dirs.cmake b/etc/initsystem/prepare-dirs.cmake
index 99dc602234..4cef831936 100644
--- a/etc/initsystem/prepare-dirs.cmake
+++ b/etc/initsystem/prepare-dirs.cmake
@@ -26,12 +26,10 @@ getent group $ICINGA2_GROUP >/dev/null 2>&1 || (echo "Icinga group '$ICINGA2_GRO
 getent group $ICINGA2_COMMAND_GROUP >/dev/null 2>&1 || (echo "Icinga command group '$ICINGA2_COMMAND_GROUP' does not exist. Exiting." && exit 6)
 
 if [ ! -e "$ICINGA2_INIT_RUN_DIR" ]; then
-	mkdir "$ICINGA2_INIT_RUN_DIR"
-	mkdir "$ICINGA2_INIT_RUN_DIR"/cmd
+	mkdir -m 755 "$ICINGA2_INIT_RUN_DIR"
+	mkdir -m 2750 "$ICINGA2_INIT_RUN_DIR"/cmd
 fi
 
-chmod 755 "$ICINGA2_INIT_RUN_DIR"
-chmod 2750 "$ICINGA2_INIT_RUN_DIR"/cmd
 chown -R $ICINGA2_USER:$ICINGA2_COMMAND_GROUP "$ICINGA2_INIT_RUN_DIR"
 
 test -e "$ICINGA2_LOG_DIR" || install -m 750 -o $ICINGA2_USER -g $ICINGA2_COMMAND_GROUP -d "$ICINGA2_LOG_DIR"
