From 695fb61707e4286b64f6e446c189b449bd07d00a Mon Sep 17 00:00:00 2001
From: acrylian <wartgart@gmx.net>
Date: Tue, 11 Dec 2018 19:37:54 +0100
Subject: [PATCH] =?UTF-8?q?Follow=20up=E2=80=A6?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 zp-core/template-functions.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zp-core/template-functions.php b/zp-core/template-functions.php
index 8b04fa1694..4c90b337d8 100644
--- a/zp-core/template-functions.php
+++ b/zp-core/template-functions.php
@@ -4135,7 +4135,7 @@ function search_(way) {
 									if (in_array($key, $query_fields)) {
 										echo ' checked="checked" ';
 									}
-									echo ' value="' . html_encode($key) . '"  /> ' . $display . "</label></li>" . "\n";
+									echo ' value="' . html_encode($key) . '"  /> ' . html_encode($display) . "</label></li>" . "\n";
 								}
 								?>
 							</ul>
