From 10f6169e39510bbf5af913886037044458b9dc9b Mon Sep 17 00:00:00 2001
From: w8tcha <thewatcher@watchersnet.de>
Date: Thu, 14 Mar 2019 10:23:17 +0100
Subject: [PATCH] fixed #94

---
 oembed/plugin.js | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/oembed/plugin.js b/oembed/plugin.js
index 78108bb..06b52c5 100644
--- a/oembed/plugin.js
+++ b/oembed/plugin.js
@@ -79,9 +79,6 @@
                                 'div(embeddedContent,oembed-provider-*) blockquote': {
                                     attributes: '*'
                                 },
-                                'div(embeddedContent,oembed-provider-*) script': {
-                                    attributes: '*'
-                                },
                                 'div(embeddedContent,oembed-provider-*) embed': {
                                     attributes: '*'
                                 }
