From f8890a645c221823ac133dbf991f8a65ae50d637 Mon Sep 17 00:00:00 2001
From: akallabeth <akallabeth@posteo.net>
Date: Tue, 31 Mar 2020 09:23:42 +0200
Subject: [PATCH] Fixed #6005: Bounds checks in update_read_bitmap_data

---
 libfreerdp/core/update.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libfreerdp/core/update.c b/libfreerdp/core/update.c
index 08646968252..0bae7f37991 100644
--- a/libfreerdp/core/update.c
+++ b/libfreerdp/core/update.c
@@ -103,6 +103,9 @@ static BOOL update_read_bitmap_data(rdpUpdate* update, wStream* s, BITMAP_DATA*
 	{
 		if (!(bitmapData->flags & NO_BITMAP_COMPRESSION_HDR))
 		{
+			if (Stream_GetRemainingLength(s) < 8)
+				return FALSE;
+
 			Stream_Read_UINT16(s,
 			                   bitmapData->cbCompFirstRowSize); /* cbCompFirstRowSize (2 bytes) */
 			Stream_Read_UINT16(s,
