From 6ed536982953d870010d8fa065dccbffeb6cae50 Mon Sep 17 00:00:00 2001
From: xiphon <xiphon@protonmail.com>
Date: Sun, 11 Oct 2020 19:51:05 +0000
Subject: [PATCH] cmake: set CMAKE_SKIP_RPATH on Linux to fix empty RPATH token
 issue

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a88f13c42c..087d46c719 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -175,6 +175,7 @@ find_package(Boost 1.58 REQUIRED COMPONENTS
 		locale)
 
 if(UNIX AND NOT APPLE)
+    set(CMAKE_SKIP_RPATH ON)
     find_package(X11 REQUIRED)
     message(STATUS "X11_FOUND = ${X11_FOUND}")
     message(STATUS "X11_INCLUDE_DIR = ${X11_INCLUDE_DIR}")
