From 7ade4e37f24d6859987bb9f6232f604325633fdd Mon Sep 17 00:00:00 2001
From: Laurent Destailleur <eldy@destailleur.fr>
Date: Wed, 18 Apr 2018 12:23:49 +0200
Subject: [PATCH] FIX CVE-2018-10094

---
 htdocs/adherents/list.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/htdocs/adherents/list.php b/htdocs/adherents/list.php
index cae878829b71..68062b895a09 100644
--- a/htdocs/adherents/list.php
+++ b/htdocs/adherents/list.php
@@ -43,7 +43,7 @@
 $result=restrictedArea($user,'adherent');
 
 $filter=GETPOST("filter",'alpha');
-$statut=GETPOST("statut",'alpha');
+$statut=GETPOST("statut",'intcomma');
 $search=GETPOST("search",'alpha');
 $search_ref=GETPOST("search_ref",'alpha');
 $search_lastname=GETPOST("search_lastname",'alpha');
