From a6240a163cb787909703d9fc649cf861f60ddd7c Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 20 Feb 2016 09:51:04 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/131

---
 coders/mat.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/coders/mat.c b/coders/mat.c
index 93a9d8710a..9bc442274a 100644
--- a/coders/mat.c
+++ b/coders/mat.c
@@ -925,6 +925,7 @@ RestoreMSCWarning
   }
       }
     } while(z-- >= 2);
+    quantum_info=DestroyQuantumInfo(quantum_info);
 ExitLoop:
 
 
