From 20f4e3d87e84663d922b937842fddd9af1b68dd9 Mon Sep 17 00:00:00 2001
From: Stephen Stack <stephen@rconfig.com>
Date: Tue, 25 Feb 2020 09:34:21 +0000
Subject: [PATCH] updated head.inc.php for 3.9.4

---
 www/includes/head.inc.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/www/includes/head.inc.php b/www/includes/head.inc.php
index 76eb847..9e17a22 100644
--- a/www/includes/head.inc.php
+++ b/www/includes/head.inc.php
@@ -13,6 +13,7 @@
  */
 if (!$session->logged_in) {
     header("Location: " . $config_basedir . "login.php");
+    exit();
 }
 $host = gethostname();
 $ip = $_SERVER['SERVER_ADDR'];
@@ -58,4 +59,4 @@
         <script type="text/javascript" src="js/toastr/toastr.min.js" ></script>       
         <!-- Custom JS -->
         <script type="text/javascript" src="js/rconfigFunctions.js"></script>
-    </head>
\ No newline at end of file
+    </head>
