From df8a43fb4bdc9c858874f72773bcba597ae9432c Mon Sep 17 00:00:00 2001
From: Jamie Cameron <jcameron@webmin.com>
Date: Sat, 3 Aug 2019 22:41:37 -0700
Subject: [PATCH] Object names cannot contact special characters

---
 web-lib-funcs.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/web-lib-funcs.pl b/web-lib-funcs.pl
index bbd51bacc1..1e3260ff05 100755
--- a/web-lib-funcs.pl
+++ b/web-lib-funcs.pl
@@ -7307,7 +7307,7 @@ sub unserialise_variable
 elsif ($v[0] eq 'UNDEF') {
 	$rv = undef;
 	}
-elsif ($v[0] =~ /^OBJECT\s+(.*)$/) {
+elsif ($v[0] =~ /^OBJECT\s+([A-Za-z0-9_:]+)$/) {
 	# An object hash that we have to re-bless
 	my $cls = $1;
 	$rv = { };
