From 18bbf2043e53f69e0119d24f8ae4edb274afb9b2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hanno=20B=C3=B6ck?= <hanno@gentoo.org>
Date: Wed, 10 Jun 2020 10:15:22 +0200
Subject: [PATCH] limit data we read from file to avoid DoS via large files

---
 freewvs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/freewvs b/freewvs
index a7a1bac..80e46df 100755
--- a/freewvs
+++ b/freewvs
@@ -125,7 +125,7 @@ for fdir in opts.dirs:
                             file = open(mfile, errors='replace')
                         except OSError:
                             continue
-                        filestr = file.read()
+                        filestr = file.read(200000)
                         file.close()
 
                         if (('extra_match' in det
