From 2ce613f1e9fe8ffab7a4cb9d1384444622285335 Mon Sep 17 00:00:00 2001
From: FusionPBX <markjcrane@gmail.com>
Date: Tue, 13 Aug 2019 00:09:47 -0600
Subject: [PATCH] Update device_imports.php

---
 app/devices/device_imports.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/devices/device_imports.php b/app/devices/device_imports.php
index b1efe5322b5..87d3c0ebecd 100644
--- a/app/devices/device_imports.php
+++ b/app/devices/device_imports.php
@@ -363,7 +363,7 @@ function get_parent($schema,$table_name) {
 	echo "		".$text['description-import']."\n";
 	echo "	</td>\n";
 	echo "	<td valign='top' width='70%' align='right'>\n";
-	echo "		<input type='button' class='btn' name='' alt='".$text['button-back']."' onclick=\"window.location='devices.php?".$_GET["query_string"]."'\" value='".$text['button-back']."'>\n";
+	echo "		<input type='button' class='btn' name='' alt='".$text['button-back']."' onclick=\"window.location='devices.php'\" value='".$text['button-back']."'>\n";
 	//echo "		<input name='submit' type='submit' class='btn' id='import' value=\"".$text['button-import']."\">\n";
 	echo "	</td>\n";
 	echo "	</tr>\n";
