From bfa3b9610c83227894c92b0d312ad327fceb6241 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Fri, 21 Jun 2019 20:33:10 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1614

---
 coders/pnm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coders/pnm.c b/coders/pnm.c
index c10558d244..d3500b3ae0 100644
--- a/coders/pnm.c
+++ b/coders/pnm.c
@@ -1762,13 +1762,13 @@ static MagickBooleanType WritePNMImage(const ImageInfo *image_info,Image *image)
           {
             *q++=(unsigned char) (GetPixelLuma(image,p) >= (QuantumRange/2.0) ?
               '0' : '1');
-            *q++=' ';
             if ((q-pixels+1) >= (ssize_t) sizeof(pixels))
               {
                 *q++='\n';
                 (void) WriteBlob(image,q-pixels,pixels);
                 q=pixels;
               }
+            *q++=' ';
             p++;
           }
           *q++='\n';
