From bad372e654cdd549e7d786acbba0035ded54c607 Mon Sep 17 00:00:00 2001
From: Sebastian Hildebrandt <hildebrandt@plus-innovations.com>
Date: Tue, 19 May 2020 17:02:51 +0200
Subject: [PATCH] improved shell sanitation

---
 lib/util.js | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/util.js b/lib/util.js
index 19df5e7a..e7a87dc1 100644
--- a/lib/util.js
+++ b/lib/util.js
@@ -503,6 +503,9 @@ function sanitizeShellString(str) {
   result = result.replace(/\$/g, "");
   result = result.replace(/#/g, "");
   result = result.replace(/\\/g, "");
+  result = result.replace(/\t/g, "");
+  result = result.replace(/\n/g, "");
+  result = result.replace(/\"/g, "");
   return result
 }
 
