From ad18ece95fa064efc0995c4ab2c985f77fb166ec Mon Sep 17 00:00:00 2001
From: jeanlf <jeanlf@gpac.io>
Date: Wed, 1 Sep 2021 13:45:04 +0200
Subject: [PATCH] fixed #1904

---
 src/isomedia/hint_track.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/isomedia/hint_track.c b/src/isomedia/hint_track.c
index 304aa2cd8..385e74625 100644
--- a/src/isomedia/hint_track.c
+++ b/src/isomedia/hint_track.c
@@ -43,7 +43,7 @@ Bool IsHintTrack(GF_TrackBox *trak)
 u32 GetHintFormat(GF_TrackBox *trak)
 {
 	GF_HintMediaHeaderBox *hmhd = (GF_HintMediaHeaderBox *)trak->Media->information->InfoHeader;
-	if (hmhd->type != GF_ISOM_BOX_TYPE_HMHD)
+	if (!hmhd || (hmhd->type != GF_ISOM_BOX_TYPE_HMHD))
 		return 0;
 		
 	if (!hmhd || !hmhd->subType) {
