From 96e0aa35653ec0c66084a7e9343465e16c1f769b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Fri, 17 Jun 2016 09:21:52 +0200
Subject: [PATCH] Fix error reporting on invalid request data
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 setup/validate.php | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/setup/validate.php b/setup/validate.php
index 893e0b70fef..c53c0e73b1e 100644
--- a/setup/validate.php
+++ b/setup/validate.php
@@ -26,6 +26,9 @@
 $values = (array)$values;
 $result = PMA\libraries\config\Validator::validate($GLOBALS['ConfigFile'], $vids, $values, true);
 if ($result === false) {
-    $result = 'Wrong data or no validation for ' . $vids;
+    $result = sprintf(
+        __('Wrong data or no validation for %s'),
+        implode(',', $vids)
+    );
 }
 echo $result !== true ? json_encode($result) : '';
