From c334dd8499a681587dd4199e90b0aa0eba814c1d Mon Sep 17 00:00:00 2001
From: soruly <soruly@gmail.com>
Date: Tue, 28 Feb 2017 11:41:22 +0800
Subject: [PATCH] Fix #8 XSS issue

---
 index.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/index.php b/index.php
index 90ae205..524fa31 100755
--- a/index.php
+++ b/index.php
@@ -121,7 +121,7 @@
 <script src="/jquery.html5uploader.min.js"></script>
 <script src="/index.js"></script>
 <?php
-if(isset($_GET["url"])){
+if(isset($_GET["url"]) && filter_var($_GET["url"], FILTER_VALIDATE_URL)){
 echo '<script>
 document.querySelector("#autoSearch").checked = true;
 document.querySelector("#messageText").classList.remove("error");
