From 604588fc35c7585abb7a9e71f69bb82e4389fefc Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Fri, 21 Jun 2019 17:30:31 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1609

---
 MagickCore/threshold.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MagickCore/threshold.c b/MagickCore/threshold.c
index f9412bb52e..0c3a488d7e 100644
--- a/MagickCore/threshold.c
+++ b/MagickCore/threshold.c
@@ -212,7 +212,7 @@ MagickExport Image *AdaptiveThresholdImage(const Image *image,
   threshold_image=CloneImage(image,0,0,MagickTrue,exception);
   if (threshold_image == (Image *) NULL)
     return((Image *) NULL);
-  if (width == 0)
+  if ((width == 0) || (height == 0))
     return(threshold_image);
   status=SetImageStorageClass(threshold_image,DirectClass,exception);
   if (status == MagickFalse)
