From 1bc1fd0ff8c555841c78829217ac81fa0598255d Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 14 Feb 2016 09:23:34 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/131

---
 coders/mat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coders/mat.c b/coders/mat.c
index b5a3f1f970..93a9d8710a 100644
--- a/coders/mat.c
+++ b/coders/mat.c
@@ -864,7 +864,7 @@ RestoreMSCWarning
       return(DestroyImageList(image));
 
   /* ----- Load raster data ----- */
-    BImgBuff = (unsigned char *) AcquireQuantumMemory((size_t) (ldblk),sizeof(unsigned char));    /* Ldblk was set in the check phase */
+    BImgBuff = (unsigned char *) AcquireQuantumMemory((size_t) (ldblk),sizeof(double));    /* Ldblk was set in the check phase */
     if (BImgBuff == NULL)
       ThrowReaderException(ResourceLimitError,"MemoryAllocationFailed");
 
