From c57d3cc7b97b5f32801032f7bb222297aa97dfea Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Tue, 26 Jan 2016 15:10:18 +0100
Subject: [PATCH] Avoid invalid invocation of SQL parser
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 libraries/sql-parser/autoload.php | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libraries/sql-parser/autoload.php b/libraries/sql-parser/autoload.php
index 85f0d6c986b..da6bfe0a8cc 100644
--- a/libraries/sql-parser/autoload.php
+++ b/libraries/sql-parser/autoload.php
@@ -14,6 +14,9 @@
 namespace SqlParser\Autoload;
 
 if (!class_exists('SqlParser\\Autoload\\ClassLoader')) {
+    if (! file_exists('./libraries/sql-parser/ClassLoader.php')) {
+        die('Invalid invocation');
+    }
     include_once './libraries/sql-parser/ClassLoader.php';
 }
 
