From cf84be1ca1d9a08ad1341dfbf8df475b13a89072 Mon Sep 17 00:00:00 2001
From: Julio Montoya <gugli100@gmail.com>
Date: Fri, 14 May 2021 11:55:20 +0200
Subject: [PATCH] Documents: Add remove_XSS

---
 main/document/document.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main/document/document.php b/main/document/document.php
index 9ebfeeac5e2..17837084b75 100755
--- a/main/document/document.php
+++ b/main/document/document.php
@@ -1995,7 +1995,7 @@ function convertModal (id, format) {
             );
 
             // Document title with link and comment
-            $titleWithLink = $link.$session_img.'<br />'.$invisibility_span_open;
+            $titleWithLink = Security::remove_XSS($link.$session_img.'<br />'.$invisibility_span_open);
             $commentText = nl2br(htmlspecialchars($document_data['comment'], ENT_QUOTES, $charset));
             if (!empty($commentText)) {
                 $titleWithLink .= '<em>'.$commentText.'</em>';
