From 72271ef057e6e4a95c6128973902ea646f7b5462 Mon Sep 17 00:00:00 2001
From: Francisco Mancardi <francisco.mancardi@gmail.com>
Date: Sat, 4 Apr 2020 20:11:36 +0200
Subject: [PATCH] viewer (not used anymore) generates sec issue

---
 login.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/login.php b/login.php
index 4bdf0ba914..1c9925642e 100644
--- a/login.php
+++ b/login.php
@@ -151,7 +151,8 @@ function init_args() {
   $args->destination = urldecode($pParams['destination']);
   $args->loginform_token = urldecode($pParams['loginform_token']);
 
-  $args->viewer = $pParams['viewer']; 
+  // $args->viewer = $pParams['viewer']; 
+  $args->viewer = '';
 
   $k2c = array('ajaxcheck' => 'do','ajaxlogin' => 'do');
   if (isset($k2c[$pParams['action']]))  {
