From 578b0bc5e74e14de9ef4975f504dc698796bdf9c Mon Sep 17 00:00:00 2001
From: doowb <brian.woodward@gmail.com>
Date: Wed, 7 Feb 2018 11:04:06 -0500
Subject: [PATCH] exclude __proto__

---
 index.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/index.js b/index.js
index 60f7049..909fbef 100644
--- a/index.js
+++ b/index.js
@@ -23,6 +23,10 @@ function mixinDeep(target, objects) {
  */
 
 function copy(val, key) {
+  if (key === '__proto__') {
+    return;
+  }
+
   var obj = this[key];
   if (isObject(val) && isObject(obj)) {
     mixinDeep(obj, val);
