From b007dd3a048097d8f58949297f5b434612e1e1a3 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Thu, 6 Jul 2017 06:41:49 -0400
Subject: [PATCH] ...

---
 coders/mpc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/coders/mpc.c b/coders/mpc.c
index fc489552a1..092367a78f 100644
--- a/coders/mpc.c
+++ b/coders/mpc.c
@@ -1008,6 +1008,7 @@ ModuleExport size_t RegisterMPCImage(void)
   entry=SetMagickInfo("CACHE");
   entry->description=ConstantString("Magick Persistent Cache image format");
   entry->module=ConstantString("MPC");
+  entry->seekable_stream=MagickTrue;
   entry->stealth=MagickTrue;
   (void) RegisterMagickInfo(entry);
   entry=SetMagickInfo("MPC");
@@ -1015,6 +1016,7 @@ ModuleExport size_t RegisterMPCImage(void)
   entry->encoder=(EncodeImageHandler *) WriteMPCImage;
   entry->magick=(IsImageFormatHandler *) IsMPC;
   entry->description=ConstantString("Magick Persistent Cache image format");
+  entry->seekable_stream=MagickTrue;
   entry->module=ConstantString("MPC");
   (void) RegisterMagickInfo(entry);
   return(MagickImageCoderSignature);
