From c3812075e30c76a800a961e7291087d357403f6b Mon Sep 17 00:00:00 2001
From: Jason Dillaman <dillaman@redhat.com>
Date: Mon, 24 Sep 2018 10:09:54 -0400
Subject: [PATCH] rbd-target-api: disable the built-in werkzeug debugger

Signed-off-by: Jason Dillaman <dillaman@redhat.com>
---
 rbd-target-api.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/rbd-target-api.py b/rbd-target-api.py
index 450f96f..394c7e0 100755
--- a/rbd-target-api.py
+++ b/rbd-target-api.py
@@ -2004,7 +2004,8 @@ def main():
     # request makes further api requests
     app.run(host='0.0.0.0',
             port=settings.config.api_port,
-            debug=True,
+            debug=settings.config.debug,
+            use_evalex=False,
             threaded=True,
             use_reloader=False,
             ssl_context=context)
