From 42b5315b7c4b64f1839eb62fd9ba9e182034e9c9 Mon Sep 17 00:00:00 2001
From: winmin <shannanshuibe_i@163.com>
Date: Mon, 22 Jun 2020 21:32:06 +0800
Subject: [PATCH] add totBytes check

---
 silk/test/Decoder.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/silk/test/Decoder.c b/silk/test/Decoder.c
index 5a4fd93..2d33a1a 100644
--- a/silk/test/Decoder.c
+++ b/silk/test/Decoder.c
@@ -361,6 +361,12 @@ int main( int argc, char* argv[] )
         for( i = 0; i < MAX_LBRR_DELAY; i++ ) {
             totBytes += nBytesPerPacket[ i + 1 ];
         }
+        /* Check if the received totBytes is valid */
+        if (totBytes < 0 || totBytes > sizeof(payload))
+        {
+            fprintf( stderr, "\rPackets decoded:             %d", totPackets );
+            return -1;
+        }
         SKP_memmove( payload, &payload[ nBytesPerPacket[ 0 ] ], totBytes * sizeof( SKP_uint8 ) );
         payloadEnd -= nBytesPerPacket[ 0 ];
         SKP_memmove( nBytesPerPacket, &nBytesPerPacket[ 1 ], MAX_LBRR_DELAY * sizeof( SKP_int16 ) );
@@ -453,6 +459,15 @@ int main( int argc, char* argv[] )
         for( i = 0; i < MAX_LBRR_DELAY; i++ ) {
             totBytes += nBytesPerPacket[ i + 1 ];
         }
+
+        /* Check if the received totBytes is valid */
+        if (totBytes < 0 || totBytes > sizeof(payload))
+        {
+            
+            fprintf( stderr, "\rPackets decoded:              %d", totPackets );
+            return -1;
+        }
+        
         SKP_memmove( payload, &payload[ nBytesPerPacket[ 0 ] ], totBytes * sizeof( SKP_uint8 ) );
         payloadEnd -= nBytesPerPacket[ 0 ];
         SKP_memmove( nBytesPerPacket, &nBytesPerPacket[ 1 ], MAX_LBRR_DELAY * sizeof( SKP_int16 ) );
