From 364732e309cccb3fb56c938ed8d8bc0e04a3ca98 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Mon, 20 Jun 2016 10:10:31 +0200
Subject: [PATCH] Escape error message from server
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 js/console.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/console.js b/js/console.js
index 802e8e1f4c9..79aedb448eb 100644
--- a/js/console.js
+++ b/js/console.js
@@ -201,7 +201,7 @@ var PMA_console = {
                 } catch (e) {
                     console.log("Invalid JSON!" + e.message);
                     if (AJAX.xhr && AJAX.xhr.status === 0 && AJAX.xhr.statusText !== 'abort') {
-                        PMA_ajaxShowMessage($('<div />',{'class':'error','html':PMA_messages.strRequestFailed+' ( '+AJAX.xhr.statusText+' )'}));
+                        PMA_ajaxShowMessage($('<div />',{'class':'error','html':PMA_messages.strRequestFailed+' ( '+escapeHtml(AJAX.xhr.statusText)+' )'}));
                         AJAX.active = false;
                         AJAX.xhr = null;
                     }
