From da9d86eab55cbf608d5c916b8b690f5b76bca462 Mon Sep 17 00:00:00 2001
From: Young Xiao <YangX92@hotmail.com>
Date: Sat, 16 Mar 2019 17:42:14 +0800
Subject: [PATCH] decompileAction: Prevent heap buffer overflow and underflow
 with using OpCode

---
 util/decompile.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/util/decompile.c b/util/decompile.c
index e9341356..1df56da5 100644
--- a/util/decompile.c
+++ b/util/decompile.c
@@ -3202,7 +3202,6 @@ decompileCAST(int n, SWF_ACTION *actions, int maxn)
 int
 decompileAction(int n, SWF_ACTION *actions, int maxn)
 {
-	if( n > maxn ) SWF_error("Action overflow!!");
 
 #ifdef DEBUG
 	fprintf(stderr,"%d:\tACTION[%3.3d]: %s\n",
@@ -3210,7 +3209,7 @@ decompileAction(int n, SWF_ACTION *actions, int maxn)
 	        actionName(actions[n].SWF_ACTIONRECORD.ActionCode));
 #endif
 
-	switch(actions[n].SWF_ACTIONRECORD.ActionCode)
+	switch(OpCode(actions, n, maxn))
 	{
 	case SWFACTION_END:
 		return 0;
