From 7d0baab0d361c377af84994f95ba76c280048548 Mon Sep 17 00:00:00 2001
From: Aravind SV <mail@arvindsv.com>
Date: Fri, 19 Feb 2021 10:46:02 +0000
Subject: [PATCH] #000 - Add missing Content-Type check

---
 .../go/apiv1/backupconfig/BackupConfigControllerV1.java       | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/api/api-backup-config-v1/src/main/java/com/thoughtworks/go/apiv1/backupconfig/BackupConfigControllerV1.java b/api/api-backup-config-v1/src/main/java/com/thoughtworks/go/apiv1/backupconfig/BackupConfigControllerV1.java
index e0f509a7cb7..27b6c38f248 100644
--- a/api/api-backup-config-v1/src/main/java/com/thoughtworks/go/apiv1/backupconfig/BackupConfigControllerV1.java
+++ b/api/api-backup-config-v1/src/main/java/com/thoughtworks/go/apiv1/backupconfig/BackupConfigControllerV1.java
@@ -65,7 +65,9 @@ public String controllerBasePath() {
     public void setupRoutes() {
         path(controllerBasePath(), () -> {
             before("", mimeType, this::setContentType);
-
+            before("/*", mimeType, this::setContentType);
+            before("", mimeType, this::verifyContentType);
+            before("/*", mimeType, this::verifyContentType);
 
             // change the line below to enable appropriate security
             before("", mimeType, this.apiAuthenticationHelper::checkAdminUserAnd403);
