From 983bda1f0fa5fa86884381208274198a620f006e Mon Sep 17 00:00:00 2001
From: Anders Brander <anders@brander.dk>
Date: Mon, 11 May 2015 14:59:52 +0200
Subject: [PATCH] Avoid overflow in ljpeg_start().

---
 plugins/load-dcraw/dcraw.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/plugins/load-dcraw/dcraw.cc b/plugins/load-dcraw/dcraw.cc
index 8ad088cd..b1c4b10a 100644
--- a/plugins/load-dcraw/dcraw.cc
+++ b/plugins/load-dcraw/dcraw.cc
@@ -890,7 +890,8 @@ struct jhead {
 
 int CLASS ljpeg_start (struct jhead *jh, int info_only)
 {
-  int c, tag, len;
+  int c, tag;
+  ushort len;
   uchar data[0x10000];
   const uchar *dp;
 
