From 51cf24b2a39138e6a7b5739ef59eb38cd7c39763 Mon Sep 17 00:00:00 2001
From: Matt Pass <matt@mattpass.com>
Date: Mon, 17 Jan 2022 12:38:29 +0000
Subject: [PATCH] rXSS cleaned username in editor info display

---
 editor.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/editor.php b/editor.php
index 417ba43f..7b7fa26f 100644
--- a/editor.php
+++ b/editor.php
@@ -150,7 +150,7 @@
             ?>
             <h2><?php echo $t['multi-user']; ?></h2>
             <span class="heading"><?php echo $t['Username']; ?></span><br>
-            <?php echo $_SESSION['username'];?><br><br>
+            <?php echo xssClean($_SESSION['username'], "html");?><br><br>
             <?php
         }
         ?>
