From 2c55b278a491bf45313ecc0825cf800e2f7ff5c1 Mon Sep 17 00:00:00 2001
From: Guillermo Rauch <rauchg@gmail.com>
Date: Wed, 30 Mar 2016 08:30:14 -0700
Subject: [PATCH] default `rejectUnauthorized` to `true`

---
 lib/socket.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/socket.js b/lib/socket.js
index 9a365635..c1364994 100644
--- a/lib/socket.js
+++ b/lib/socket.js
@@ -90,7 +90,7 @@ function Socket (uri, opts) {
   this.cert = opts.cert || null;
   this.ca = opts.ca || null;
   this.ciphers = opts.ciphers || null;
-  this.rejectUnauthorized = opts.rejectUnauthorized === undefined ? null : opts.rejectUnauthorized;
+  this.rejectUnauthorized = opts.rejectUnauthorized === undefined ? true : opts.rejectUnauthorized;
 
   // other options for Node.js client
   var freeGlobal = typeof global === 'object' && global;
