From f17bfd9f1da05f30f23a4dd05e9d2363e1406948 Mon Sep 17 00:00:00 2001
From: radare <pancake@nopcode.org>
Date: Thu, 22 Nov 2018 10:48:39 +0100
Subject: [PATCH] Fix #12242 - Crash in x86.nz assembler (#12266)

---
 libr/asm/p/asm_x86_nz.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libr/asm/p/asm_x86_nz.c b/libr/asm/p/asm_x86_nz.c
index 8d4e28941f4..812d13d3691 100644
--- a/libr/asm/p/asm_x86_nz.c
+++ b/libr/asm/p/asm_x86_nz.c
@@ -2023,9 +2023,9 @@ static int opmov(RAsm *a, ut8 *data, const Opcode *op) {
 			if (op->operands[1].scale[0] == 0) {
 				return -1;
 			}
-			data[l++] = SEG_REG_PREFIXES[op->operands[1].regs[0]];
+			data[l++] = SEG_REG_PREFIXES[op->operands[1].regs[0] % 6];
 			data[l++] = 0x8b;
-			data[l++] = op->operands[0].reg << 3 | 0x5;
+			data[l++] = (((ut32)op->operands[0].reg) << 3) | 0x5;
 			data[l++] = offset;
 			data[l++] = offset >> 8;
 			data[l++] = offset >> 16;
