From a01cbc8c3d3b8bb7720bf3ff234671b4c0e1859c Mon Sep 17 00:00:00 2001
From: "Ycarus (Yannick Chabanois)" <ycarus@zugaina.org>
Date: Thu, 8 Apr 2021 17:37:19 +0200
Subject: [PATCH] Fix timing attacks potential vulnerability

---
 omr-admin.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/omr-admin.py b/omr-admin.py
index 802136e..9cf4395 100755
--- a/omr-admin.py
+++ b/omr-admin.py
@@ -776,7 +776,7 @@ def set_lastchange(sync=0):
 fake_users_db = omr_config_data['users'][0]
 
 def verify_password(plain_password, user_password):
-    if plain_password == user_password:
+    if secrets.compare_digest(plain_password,user_password):
         LOG.debug("password true")
         return True
     return False
