From 55e6dc49f1a381d9d511ee2f888fdc3e3c3e3953 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Fri, 21 Jun 2019 16:52:14 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/1608

---
 magick/threshold.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/magick/threshold.c b/magick/threshold.c
index 763eb8552d..a53e839e6c 100644
--- a/magick/threshold.c
+++ b/magick/threshold.c
@@ -202,6 +202,8 @@ MagickExport Image *AdaptiveThresholdImage(const Image *image,
   threshold_image=CloneImage(image,0,0,MagickTrue,exception);
   if (threshold_image == (Image *) NULL)
     return((Image *) NULL);
+  if (width == 0)
+    return(threshold_image);
   if (SetImageStorageClass(threshold_image,DirectClass) == MagickFalse)
     {
       InheritException(exception,&threshold_image->exception);
