From 75a55824012406a08c4debf5ddb7ae41c32a7dbc Mon Sep 17 00:00:00 2001
From: Madhura Jayaratne <madhura.cj@gmail.com>
Date: Tue, 19 Jan 2016 10:12:32 +1100
Subject: [PATCH] Fix XSS in DB_search.php

Signed-off-by: Madhura Jayaratne <madhura.cj@gmail.com>
---
 libraries/DbSearch.class.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libraries/DbSearch.class.php b/libraries/DbSearch.class.php
index bce102f90d4..c79beaddf2c 100644
--- a/libraries/DbSearch.class.php
+++ b/libraries/DbSearch.class.php
@@ -344,7 +344,8 @@ private function _getResultsRow($each_table, $newsearchsqls, $odd_row, $res_cnt)
             $browse_result_path = 'sql.php' . PMA_URL_getCommon($this_url_params);
             $html_output .= '<td><a name="browse_search" class="ajax" href="'
                 . $browse_result_path . '" onclick="loadResult(\''
-                . $browse_result_path . '\',\'' . $each_table . '\',\''
+                . $browse_result_path . '\',\''
+                . PMA_escapeJsString(htmlspecialchars($each_table)) . '\',\''
                 . PMA_URL_getCommon(
                     array(
                         'db' => $GLOBALS['db'], 'table' => $each_table
