From b35530fa0681b27eba084de5527037ebfb397422 Mon Sep 17 00:00:00 2001
From: pancake <pancake@nopcode.org>
Date: Wed, 18 Apr 2018 13:15:11 +0200
Subject: [PATCH] Fix oobread in avr

---
 libr/anal/p/anal_avr.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libr/anal/p/anal_avr.c b/libr/anal/p/anal_avr.c
index 9361b404a7e..4e62158513c 100644
--- a/libr/anal/p/anal_avr.c
+++ b/libr/anal/p/anal_avr.c
@@ -1526,6 +1526,9 @@ OPCODE_DESC opcodes[] = {
 
 static OPCODE_DESC* avr_op_analyze(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *buf, int len, CPU_MODEL *cpu) {
 	OPCODE_DESC *opcode_desc;
+	if (len < 2) {
+		return NULL;
+	}
 	ut16 ins = (buf[1] << 8) | buf[0];
 	int fail;
 	char *t;
