From 9706d125b42481178d2b423de245f5d2faadbf40 Mon Sep 17 00:00:00 2001
From: mdipierro <massimo.dipierro@gmail.com>
Date: Mon, 14 Mar 2016 14:52:41 -0500
Subject: [PATCH] DO NOT LEAK GLOBALS SETTINGS INTO request object

---
 gluon/globals.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/gluon/globals.py b/gluon/globals.py
index 5581ade84..4f9ff8110 100644
--- a/gluon/globals.py
+++ b/gluon/globals.py
@@ -186,7 +186,6 @@ def __init__(self, env):
         Storage.__init__(self)
         self.env = Storage(env)
         self.env.web2py_path = global_settings.applications_parent
-        self.env.update(global_settings)
         self.cookies = Cookie.SimpleCookie()
         self._get_vars = None
         self._post_vars = None
@@ -202,7 +201,6 @@ def __init__(self, env):
         self.is_restful = False
         self.is_https = False
         self.is_local = False
-        self.global_settings = settings.global_settings
         self._uuid = None
 
     def parse_get_vars(self):
