From 3284eb342cd0ed1a18786e3fcdf0cdd7e76676bd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Tue, 1 Aug 2017 11:16:47 +0200
Subject: [PATCH] lib/puny_decode: Fix integer overflow (found by fuzzing)

---
 lib/puny_decode.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/puny_decode.c b/lib/puny_decode.c
index 3a1dd4f..7fb5305 100644
--- a/lib/puny_decode.c
+++ b/lib/puny_decode.c
@@ -93,10 +93,10 @@ enum { base = 36, tmin = 1, tmax = 26, skew = 38, damp = 700,
 /* point (for use in representing integers) in the range 0 to */
 /* base-1, or base if cp does not represent a value.          */
 
-static punycode_uint decode_digit(punycode_uint cp)
+static unsigned decode_digit(int cp)
 {
-  return  cp - 48 < 10 ? cp - 22 :  cp - 65 < 26 ? cp - 65 :
-          cp - 97 < 26 ? cp - 97 :  base;
+  return (unsigned) (cp - 48 < 10 ? cp - 22 :  cp - 65 < 26 ? cp - 65 :
+         cp - 97 < 26 ? cp - 97 :  base);
 }
 
 /*** Platform-specific constants ***/
-- 
GitLab

