From 16853b6973a1e72fee2b7cccda85472cb9951305 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Tue, 1 Aug 2017 11:15:10 +0200
Subject: [PATCH] lib/bidi: Fix integer overflow (found by fuzzing)

---
 lib/bidi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/bidi.c b/lib/bidi.c
index 7582981..47b6ff0 100644
--- a/lib/bidi.c
+++ b/lib/bidi.c
@@ -30,6 +30,7 @@
 
 #include "idn2.h"
 
+#include <sys/types.h>
 #include <stdbool.h>
 
 #include "bidi.h"
@@ -39,7 +40,7 @@
 static bool
 _isBidi (const uint32_t *label, size_t llen)
 {
-  while (llen-- > 0) {
+  for (; (ssize_t) llen > 0; llen--) {
     int bc = uc_bidi_category (*label++);
 
     if (bc == UC_BIDI_R || bc == UC_BIDI_AL || bc == UC_BIDI_AN)
-- 
GitLab

