From 9347b5c01dc52682cb6be11539d9b7ebceae4416 Mon Sep 17 00:00:00 2001
From: JerikoOne <jeriko.one@gmx.us>
Date: Fri, 13 Jul 2018 12:24:58 -0700
Subject: [PATCH] Handle NO response without message properly

---
 imap/command.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imap/command.c b/imap/command.c
index 82bf54c4..bb267f62 100644
--- a/imap/command.c
+++ b/imap/command.c
@@ -586,7 +586,7 @@ static int cmd_handle_untagged (IMAP_DATA* idata)
     dprint (2, (debugfile, "Handling untagged NO\n"));
 
     /* Display the warning message from the server */
-    mutt_error ("%s", s+3);
+    mutt_error ("%s", s+2);
     mutt_sleep (2);
   }
 
-- 
GitLab

