From 946565d8eb05fbf7970ea366e817581bb5a90910 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@gnutls.org>
Date: Tue, 16 Jan 2018 18:45:12 +0100
Subject: [PATCH] released 4.13

Signed-off-by: Nikos Mavrogiannopoulos <nmav@gnutls.org>
---
 NEWS           | 2 +-
 lib/libtasn1.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/NEWS b/NEWS
index f9ac241..3fac146 100644
--- a/NEWS
+++ b/NEWS
@@ -1,6 +1,6 @@
 GNU Libtasn1 NEWS                                     -*- outline -*-
 
-* Noteworthy changes in release 4.13 (unreleased) [stable]
+* Noteworthy changes in release 4.13 (released 2018-01-16) [stable]
 - On indefinite string decoding, set a maximum level of allowed recursions
   (3) to protect the BER decoder from a stack exhaustion.
 
diff --git a/lib/libtasn1.h b/lib/libtasn1.h
index 626b11e..ea26b78 100644
--- a/lib/libtasn1.h
+++ b/lib/libtasn1.h
@@ -44,7 +44,7 @@ extern "C"
 {
 #endif
 
-#define ASN1_VERSION "4.12"
+#define ASN1_VERSION "4.13"
 
 #if defined(__GNUC__) && !defined(ASN1_INTERNAL_BUILD)
 # define _ASN1_GCC_VERSION (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
-- 
GitLab

