From 614117ef6e4c60e1950d742e3edf0a0ef8d389de Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Wed, 24 Jul 2019 11:19:20 +0200
Subject: [PATCH] Update NEWS for CVE-2019-12290

---
 NEWS | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/NEWS b/NEWS
index f5f6d6f..46c3b3f 100644
--- a/NEWS
+++ b/NEWS
@@ -3,6 +3,14 @@ Copyright (C) 2011-2017 Simon Josefsson
 Copyright (C) 2018-2019 Tim Ruehsen
 See the end for copying conditions.
 
+* Version 2.x.x (unreleased)
+
+** Mitre has assigned CVE-2019-12290 which was fixed by
+   the roundtrip feature introduced in 2.2.0 (commit 241e8f48)
+
+** Update the data tables from Unicode 6.3.0 to Unicode 11.0
+
+
 * Version 2.2.0 (released 2019-05-23)
 
 ** Perform A-Label roundtrip for lookup functions by default
-- 
GitLab

