From 5b595e8e52653f6c5726a4cdd8fddeb6e83804d2 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@gnutls.org>
Date: Tue, 31 Mar 2020 07:40:21 +0200
Subject: [PATCH] NEWS: updated for release

Signed-off-by: Nikos Mavrogiannopoulos <nmav@gnutls.org>
---
 NEWS | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/NEWS b/NEWS
index 5152049da4..50cb2253a5 100644
--- a/NEWS
+++ b/NEWS
@@ -5,11 +5,12 @@ Copyright (C) 2000-2016 Free Software Foundation, Inc.
 Copyright (C) 2013-2019 Nikos Mavrogiannopoulos
 See the end for copying conditions.
 
-* Version 3.6.13 (unreleased)
+* Version 3.6.13 (released 2020-03-31)
 
 ** libgnutls: Fix a DTLS-protocol regression (caused by TLS1.3 support), since 3.6.3.
    The DTLS client would not contribute any randomness to the DTLS negotiation,
-   breaking the security guarantees of the DTLS protocol (#960) [CVSS: high]
+   breaking the security guarantees of the DTLS protocol (#960)
+   [GNUTLS-SA-2020-03-31, CVSS: high]
 
 ** libgnutls: Added new APIs to access KDF algorithms (#813).
 
-- 
GitLab

