From d12d6ea7b025d3b3f81c8a71063fe9f89e0c4bf1 Mon Sep 17 00:00:00 2001
From: Jonny Bradley <jmail@nospaces.net>
Date: Wed, 29 Jul 2020 10:18:10 +0100
Subject: [PATCH] [FIX] Improve filter pattern

---
 lib/core/TikiFilter/PreventXss.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/core/TikiFilter/PreventXss.php b/lib/core/TikiFilter/PreventXss.php
index 3fddd85a726..22818535640 100644
--- a/lib/core/TikiFilter/PreventXss.php
+++ b/lib/core/TikiFilter/PreventXss.php
@@ -150,7 +150,7 @@ class TikiFilter_PreventXss implements Laminas\Filter\FilterInterface
 		// keep replacing as long as the previous round replaced something
 		while ($this->RemoveXSSchars($val)
 			|| $this->RemoveXSSregexp($ra_as_tag_only, $val, '(\<|\[\\\\xC0\]\[\\\\xBC\])\??')
-			|| $this->RemoveXSSregexp($ra_as_attribute, $val, '\s')
+			|| $this->RemoveXSSregexp($ra_as_attribute, $val, '[\s\/"\']')
 			|| $this->RemoveXSSregexp($ra_as_content, $val, '[\.\\\\+\*\?\[\^\]\$\(\)\{\}\=\!\<\|\:;\-\/`#"\']', '(?!\s*[a-z0-9])', true)
 			|| $this->RemoveXSSregexp($ra_javascript, $val, '', ':', true)
 	///		|| RemoveXSSregexp($ra_style, $val, '[^a-z0-9]', '=') // Commented as it has been considered as a bit too aggressive
-- 
GitLab

