From 6d854cef4211cdcdbc7446c978f23ab859847cdd Mon Sep 17 00:00:00 2001
From: Benjamin Dauvergne <bdauvergne@entrouvert.com>
Date: Thu, 24 Apr 2014 12:31:13 +0200
Subject: lasso/xml/tools.c: fix reference to unitialized memory


diff --git a/lasso/xml/tools.c b/lasso/xml/tools.c
index 1bca859..2243e08 100644
--- a/lasso/xml/tools.c
+++ b/lasso/xml/tools.c
@@ -2461,7 +2461,7 @@ get_or_define_ns(xmlNode *xmlnode, const xmlChar *ns_uri, const xmlChar *advised
 		return ns;
 	/* Try with the advised prefix */
 	if (advised_prefix) {
-		ns = xmlSearchNs(NULL, xmlnode, BAD_CAST prefix);
+		ns = xmlSearchNs(NULL, xmlnode, BAD_CAST advised_prefix);
 		if (! ns) { /* If not taken, use it */
 			return xmlNewNs(xmlnode, ns_uri, BAD_CAST advised_prefix);
 		}
-- 
cgit v0.10.2

